<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Steve Pederson" />


<title>Differential Expression</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">20170327_Psen2S4Ter_RNASeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential Expression</h1>
<h4 class="author">Steve Pederson</h4>
<h4 class="date">25 January, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-01-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>20170327_Psen2S4Ter_RNASeq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200119code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200119)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200119code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomUofABioinformaticsHub20170327Psen2S4TerRNASeqtree07a6053b4c3f733eccd3a7d7a76419fb3c255b2atargetblank07a6053a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/tree/07a6053b4c3f733eccd3a7d7a76419fb3c255b2a" target="_blank">07a6053</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomUofABioinformaticsHub20170327Psen2S4TerRNASeqtree07a6053b4c3f733eccd3a7d7a76419fb3c255b2atargetblank07a6053a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/3_Enrichment.Rmd
    Untracked:  analysis/figure/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/07a6053b4c3f733eccd3a7d7a76419fb3c255b2a/analysis/2_DifferentialExpression.Rmd" target="_blank">07a6053</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
<td>
Corrected paths
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/09180e5a4d25468a130dff1e8b8b0e5be091f2fa/analysis/2_DifferentialExpression.Rmd" target="_blank">09180e5</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
<td>
Added data export
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/fca554f4e208e95d1aa206c8086e2b44597e4d52/analysis/2_DifferentialExpression.Rmd" target="_blank">fca554f</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
<td>
Tweaked explanations &amp; labels
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/be95a600302098eba9391210983d011dcff1ee3a/analysis/2_DifferentialExpression.Rmd" target="_blank">be95a60</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
<td>
Finished first pass of DE Analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/be95a600302098eba9391210983d011dcff1ee3a/docs/2_DifferentialExpression.html" target="_blank">be95a60</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
<td>
Finished first pass of DE Analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/f04bb47164970b13abe0de3b1476050daa64b258/docs/2_DifferentialExpression.html" target="_blank">f04bb47</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
<td>
Minor typos
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/analysis/2_DifferentialExpression.Rmd" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
<td>
Tweaks
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7fea1563b2a78daf45cb02410457f896a3c53143/analysis/2_DifferentialExpression.Rmd" target="_blank">7fea156</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
<td>
Added rRNA checks &amp; heatmaps
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/9bff5161683ed7d4da70a0d339dbe7895142271b/docs/2_DifferentialExpression.html" target="_blank">9bff516</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-24
</td>
<td>
Added analysis without CQN
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7b680b27fab5695534596b62e9eefd48c5e1dcb3/analysis/2_DifferentialExpression.Rmd" target="_blank">7b680b2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-24
</td>
<td>
Added analysis without CQN
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/95ccc9066e528326f3fb6c4c248f6de5d4a8991b/docs/2_DifferentialExpression.html" target="_blank">95ccc90</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
<td>
Compiled DE so far
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/1858f49ccbba24537a206cf64f350795ef8f8870/analysis/2_DifferentialExpression.Rmd" target="_blank">1858f49</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
<td>
Started presenting actual DE genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/1858f49ccbba24537a206cf64f350795ef8f8870/docs/2_DifferentialExpression.html" target="_blank">1858f49</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
<td>
Started presenting actual DE genes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/10a285b55e3d5e8c883f78710a0a9f9dd3a6e75a/analysis/2_DifferentialExpression.Rmd" target="_blank">10a285b</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
<td>
Expanded description in index and tried reusing an image
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/10a285b55e3d5e8c883f78710a0a9f9dd3a6e75a/docs/2_DifferentialExpression.html" target="_blank">10a285b</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
<td>
Expanded description in index and tried reusing an image
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/aabb5708d27867bd0815852e026fa2849d9336d2/docs/2_DifferentialExpression.html" target="_blank">aabb570</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
<td>
Rebuilt to get rid of warnings
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/71b8832323504c21b9da26d8bc819b8cc116ea97/analysis/2_DifferentialExpression.Rmd" target="_blank">71b8832</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
<td>
Added DE QC for GC bias
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/71b8832323504c21b9da26d8bc819b8cc116ea97/docs/2_DifferentialExpression.html" target="_blank">71b8832</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
<td>
Added DE QC for GC bias
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/e8256379fb08fe9d10a9db0f184f160465b8a9bd/analysis/2_DifferentialExpression.Rmd" target="_blank">e825637</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
<td>
Minor updates to DE plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/01512dac59935dca1dd5e74c9c12478b0b0239eb/docs/2_DifferentialExpression.html" target="_blank">01512da</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
<td>
Added initial DE analysis to index
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/fbb62422891af2a7c033ef9780b9328b6c71a203/analysis/2_DifferentialExpression.Rmd" target="_blank">fbb6242</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
<td>
Paused DE analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/c5606377a61a29d83d3b20745b11c254265d3979/analysis/2_DifferentialExpression.Rmd" target="_blank">c560637</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-20
</td>
<td>
Started DE analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level1">
<h1>Setup</h1>
<pre class="r"><code>library(ngsReports)
library(tidyverse)
library(magrittr)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(scales)
library(pander)
library(cqn)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)</code></pre>
<pre class="r"><code>if (interactive()) setwd(here::here())
theme_set(theme_bw())
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;table.style&quot;, &quot;rmarkdown&quot;)</code></pre>
<div id="annotations" class="section level2">
<h2>Annotations</h2>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
    subset(species == &quot;Danio rerio&quot;) %&gt;%
    subset(rdataclass == &quot;EnsDb&quot;)
ensDb &lt;- ah[[&quot;AH74989&quot;]]</code></pre>
<pre class="r"><code>grTrans &lt;- transcripts(ensDb)
trLengths &lt;- exonsBy(ensDb, &quot;tx&quot;) %&gt;%
    width() %&gt;%
    vapply(sum, integer(1))
mcols(grTrans)$length &lt;- trLengths[names(grTrans)]</code></pre>
<pre class="r"><code>gcGene &lt;- grTrans %&gt;%
    mcols() %&gt;%
    as.data.frame() %&gt;%
    dplyr::select(gene_id, tx_id, gc_content, length) %&gt;%
    as_tibble() %&gt;%
    group_by(gene_id) %&gt;%
    summarise(
        gc_content = sum(gc_content*length) / sum(length),
        length = ceiling(median(length))
    )
grGenes &lt;- genes(ensDb)
mcols(grGenes) %&lt;&gt;%
    as.data.frame() %&gt;%
    left_join(gcGene) %&gt;%
    as.data.frame() %&gt;%
    DataFrame()</code></pre>
<p>Similarly to the Quality Assessment steps, <code>GRanges</code> objects were formed at the gene and transcript levels, to enable estimation of GC content and length for each transcript and gene. GC content and transcript length are available for each transcript, and for gene-level estimates, GC content was taken as the sum of all GC bases divided by the sum of all transcript lengths, effectively averaging across all transcripts. Gene length was defined as the median transcript length.</p>
<pre class="r"><code>samples &lt;- read_csv(&quot;data/samples.csv&quot;) %&gt;%
  distinct(sampleName, .keep_all = TRUE) %&gt;%
  dplyr::select(sample = sampleName, sampleID, genotype) %&gt;%
  mutate(
    genotype = factor(genotype, levels = c(&quot;WT&quot;, &quot;Het&quot;, &quot;Hom&quot;)),
    mutant = genotype %in% c(&quot;Het&quot;, &quot;Hom&quot;),
    homozygous = genotype == &quot;Hom&quot;
  )
genoCols &lt;- samples$genotype %&gt;%
  levels() %&gt;%
  length() %&gt;%
  brewer.pal(&quot;Set1&quot;) %&gt;%
  setNames(levels(samples$genotype))</code></pre>
<p>Sample metadata was also loaded, with only the sampleID and genotype being retained. All other fields were considered irrelevant.</p>
</div>
<div id="count-data" class="section level2">
<h2>Count Data</h2>
<pre class="r"><code>minCPM &lt;- 1.5
minSamples &lt;- 4
dgeList &lt;- file.path(&quot;data&quot;, &quot;2_alignedData&quot;, &quot;featureCounts&quot;, &quot;genes.out&quot;) %&gt;%
    read_delim(delim = &quot;\t&quot;) %&gt;%
    set_names(basename(names(.))) %&gt;%
    as.data.frame() %&gt;%
    column_to_rownames(&quot;Geneid&quot;) %&gt;%
    as.matrix() %&gt;% 
    set_colnames(str_remove(colnames(.), &quot;Aligned.sortedByCoord.out.bam&quot;)) %&gt;%
    .[rowSums(cpm(.) &gt;= minCPM) &gt;= minCPM,] %&gt;%
    DGEList(
        samples = tibble(sample = colnames(.)) %&gt;%
            left_join(samples),
        genes = grGenes[rownames(.)] %&gt;%
            as.data.frame() %&gt;%
            dplyr::select(
                chromosome = seqnames, start, end, 
                gene_id, gene_name, gene_biotype, description, 
                entrezid, gc_content, length
            )
    ) %&gt;%
    .[!grepl(&quot;rRNA&quot;, .$genes$gene_biotype),] %&gt;%
    calcNormFactors()</code></pre>
<p>Gene-level count data as output by <code>featureCounts</code>, was loaded and formed into a <code>DGEList</code> object. During this process, genes were removed if:</p>
<ul>
<li>They were not considered as detectable (CPM &lt; 1.5 in &gt; 8 samples). This translates to &gt; 18 reads assigned a gene in all samples from one or more of the genotype groups</li>
<li>The <code>gene_biotype</code> was any type of <code>rRNA</code>.</li>
</ul>
<p>These filtering steps returned gene-level counts for 16,640 genes, with total library sizes between <em>11,852,141</em> and <em>16,997,219</em> reads assigned to genes. It was noted that these library sizes were about 1.5-fold larger than the transcript-level counts used for the QA steps.</p>
<pre class="r"><code>cpm(dgeList, log = TRUE) %&gt;%
  as.data.frame() %&gt;%
  pivot_longer(
    cols = everything(),
    names_to = &quot;sample&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  split(f = .$sample) %&gt;%
  lapply(function(x){
    d &lt;- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %&gt;%
  bind_rows() %&gt;%
  left_join(samples) %&gt;%
  ggplot(aes(x, y, colour = genotype, group = sample)) +
  geom_line() +
  scale_colour_manual(
    values = genoCols
  ) +
  labs(
    x = &quot;logCPM&quot;,
    y = &quot;Density&quot;,
    colour = &quot;Genotype&quot;
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/plotDensities-1.png" alt="*Expression density plots for all samples after filtering, showing logCPM values.*" width="480" />
<p class="caption">
<em>Expression density plots for all samples after filtering, showing logCPM values.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotDensities-1">
Past versions of plotDensities-1.png
</button>
</p>
<div id="fig-plotDensities-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/plotDensities-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/1858f49ccbba24537a206cf64f350795ef8f8870/docs/figure/2_DifferentialExpression.Rmd/plotDensities-1.png" target="_blank">1858f49</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/01512dac59935dca1dd5e74c9c12478b0b0239eb/docs/figure/2_DifferentialExpression.Rmd/plotDensities-1.png" target="_blank">01512da</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="additional-functions" class="section level3">
<h3>Additional Functions</h3>
<pre class="r"><code>contLabeller &lt;- as_labeller(
    c(
        HetVsWT = &quot;S4Ter/+ Vs +/+&quot;,
        HomVsWT = &quot;S4Ter/S4Ter Vs +/+&quot;,
        HomVsHet = &quot;S4Ter/S4Ter Vs S4Ter/+&quot;,
        Hom = &quot;S4Ter/S4Ter&quot;,
        Het = &quot;S4Ter/+&quot;,
        WT = &quot;+/+&quot;,
        mutant = &quot;S4Ter Vs WT&quot;,
        homozygous = &quot;S4Ter/S4Ter Vs S4Ter/+&quot;
    )
)
geneLabeller &lt;- structure(grGenes$gene_name, names = grGenes$gene_id) %&gt;%
    as_labeller()</code></pre>
<p>Labeller functions for genotypes, contrasts and gene names were additionally defined for simpler plotting using <code>ggplot2</code>.</p>
</div>
</div>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<div id="pca" class="section level2">
<h2>PCA</h2>
<pre class="r"><code>pca &lt;- dgeList %&gt;%
    cpm(log = TRUE) %&gt;%
    t() %&gt;%
    prcomp() 
pcaVars &lt;- percent_format(0.1)(summary(pca)$importance[&quot;Proportion of Variance&quot;,])</code></pre>
<pre class="r"><code>pca$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;sample&quot;) %&gt;%
  left_join(samples) %&gt;%
  as_tibble() %&gt;%
  ggplot(aes(PC1, PC2, colour = genotype, fill = genotype)) +
  geom_point() +
  geom_text_repel(aes(label = sampleID), show.legend = FALSE) +
  stat_ellipse(geom = &quot;polygon&quot;, alpha = 0.05, show.legend = FALSE) +
  guides(fill = FALSE) +
  scale_colour_manual(
    values = genoCols
  ) +
  labs(
    x = paste0(&quot;PC1 (&quot;, pcaVars[[&quot;PC1&quot;]], &quot;)&quot;),
    y = paste0(&quot;PC2 (&quot;, pcaVars[[&quot;PC2&quot;]], &quot;)&quot;),
    colour = &quot;Genotype&quot;
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/plotPCA-1.png" alt="*PCA of gene-level counts.*" width="480" />
<p class="caption">
<em>PCA of gene-level counts.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotPCA-1">
Past versions of plotPCA-1.png
</button>
</p>
<div id="fig-plotPCA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/plotPCA-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/1858f49ccbba24537a206cf64f350795ef8f8870/docs/figure/2_DifferentialExpression.Rmd/plotPCA-1.png" target="_blank">1858f49</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/01512dac59935dca1dd5e74c9c12478b0b0239eb/docs/figure/2_DifferentialExpression.Rmd/plotPCA-1.png" target="_blank">01512da</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>A Principal Component Analysis (PCA) was also performed using logCPM values from each sample. Both mutant genotypes appear to cluster together, however it has previously been noted that GC content appears to track closely with PC1, as a result of variable rRNA removal.</p>
</div>
<div id="model-description" class="section level2">
<h2>Model Description</h2>
<p><img src="figure/index.Rmd/plotLayout-1.png" width="50%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotLayout-1">
Past versions of plotLayout-1.png
</button>
</p>
<div id="fig-plotLayout-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/10a285b55e3d5e8c883f78710a0a9f9dd3a6e75a/docs/figure/index.Rmd/plotLayout-1.png" target="_blank">10a285b</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The initial expectation was to perform three pairwise comparisons as described in the above figure. However, given there was a strong similarity between mutants, the model matrix was defined as containing an intercept (i.e. baseline expression in wild-type), with additional columns defining presence of any mutant alleles, and the final column capturing the difference between mutants. This avoids any issues with the use of hard cutoffs when combing lists, such as would occur when comparing separate results from both mutant genotypes against wild-type samples.</p>
<pre class="r"><code>d &lt;- model.matrix(~mutant + homozygous, data = dgeList$samples) %&gt;%
  set_colnames(str_remove(colnames(.), &quot;TRUE&quot;))</code></pre>
<pre class="r"><code>pheatmap(
  d, 
  cluster_cols = FALSE, 
  cluster_rows = FALSE, 
  color = c(&quot;white&quot;, &quot;grey50&quot;),
  annotation_row = dgeList$samples[&quot;genotype&quot;],
  annotation_colors = list(genotype = genoCols),
  legend = FALSE
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/mmVisualisation-1.png" alt="*Visualisation of the design matrix*" width="576" />
<p class="caption">
<em>Visualisation of the design matrix</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-mmVisualisation-1">
Past versions of mmVisualisation-1.png
</button>
</p>
<div id="fig-mmVisualisation-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/mmVisualisation-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="normalisation" class="section level3">
<h3>Normalisation</h3>
<p>As GC content and length was noted as being of concern for this dataset, <em>conditional-quantile normalisation</em> was performed using the <code>cqn</code> package. This adds a gene and sample-level offset for each count which takes into account any systemic bias, such as that identified previously as an artefact of variable rRNA removal. The resultant <code>glm.offset</code> values were added to the original <code>DGEList</code> object, and all dispersion estimates were calculated.</p>
<pre class="r"><code>gcCqn &lt;- cqn(
    counts = dgeList$counts,
    x = dgeList$genes$gc_content,
    lengths = dgeList$genes$length,
    sizeFactors = dgeList$samples$lib.size
)</code></pre>
<pre class="r"><code>par(mfrow = c(1, 2))
cqnplot(gcCqn, n = 1, xlab = &quot;GC Content&quot;, col = genoCols)
cqnplot(gcCqn, n = 2, xlab = &quot;Length&quot;, col = genoCols)
legend(&quot;bottomright&quot;, legend = levels(samples$genotype), col = genoCols, lty = 1)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/plotCQN-1.png" alt="*Model fits for GC content and gene length under the CQN model. Genotype-specific effects are clearly visible.*" width="672" />
<p class="caption">
<em>Model fits for GC content and gene length under the CQN model. Genotype-specific effects are clearly visible.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotCQN-1">
Past versions of plotCQN-1.png
</button>
</p>
<div id="fig-plotCQN-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/plotCQN-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/01512dac59935dca1dd5e74c9c12478b0b0239eb/docs/figure/2_DifferentialExpression.Rmd/plotCQN-1.png" target="_blank">01512da</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>par(mfrow = c(1, 1))</code></pre>
<pre class="r"><code>dgeList$offset &lt;- gcCqn$glm.offset 
dgeList %&lt;&gt;% estimateDisp(design = d)</code></pre>
</div>
</div>
<div id="model-fitting" class="section level2">
<h2>Model Fitting</h2>
<pre class="r"><code>minLfc &lt;- log2(2)
alpha &lt;- 0.01
fit &lt;- glmFit(dgeList)
topTables &lt;- colnames(d)[2:3] %&gt;%
  sapply(function(x){
    glmLRT(fit, coef = x) %&gt;%
      topTags(n = Inf) %&gt;%
      .[[&quot;table&quot;]] %&gt;%
      as_tibble() %&gt;%
      arrange(PValue) %&gt;%
      dplyr::select(
        gene_id, gene_name, logFC, logCPM, PValue, FDR, everything()  
      ) %&gt;%
      mutate(
        coef = x,
        bonfP = p.adjust(PValue, &quot;bonf&quot;),
        DE = case_when(
          bonfP &lt; alpha ~ TRUE,
          FDR &lt; alpha &amp; abs(logFC) &gt; minLfc ~ TRUE
        ),
        DE = ifelse(is.na(DE), FALSE, DE)
      )
  }, simplify = FALSE)</code></pre>
<p>Models were fit using the negative-binomial approaches of <code>glmFit()</code>. Top Tables were then obtained using likelihood-ratio tests in <code>glmLRT()</code>. These test the standard <span class="math inline">\(H_0\)</span> that the true value of the estimated model coefficient is zero. These model coefficients effectively estimate:</p>
<ol style="list-style-type: lower-alpha">
<li>the effect of the presence of a mutation, and</li>
<li>the difference in heterozygous and heterozygous mutants</li>
</ol>
<p>For enrichment testing, genes were initially considered to be DE using:</p>
<ol style="list-style-type: lower-alpha">
<li>An Bonferroni-adjusted p-value &lt; 0.01</li>
<li>An FDR-adjusted p-value &lt; 0.01 along with an estimated logFC outside of the range <span class="math inline">\(\pm \log_2(2)\)</span>.</li>
</ol>
<p>As fewer genes were detected in the comparisons between homozygous and heterozygous mutants, a simple FDR of 0.05 was subsequently chosen for this comparison.</p>
<pre class="r"><code>topTables$homozygous %&lt;&gt;%
  mutate(DE = FDR &lt; 0.05)</code></pre>
<p>Using these criteria, the following initial DE gene-sets were defined:</p>
<pre class="r"><code>topTables %&gt;%
  lapply(dplyr::filter, DE) %&gt;% 
  vapply(nrow, integer(1)) %&gt;%
  set_names(
    case_when(
      names(.) == &quot;mutant&quot; ~ &quot;psen2 mutant&quot;,
      names(.) == &quot;homozygous&quot; ~ &quot;HomVsHet&quot;
    )
  ) %&gt;%
  pander()</code></pre>
<table>
<thead>
<tr class="header">
<th align="center">psen2 mutant</th>
<th align="center">HomVsHet</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">615</td>
<td align="center">7</td>
</tr>
</tbody>
</table>
<pre class="r"><code>deCols &lt;- c(
  `FALSE` = rgb(0.5, 0.5, 0.5, 0.4), 
  `TRUE` = rgb(1, 0, 0, 0.7)
)</code></pre>
</div>
<div id="model-checking" class="section level2">
<h2>Model Checking</h2>
<div id="gc-and-length-bias" class="section level3">
<h3>GC and Length Bias</h3>
<pre class="r"><code>topTables %&gt;%
    bind_rows() %&gt;%
    mutate(stat = -sign(logFC)*log10(PValue)) %&gt;%
    ggplot(aes(gc_content, stat)) +
    geom_point(aes(colour = DE), alpha = 0.4) +
    geom_smooth(se = FALSE) +
    facet_wrap(~coef, labeller = contLabeller)  +
    labs(
        x = &quot;GC content (%)&quot;,
        y = &quot;Ranking Statistic&quot;
    ) +
    coord_cartesian(ylim = c(-10, 10)) +
    scale_colour_manual(values = deCols) +
    theme(legend.position = &quot;none&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/checkGCBias-1.png" alt="*Checks for GC bias in differential expression. GC content is shown against the ranking statistic, using -log10(p) multiplied by the sign of log fold-change. A small amount of bias was noted particularly in the comparison between mutants and wild-type.*" width="768" />
<p class="caption">
<em>Checks for GC bias in differential expression. GC content is shown against the ranking statistic, using -log10(p) multiplied by the sign of log fold-change. A small amount of bias was noted particularly in the comparison between mutants and wild-type.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-checkGCBias-1">
Past versions of checkGCBias-1.png
</button>
</p>
<div id="fig-checkGCBias-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/be95a600302098eba9391210983d011dcff1ee3a/docs/figure/2_DifferentialExpression.Rmd/checkGCBias-1.png" target="_blank">be95a60</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/checkGCBias-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/1858f49ccbba24537a206cf64f350795ef8f8870/docs/figure/2_DifferentialExpression.Rmd/checkGCBias-1.png" target="_blank">1858f49</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/71b8832323504c21b9da26d8bc819b8cc116ea97/docs/figure/2_DifferentialExpression.Rmd/checkGCBias-1.png" target="_blank">71b8832</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>topTables %&gt;%
    bind_rows() %&gt;%
    mutate(stat = -sign(logFC)*log10(PValue)) %&gt;%
    ggplot(aes(length, stat)) +
    geom_point(aes(colour = DE), alpha = 0.4) +
    geom_smooth(se = FALSE) +
    facet_wrap(~coef, labeller = contLabeller)  +
    labs(
        x = &quot;Gene Length (bp)&quot;,
        y = &quot;Ranking Statistic&quot;
    ) +
    coord_cartesian(ylim = c(-10, 10)) +
    scale_x_log10(labels = comma) +
    scale_colour_manual(values = deCols) +
    theme(legend.position = &quot;none&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/checkLengthBias-1.png" alt="*Checks for length bias in differential expression. Gene length is shown against the ranking statistic, using -log10(p) multiplied by the sign of log fold-change. Again, a small amount of bias was noted particularly in the comparison between mutants and wild-type.*" width="768" />
<p class="caption">
<em>Checks for length bias in differential expression. Gene length is shown against the ranking statistic, using -log10(p) multiplied by the sign of log fold-change. Again, a small amount of bias was noted particularly in the comparison between mutants and wild-type.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-checkLengthBias-1">
Past versions of checkLengthBias-1.png
</button>
</p>
<div id="fig-checkLengthBias-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/be95a600302098eba9391210983d011dcff1ee3a/docs/figure/2_DifferentialExpression.Rmd/checkLengthBias-1.png" target="_blank">be95a60</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/checkLengthBias-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/1858f49ccbba24537a206cf64f350795ef8f8870/docs/figure/2_DifferentialExpression.Rmd/checkLengthBias-1.png" target="_blank">1858f49</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/71b8832323504c21b9da26d8bc819b8cc116ea97/docs/figure/2_DifferentialExpression.Rmd/checkLengthBias-1.png" target="_blank">71b8832</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Checks for both GC and length bias on differential expression showed that a small bias remained evident, despite using conditional-quantile normalisation. However, <a href="2a_NoCQN.html">an alternative analysis on the same dataset <strong>excluding the CQN steps</strong></a> revealed vastly different and exaggerated bias. As such, the impact of CQN normalisation was considered to be appropriate.</p>
</div>
</div>
<div id="checks-for-rrna-impacts" class="section level2">
<h2>Checks for rRNA impacts</h2>
<pre class="r"><code>rawFqc &lt;- list.files(
  path = &quot;data/0_rawData/FastQC/&quot;,
  pattern = &quot;zip&quot;,
  full.names = TRUE
) %&gt;%
  FastqcDataList()
rawGC &lt;- getModule(rawFqc, &quot;Per_sequence_GC&quot;) %&gt;%
  group_by(Filename) %&gt;% 
  mutate(Freq = Count / sum(Count)) %&gt;%
  dplyr::filter(GC_Content &gt; 70) %&gt;% 
  summarise(Freq = sum(Freq)) %&gt;% 
  arrange(desc(Freq)) %&gt;%
  mutate(sample = str_remove(Filename, &quot;_R[12].fastq.gz&quot;)) %&gt;%
  group_by(sample) %&gt;%
  summarise(Freq = mean(Freq)) %&gt;%
  left_join(samples) 
riboVec &lt;- structure(rawGC$Freq, names = rawGC$sample)</code></pre>
<pre class="r"><code>riboCors &lt;- cpm(dgeList, log = TRUE)%&gt;%
  apply(1, function(x){
    cor(x, riboVec[names(x)])
  })</code></pre>
<p>Given the previously identified concerns about variable rRNA removal, correlations were calculated between each gene’s expression values and the proportion of the raw libraries with &gt; 70% GC. Those with the strongest correlation, and for which the FDR is &lt; 0.01 are shown below.</p>
<p>Most notably, the most correlated RNA with these rRNA proportions was <code>eif1b</code> which is the initiation factor for translation. As this is a protein coding gene with the <em>protein product</em> interacting with the ribosome, the reason for this correlation <em>at the RNA level</em> raises difficult questions which cannot be answered here.</p>
<pre class="r"><code>topTables$mutant %&gt;%
  mutate(riboCors = riboCors[gene_id]) %&gt;%
  dplyr::filter(
    FDR &lt; alpha
  ) %&gt;%
  dplyr::select(gene_id, gene_name, logFC, logCPM, FDR, riboCors, DE) %&gt;%
  arrange(desc(riboCors)) %&gt;%
  mutate(FDR = case_when(
    FDR &gt;= 0.0001 ~ sprintf(&quot;%.4f&quot;, FDR),
    FDR &lt; 0.0001 ~ sprintf(&quot;%.2e&quot;, FDR)
    )
  ) %&gt;%
  dplyr::slice(1:40) %&gt;%
  pander(
    justify = &quot;llrrrrl&quot;,
    style = &quot;rmarkdown&quot;,
    caption = paste(
      &quot;The&quot;, nrow(.), &quot;genes most correlated with the original high GC content.&quot;,
      &quot;Many failed the selection criteria for differential expression,&quot;,
      &quot;primarily due to the stringent logFC filter.&quot;,
      &quot;However, the unusually high number of ribosomal protein coding genes&quot;,
      &quot;is currently difficult to explain, but notable.&quot;,
      &quot;Two possibilities are these RNAs are somehow translated with physical&quot;,
      &quot;linkage to the ribosomal complex, or that there is genuinely a global&quot;,
      &quot;shift in the level of ribosomal activity.&quot;,
      &quot;Neither explanation is initially satisfying.&quot;
    )
  )</code></pre>
<table>
<caption>The 40 genes most correlated with the original high GC content. Many failed the selection criteria for differential expression, primarily due to the stringent logFC filter. However, the unusually high number of ribosomal protein coding genes is currently difficult to explain, but notable. Two possibilities are these RNAs are somehow translated with physical linkage to the ribosomal complex, or that there is genuinely a global shift in the level of ribosomal activity. Neither explanation is initially satisfying.</caption>
<thead>
<tr class="header">
<th align="left">gene_id</th>
<th align="left">gene_name</th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">FDR</th>
<th align="right">riboCors</th>
<th align="left">DE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSDARG00000012688</td>
<td align="left">eif1b</td>
<td align="right">0.5229</td>
<td align="right">7.482</td>
<td align="right">0.0024</td>
<td align="right">0.9906</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000012871</td>
<td align="left">npepl1</td>
<td align="right">0.6049</td>
<td align="right">4.485</td>
<td align="right">0.0028</td>
<td align="right">0.9823</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000103994</td>
<td align="left">ppiab</td>
<td align="right">0.8891</td>
<td align="right">7.372</td>
<td align="right">6.18e-05</td>
<td align="right">0.9814</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000034897</td>
<td align="left">rps10</td>
<td align="right">0.8809</td>
<td align="right">5.898</td>
<td align="right">0.0001</td>
<td align="right">0.9795</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000068995</td>
<td align="left">h2afx1</td>
<td align="right">0.644</td>
<td align="right">6.605</td>
<td align="right">0.0025</td>
<td align="right">0.9734</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000077291</td>
<td align="left">rps2</td>
<td align="right">1.305</td>
<td align="right">7.52</td>
<td align="right">1.58e-07</td>
<td align="right">0.9733</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000002240</td>
<td align="left">psmb6</td>
<td align="right">0.6407</td>
<td align="right">4.702</td>
<td align="right">0.0056</td>
<td align="right">0.9733</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000036298</td>
<td align="left">rps13</td>
<td align="right">0.6762</td>
<td align="right">6.076</td>
<td align="right">8.26e-05</td>
<td align="right">0.9707</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000037071</td>
<td align="left">rps26</td>
<td align="right">0.651</td>
<td align="right">5.323</td>
<td align="right">0.0045</td>
<td align="right">0.97</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000057026</td>
<td align="left">ran</td>
<td align="right">0.6265</td>
<td align="right">7.071</td>
<td align="right">0.0002</td>
<td align="right">0.9689</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000111753</td>
<td align="left">hist1h4l</td>
<td align="right">0.9775</td>
<td align="right">2.015</td>
<td align="right">0.0061</td>
<td align="right">0.9685</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000038028</td>
<td align="left">ndufa6</td>
<td align="right">1.051</td>
<td align="right">4.693</td>
<td align="right">2.57e-05</td>
<td align="right">0.9667</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000046157</td>
<td align="left">RPS17</td>
<td align="right">0.7725</td>
<td align="right">6.74</td>
<td align="right">0.0002</td>
<td align="right">0.9667</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000078929</td>
<td align="left">abhd16a</td>
<td align="right">0.5358</td>
<td align="right">4.906</td>
<td align="right">0.0013</td>
<td align="right">0.9664</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000099226</td>
<td align="left">CABZ01076667.1</td>
<td align="right">0.5407</td>
<td align="right">3.973</td>
<td align="right">0.0008</td>
<td align="right">0.9652</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000014690</td>
<td align="left">rps4x</td>
<td align="right">0.6511</td>
<td align="right">7.368</td>
<td align="right">0.0003</td>
<td align="right">0.9649</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000092553</td>
<td align="left">slc25a5</td>
<td align="right">0.9012</td>
<td align="right">8.952</td>
<td align="right">2.64e-05</td>
<td align="right">0.9648</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000011665</td>
<td align="left">aldoaa</td>
<td align="right">0.6082</td>
<td align="right">7.763</td>
<td align="right">0.0029</td>
<td align="right">0.9639</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000045447</td>
<td align="left">slc35g2b</td>
<td align="right">0.766</td>
<td align="right">4.873</td>
<td align="right">0.0002</td>
<td align="right">0.9627</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000034534</td>
<td align="left">atp6v1aa</td>
<td align="right">0.6212</td>
<td align="right">7.049</td>
<td align="right">0.0003</td>
<td align="right">0.9618</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000099766</td>
<td align="left">myl12.1</td>
<td align="right">0.3841</td>
<td align="right">6.468</td>
<td align="right">0.0058</td>
<td align="right">0.9595</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000037962</td>
<td align="left">psmb7</td>
<td align="right">0.7979</td>
<td align="right">4.104</td>
<td align="right">3.30e-05</td>
<td align="right">0.9592</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000019230</td>
<td align="left">rpl7a</td>
<td align="right">0.9036</td>
<td align="right">7.71</td>
<td align="right">5.34e-05</td>
<td align="right">0.9591</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000099104</td>
<td align="left">rpl24</td>
<td align="right">0.662</td>
<td align="right">6.729</td>
<td align="right">0.0004</td>
<td align="right">0.9571</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000055475</td>
<td align="left">rps27.2</td>
<td align="right">0.6175</td>
<td align="right">6.719</td>
<td align="right">0.0019</td>
<td align="right">0.9566</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000026322</td>
<td align="left">dhrs13a.1</td>
<td align="right">0.7388</td>
<td align="right">2.804</td>
<td align="right">0.0032</td>
<td align="right">0.9563</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000025581</td>
<td align="left">rpl10</td>
<td align="right">1.096</td>
<td align="right">6.381</td>
<td align="right">0.0003</td>
<td align="right">0.9562</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000092807</td>
<td align="left">si:dkey-151g10.6</td>
<td align="right">0.8303</td>
<td align="right">6.491</td>
<td align="right">9.37e-07</td>
<td align="right">0.9551</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000053365</td>
<td align="left">rpl31</td>
<td align="right">1.215</td>
<td align="right">6.638</td>
<td align="right">8.99e-06</td>
<td align="right">0.9549</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000075445</td>
<td align="left">psmb5</td>
<td align="right">0.8405</td>
<td align="right">5.59</td>
<td align="right">5.33e-05</td>
<td align="right">0.9544</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000030237</td>
<td align="left">pgrmc2</td>
<td align="right">0.3734</td>
<td align="right">6.21</td>
<td align="right">0.0046</td>
<td align="right">0.9538</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000011405</td>
<td align="left">rps9</td>
<td align="right">0.823</td>
<td align="right">6.209</td>
<td align="right">0.0006</td>
<td align="right">0.9537</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000035808</td>
<td align="left">clcn4</td>
<td align="right">0.6887</td>
<td align="right">5.678</td>
<td align="right">0.0014</td>
<td align="right">0.9528</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000043561</td>
<td align="left">psmc1b</td>
<td align="right">0.6844</td>
<td align="right">4.673</td>
<td align="right">0.0004</td>
<td align="right">0.9517</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000017235</td>
<td align="left">eif5a</td>
<td align="right">0.7003</td>
<td align="right">7.388</td>
<td align="right">5.54e-06</td>
<td align="right">0.9514</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000034291</td>
<td align="left">rpl37</td>
<td align="right">1.085</td>
<td align="right">5.662</td>
<td align="right">5.03e-06</td>
<td align="right">0.9513</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000104173</td>
<td align="left">tufm</td>
<td align="right">0.7123</td>
<td align="right">5.051</td>
<td align="right">0.0013</td>
<td align="right">0.9511</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000058451</td>
<td align="left">rpl6</td>
<td align="right">0.7848</td>
<td align="right">7.877</td>
<td align="right">0.0004</td>
<td align="right">0.9497</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000014867</td>
<td align="left">rpl8</td>
<td align="right">0.9139</td>
<td align="right">6.716</td>
<td align="right">1.50e-06</td>
<td align="right">0.9493</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000089976</td>
<td align="left">spcs3</td>
<td align="right">0.7379</td>
<td align="right">3.54</td>
<td align="right">0.0034</td>
<td align="right">0.949</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
<div id="other-artefacts" class="section level3">
<h3>Other Artefacts</h3>
<pre class="r"><code>topTables %&gt;%
  bind_rows() %&gt;%
  arrange(DE) %&gt;%
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = DE)) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %&gt;% dplyr::filter(DE &amp; abs(logFC) &gt; 2.9)
  ) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %&gt;% dplyr::filter(FDR &lt; 0.05 &amp; coef == &quot;homozygous&quot;)
  ) +
  geom_smooth(se = FALSE) +
  geom_hline(
    yintercept = c(-1, 1)*minLfc,
    linetype = 2,
    colour = &quot;red&quot;
  ) +
  facet_wrap(~coef, nrow = 2, labeller = contLabeller) +
  scale_y_continuous(breaks = seq(-8, 8, by = 2)) +
  scale_colour_manual(values = deCols) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/maPlots-1.png" alt="*MA plots checking for any logFC bias across the range of expression values. The small curve in the average at the low end of expression values was considered to be an artefact of the sparse points at this end. Initial DE genes are shown in red, with select points labelled.*" width="768" />
<p class="caption">
<em>MA plots checking for any logFC bias across the range of expression values. The small curve in the average at the low end of expression values was considered to be an artefact of the sparse points at this end. Initial DE genes are shown in red, with select points labelled.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-maPlots-1">
Past versions of maPlots-1.png
</button>
</p>
<div id="fig-maPlots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/be95a600302098eba9391210983d011dcff1ee3a/docs/figure/2_DifferentialExpression.Rmd/maPlots-1.png" target="_blank">be95a60</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/maPlots-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/1858f49ccbba24537a206cf64f350795ef8f8870/docs/figure/2_DifferentialExpression.Rmd/maPlots-1.png" target="_blank">1858f49</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/71b8832323504c21b9da26d8bc819b8cc116ea97/docs/figure/2_DifferentialExpression.Rmd/maPlots-1.png" target="_blank">71b8832</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/01512dac59935dca1dd5e74c9c12478b0b0239eb/docs/figure/2_DifferentialExpression.Rmd/maPlots-1.png" target="_blank">01512da</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>topTables %&gt;%
  bind_rows() %&gt;%
  ggplot(aes(PValue)) +
  geom_histogram(
    binwidth = 0.02,
    colour = &quot;black&quot;, fill = &quot;grey90&quot;
    ) +
  facet_wrap(~coef, labeller = contLabeller)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/pHist-1.png" alt="*Histograms of p-values for both sets of coefficients. Values for the difference between mutants follow the expected distribution for when there are very few differences, whilst values for the presence of a mutant allele show the expected distribution for when there are many differences. In particular the spike near zero indicates many genes which are differentially expressed.*" width="768" />
<p class="caption">
<em>Histograms of p-values for both sets of coefficients. Values for the difference between mutants follow the expected distribution for when there are very few differences, whilst values for the presence of a mutant allele show the expected distribution for when there are many differences. In particular the spike near zero indicates many genes which are differentially expressed.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pHist-1">
Past versions of pHist-1.png
</button>
</p>
<div id="fig-pHist-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/be95a600302098eba9391210983d011dcff1ee3a/docs/figure/2_DifferentialExpression.Rmd/pHist-1.png" target="_blank">be95a60</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/pHist-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<pre class="r"><code>topTables %&gt;%
    bind_rows() %&gt;%
    ggplot(aes(logFC, -log10(PValue), colour = DE)) +
    geom_point(alpha = 0.4) +
    geom_text_repel(
        aes(label = gene_name),
        data = . %&gt;% dplyr::filter(PValue &lt; 1e-12 | abs(logFC) &gt; 4)
    ) +
    geom_text_repel(
        aes(label = gene_name),
        data = . %&gt;% dplyr::filter(FDR &lt; 0.05 &amp; coef == &quot;homozygous&quot;)
    ) +
  geom_vline(
    xintercept = c(-1, 1)*minLfc,
    linetype = 2,
    colour = &quot;red&quot;
  ) +
  facet_wrap(~coef, nrow = 1, labeller = contLabeller) +
  scale_colour_manual(values = deCols) +
  scale_x_continuous(breaks = seq(-8, 8, by = 2)) +
  theme(legend.position = &quot;none&quot;) </code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/volcanoPlots-1.png" alt="*Volcano Plots showing DE genes against logFC.*" width="768" />
<p class="caption">
<em>Volcano Plots showing DE genes against logFC.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcanoPlots-1">
Past versions of volcanoPlots-1.png
</button>
</p>
<div id="fig-volcanoPlots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/be95a600302098eba9391210983d011dcff1ee3a/docs/figure/2_DifferentialExpression.Rmd/volcanoPlots-1.png" target="_blank">be95a60</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/volcanoPlots-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/1858f49ccbba24537a206cf64f350795ef8f8870/docs/figure/2_DifferentialExpression.Rmd/volcanoPlots-1.png" target="_blank">1858f49</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/71b8832323504c21b9da26d8bc819b8cc116ea97/docs/figure/2_DifferentialExpression.Rmd/volcanoPlots-1.png" target="_blank">71b8832</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/01512dac59935dca1dd5e74c9c12478b0b0239eb/docs/figure/2_DifferentialExpression.Rmd/volcanoPlots-1.png" target="_blank">01512da</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>deGenes &lt;- topTables %&gt;%
  lapply(dplyr::filter, DE) %&gt;%
  lapply(magrittr::extract2, &quot;gene_id&quot;) </code></pre>
<div id="genes-tracking-with-psen2s4ter" class="section level2">
<h2>Genes Tracking with <em>psen2<sup>S4Ter</sup></em></h2>
<pre class="r"><code>n &lt;- 40</code></pre>
<p>A set of 615 genes was identified as DE in the presence of the mutant <em>psen2<sup>S4Ter</sup></em> transcript. The most highly ranked 40 genes are shown in the following heatmap. The presence of <code>SNORNA70</code> in this list is of some concern as this is an ncRNA which is known to interact with ribosomes. In particular the expression pattern follows very closely the patterns previosuly identified for rRNA depletion using the percentage of libraries with GC &gt; 70. For example, WT_86 was the sample with most complete rRNA depletion whilst, Het_84 showed the largest proportion of the library as being derived from rRNA. These represent the samples with lowest and highest expression of SNORNA70 respectively.</p>
<pre class="r"><code>genoCols &lt;- RColorBrewer::brewer.pal(3, &quot;Set1&quot;) %&gt;%
  setNames(levels(dgeList$samples$genotype))
dgeList %&gt;%
  cpm(log = TRUE) %&gt;% 
  extract(deGenes$mutant[seq_len(n)],) %&gt;%
  as.data.frame() %&gt;%
  set_rownames(unlist(geneLabeller(rownames(.)))) %&gt;%
  pheatmap::pheatmap(
    color = viridis_pal(option = &quot;magma&quot;)(100),
    labels_col = colnames(.) %&gt;%
      str_replace_all(&quot;.+(Het|Hom|WT).+F3(_[0-9]{2}).+&quot;, &quot;\\1\\2&quot;),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), &quot;logCPM\n&quot;),
    annotation_col = dgeList$samples %&gt;%
      dplyr::select(Genotype = genotype),
    annotation_names_col = FALSE,
    annotation_colors = list(Genotype = genoCols),
    cutree_cols = 2,
    cutree_rows = 1#6
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/commonSummaryHeatmap-1.png" alt="*The 40 most highly-ranked genes by FDR which are commonly considered DE between mutants and WT samples*" width="768" />
<p class="caption">
<em>The 40 most highly-ranked genes by FDR which are commonly considered DE between mutants and WT samples</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-commonSummaryHeatmap-1">
Past versions of commonSummaryHeatmap-1.png
</button>
</p>
<div id="fig-commonSummaryHeatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/commonSummaryHeatmap-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/1858f49ccbba24537a206cf64f350795ef8f8870/docs/figure/2_DifferentialExpression.Rmd/commonSummaryHeatmap-1.png" target="_blank">1858f49</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="genes-showing-differences-between-mutants" class="section level2">
<h2>Genes showing differences between mutants</h2>
<p>When inspecting the genes showing differences between mutants, it was noted that the WT and Homozygous mutant samples clustered together, implying that there was a unique effect on these genes which was specific to the heterozygous condition.</p>
<pre class="r"><code>dgeList %&gt;%
  cpm(log = TRUE) %&gt;% 
  extract(deGenes$homozygous,) %&gt;%
  as.data.frame() %&gt;%
  set_rownames(unlist(geneLabeller(rownames(.)))) %&gt;%
  pheatmap::pheatmap(
    color = viridis_pal(option = &quot;magma&quot;)(100),
    labels_col = colnames(.) %&gt;%
      str_replace_all(&quot;.+(Het|Hom|WT).+F3(_[0-9]{2}).+&quot;, &quot;\\1\\2&quot;),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), &quot;logCPM\n&quot;),
    annotation_col = dgeList$samples %&gt;%
      dplyr::select(Genotype = genotype),
    annotation_names_col = FALSE,
    annotation_colors = list(Genotype = genoCols),
    cutree_cols = 2,
    cutree_rows = 3
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/2_DifferentialExpression.Rmd/homozygousHeatmap-1.png" alt="*The 7 most highly-ranked genes by FDR which are DE between between mutants*" width="672" />
<p class="caption">
<em>The 7 most highly-ranked genes by FDR which are DE between between mutants</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-homozygousHeatmap-1">
Past versions of homozygousHeatmap-1.png
</button>
</p>
<div id="fig-homozygousHeatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7251bd286b8818222ba7aa1dcfaf73e53c84eab7/docs/figure/2_DifferentialExpression.Rmd/homozygousHeatmap-1.png" target="_blank">7251bd2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="data-export" class="section level1">
<h1>Data Export</h1>
<p>Final gene lists were exported as separate csv files, along with fit and dgeList objects.</p>
<pre class="r"><code>write_csv(topTables$mutant, &quot;output/psen2VsWT.csv&quot;)
write_csv(topTables$homozygous, &quot;output/psen2HomVsHet.csv&quot;)</code></pre>
<pre class="r"><code>write_rds(fit, &quot;data/fit.rds&quot;)
write_rds(dgeList, &quot;data/dgeList.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>devtools::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.2 (2019-12-12)
 os       Ubuntu 18.04.3 LTS          
 system   x86_64, linux-gnu           
 ui       X11                         
 language en_AU:en                    
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Adelaide          
 date     2020-01-25                  

─ Packages ───────────────────────────────────────────────────────────────────
 package                * version  date       lib source        
 AnnotationDbi          * 1.48.0   2019-10-29 [2] Bioconductor  
 AnnotationFilter       * 1.10.0   2019-10-29 [2] Bioconductor  
 AnnotationHub          * 2.18.0   2019-10-29 [2] Bioconductor  
 askpass                  1.1      2019-01-13 [2] CRAN (R 3.6.0)
 assertthat               0.2.1    2019-03-21 [2] CRAN (R 3.6.0)
 backports                1.1.5    2019-10-02 [2] CRAN (R 3.6.1)
 Biobase                * 2.46.0   2019-10-29 [2] Bioconductor  
 BiocFileCache          * 1.10.2   2019-11-08 [2] Bioconductor  
 BiocGenerics           * 0.32.0   2019-10-29 [2] Bioconductor  
 BiocManager              1.30.10  2019-11-16 [2] CRAN (R 3.6.1)
 BiocParallel             1.20.1   2019-12-21 [2] Bioconductor  
 BiocVersion              3.10.1   2019-06-06 [2] Bioconductor  
 biomaRt                  2.42.0   2019-10-29 [2] Bioconductor  
 Biostrings               2.54.0   2019-10-29 [2] Bioconductor  
 bit                      1.1-15.1 2020-01-14 [2] CRAN (R 3.6.2)
 bit64                    0.9-7    2017-05-08 [2] CRAN (R 3.6.0)
 bitops                   1.0-6    2013-08-17 [2] CRAN (R 3.6.0)
 blob                     1.2.1    2020-01-20 [2] CRAN (R 3.6.2)
 broom                    0.5.3    2019-12-14 [2] CRAN (R 3.6.2)
 callr                    3.4.0    2019-12-09 [2] CRAN (R 3.6.2)
 cellranger               1.1.0    2016-07-27 [2] CRAN (R 3.6.0)
 cli                      2.0.1    2020-01-08 [2] CRAN (R 3.6.2)
 cluster                  2.1.0    2019-06-19 [2] CRAN (R 3.6.1)
 colorspace               1.4-1    2019-03-18 [2] CRAN (R 3.6.0)
 cqn                    * 1.32.0   2019-10-29 [2] Bioconductor  
 crayon                   1.3.4    2017-09-16 [2] CRAN (R 3.6.0)
 curl                     4.3      2019-12-02 [2] CRAN (R 3.6.2)
 data.table               1.12.8   2019-12-09 [2] CRAN (R 3.6.2)
 DBI                      1.1.0    2019-12-15 [2] CRAN (R 3.6.2)
 dbplyr                 * 1.4.2    2019-06-17 [2] CRAN (R 3.6.0)
 DelayedArray             0.12.2   2020-01-06 [2] Bioconductor  
 desc                     1.2.0    2018-05-01 [2] CRAN (R 3.6.0)
 devtools                 2.2.1    2019-09-24 [2] CRAN (R 3.6.1)
 digest                   0.6.23   2019-11-23 [2] CRAN (R 3.6.1)
 dplyr                  * 0.8.3    2019-07-04 [2] CRAN (R 3.6.1)
 edgeR                  * 3.28.0   2019-10-29 [2] Bioconductor  
 ellipsis                 0.3.0    2019-09-20 [2] CRAN (R 3.6.1)
 ensembldb              * 2.10.2   2019-11-20 [2] Bioconductor  
 evaluate                 0.14     2019-05-28 [2] CRAN (R 3.6.0)
 FactoMineR               2.1      2020-01-17 [2] CRAN (R 3.6.2)
 fansi                    0.4.1    2020-01-08 [2] CRAN (R 3.6.2)
 farver                   2.0.3    2020-01-16 [2] CRAN (R 3.6.2)
 fastmap                  1.0.1    2019-10-08 [2] CRAN (R 3.6.1)
 flashClust               1.01-2   2012-08-21 [2] CRAN (R 3.6.1)
 forcats                * 0.4.0    2019-02-17 [2] CRAN (R 3.6.0)
 fs                       1.3.1    2019-05-06 [2] CRAN (R 3.6.0)
 generics                 0.0.2    2018-11-29 [2] CRAN (R 3.6.0)
 GenomeInfoDb           * 1.22.0   2019-10-29 [2] Bioconductor  
 GenomeInfoDbData         1.2.2    2019-11-21 [2] Bioconductor  
 GenomicAlignments        1.22.1   2019-11-12 [2] Bioconductor  
 GenomicFeatures        * 1.38.1   2020-01-22 [2] Bioconductor  
 GenomicRanges          * 1.38.0   2019-10-29 [2] Bioconductor  
 ggdendro                 0.1-20   2016-04-27 [2] CRAN (R 3.6.0)
 ggplot2                * 3.2.1    2019-08-10 [2] CRAN (R 3.6.1)
 ggrepel                * 0.8.1    2019-05-07 [2] CRAN (R 3.6.0)
 git2r                    0.26.1   2019-06-29 [2] CRAN (R 3.6.1)
 glue                     1.3.1    2019-03-12 [2] CRAN (R 3.6.0)
 gtable                   0.3.0    2019-03-25 [2] CRAN (R 3.6.0)
 haven                    2.2.0    2019-11-08 [2] CRAN (R 3.6.1)
 highr                    0.8      2019-03-20 [2] CRAN (R 3.6.0)
 hms                      0.5.3    2020-01-08 [2] CRAN (R 3.6.2)
 htmltools                0.4.0    2019-10-04 [2] CRAN (R 3.6.1)
 htmlwidgets              1.5.1    2019-10-08 [2] CRAN (R 3.6.1)
 httpuv                   1.5.2    2019-09-11 [2] CRAN (R 3.6.1)
 httr                     1.4.1    2019-08-05 [2] CRAN (R 3.6.1)
 hwriter                  1.3.2    2014-09-10 [2] CRAN (R 3.6.0)
 interactiveDisplayBase   1.24.0   2019-10-29 [2] Bioconductor  
 IRanges                * 2.20.2   2020-01-13 [2] Bioconductor  
 jpeg                     0.1-8.1  2019-10-24 [2] CRAN (R 3.6.1)
 jsonlite                 1.6      2018-12-07 [2] CRAN (R 3.6.0)
 kableExtra               1.1.0    2019-03-16 [2] CRAN (R 3.6.1)
 knitr                    1.27     2020-01-16 [2] CRAN (R 3.6.2)
 labeling                 0.3      2014-08-23 [2] CRAN (R 3.6.0)
 later                    1.0.0    2019-10-04 [2] CRAN (R 3.6.1)
 lattice                  0.20-38  2018-11-04 [4] CRAN (R 3.6.0)
 latticeExtra             0.6-29   2019-12-19 [2] CRAN (R 3.6.2)
 lazyeval                 0.2.2    2019-03-15 [2] CRAN (R 3.6.0)
 leaps                    3.1      2020-01-16 [2] CRAN (R 3.6.2)
 lifecycle                0.1.0    2019-08-01 [2] CRAN (R 3.6.1)
 limma                  * 3.42.0   2019-10-29 [2] Bioconductor  
 locfit                   1.5-9.1  2013-04-20 [2] CRAN (R 3.6.0)
 lubridate                1.7.4    2018-04-11 [2] CRAN (R 3.6.0)
 magrittr               * 1.5      2014-11-22 [2] CRAN (R 3.6.0)
 MASS                     7.3-51.5 2019-12-20 [4] CRAN (R 3.6.2)
 Matrix                   1.2-18   2019-11-27 [2] CRAN (R 3.6.1)
 MatrixModels             0.4-1    2015-08-22 [2] CRAN (R 3.6.1)
 matrixStats              0.55.0   2019-09-07 [2] CRAN (R 3.6.1)
 mclust                 * 5.4.5    2019-07-08 [2] CRAN (R 3.6.1)
 memoise                  1.1.0    2017-04-21 [2] CRAN (R 3.6.0)
 mgcv                     1.8-31   2019-11-09 [4] CRAN (R 3.6.1)
 mime                     0.8      2019-12-19 [2] CRAN (R 3.6.2)
 modelr                   0.1.5    2019-08-08 [2] CRAN (R 3.6.1)
 munsell                  0.5.0    2018-06-12 [2] CRAN (R 3.6.0)
 ngsReports             * 1.1.2    2019-10-16 [1] Bioconductor  
 nlme                     3.1-143  2019-12-10 [4] CRAN (R 3.6.2)
 nor1mix                * 1.3-0    2019-06-13 [2] CRAN (R 3.6.1)
 openssl                  1.4.1    2019-07-18 [2] CRAN (R 3.6.1)
 pander                 * 0.6.3    2018-11-06 [2] CRAN (R 3.6.0)
 pheatmap               * 1.0.12   2019-01-04 [2] CRAN (R 3.6.0)
 pillar                   1.4.3    2019-12-20 [2] CRAN (R 3.6.2)
 pkgbuild                 1.0.6    2019-10-09 [2] CRAN (R 3.6.1)
 pkgconfig                2.0.3    2019-09-22 [2] CRAN (R 3.6.1)
 pkgload                  1.0.2    2018-10-29 [2] CRAN (R 3.6.0)
 plotly                   4.9.1    2019-11-07 [2] CRAN (R 3.6.1)
 plyr                     1.8.5    2019-12-10 [2] CRAN (R 3.6.2)
 png                      0.1-7    2013-12-03 [2] CRAN (R 3.6.0)
 preprocessCore         * 1.48.0   2019-10-29 [2] Bioconductor  
 prettyunits              1.1.1    2020-01-24 [2] CRAN (R 3.6.2)
 processx                 3.4.1    2019-07-18 [2] CRAN (R 3.6.1)
 progress                 1.2.2    2019-05-16 [2] CRAN (R 3.6.0)
 promises                 1.1.0    2019-10-04 [2] CRAN (R 3.6.1)
 ProtGenerics             1.18.0   2019-10-29 [2] Bioconductor  
 ps                       1.3.0    2018-12-21 [2] CRAN (R 3.6.0)
 purrr                  * 0.3.3    2019-10-18 [2] CRAN (R 3.6.1)
 quantreg               * 5.54     2019-12-13 [2] CRAN (R 3.6.2)
 R6                       2.4.1    2019-11-12 [2] CRAN (R 3.6.1)
 rappdirs                 0.3.1    2016-03-28 [2] CRAN (R 3.6.0)
 RColorBrewer           * 1.1-2    2014-12-07 [2] CRAN (R 3.6.0)
 Rcpp                     1.0.3    2019-11-08 [2] CRAN (R 3.6.1)
 RCurl                    1.98-1.1 2020-01-19 [2] CRAN (R 3.6.2)
 readr                  * 1.3.1    2018-12-21 [2] CRAN (R 3.6.0)
 readxl                   1.3.1    2019-03-13 [2] CRAN (R 3.6.0)
 remotes                  2.1.0    2019-06-24 [2] CRAN (R 3.6.0)
 reprex                   0.3.0    2019-05-16 [2] CRAN (R 3.6.0)
 reshape2                 1.4.3    2017-12-11 [2] CRAN (R 3.6.0)
 rlang                    0.4.2    2019-11-23 [2] CRAN (R 3.6.1)
 rmarkdown                2.1      2020-01-20 [2] CRAN (R 3.6.2)
 rprojroot                1.3-2    2018-01-03 [2] CRAN (R 3.6.0)
 Rsamtools                2.2.1    2019-11-06 [2] Bioconductor  
 RSQLite                  2.2.0    2020-01-07 [2] CRAN (R 3.6.2)
 rstudioapi               0.10     2019-03-19 [2] CRAN (R 3.6.0)
 rtracklayer              1.46.0   2019-10-29 [2] Bioconductor  
 rvest                    0.3.5    2019-11-08 [2] CRAN (R 3.6.1)
 S4Vectors              * 0.24.3   2020-01-18 [2] Bioconductor  
 scales                 * 1.1.0    2019-11-18 [2] CRAN (R 3.6.1)
 scatterplot3d            0.3-41   2018-03-14 [2] CRAN (R 3.6.1)
 sessioninfo              1.1.1    2018-11-05 [2] CRAN (R 3.6.0)
 shiny                    1.4.0    2019-10-10 [2] CRAN (R 3.6.1)
 ShortRead                1.44.1   2019-12-19 [2] Bioconductor  
 SparseM                * 1.78     2019-12-13 [2] CRAN (R 3.6.2)
 stringi                  1.4.5    2020-01-11 [2] CRAN (R 3.6.2)
 stringr                * 1.4.0    2019-02-10 [2] CRAN (R 3.6.0)
 SummarizedExperiment     1.16.1   2019-12-19 [2] Bioconductor  
 testthat                 2.3.1    2019-12-01 [2] CRAN (R 3.6.1)
 tibble                 * 2.1.3    2019-06-06 [2] CRAN (R 3.6.0)
 tidyr                  * 1.0.0    2019-09-11 [2] CRAN (R 3.6.1)
 tidyselect               0.2.5    2018-10-11 [2] CRAN (R 3.6.0)
 tidyverse              * 1.3.0    2019-11-21 [2] CRAN (R 3.6.1)
 truncnorm                1.0-8    2018-02-27 [2] CRAN (R 3.6.0)
 usethis                  1.5.1    2019-07-04 [2] CRAN (R 3.6.1)
 vctrs                    0.2.1    2019-12-17 [2] CRAN (R 3.6.2)
 viridisLite              0.3.0    2018-02-01 [2] CRAN (R 3.6.0)
 webshot                  0.5.2    2019-11-22 [2] CRAN (R 3.6.1)
 whisker                  0.4      2019-08-28 [2] CRAN (R 3.6.1)
 withr                    2.1.2    2018-03-15 [2] CRAN (R 3.6.0)
 workflowr              * 1.6.0    2019-12-19 [2] CRAN (R 3.6.2)
 xfun                     0.12     2020-01-13 [2] CRAN (R 3.6.2)
 XML                      3.99-0.3 2020-01-20 [2] CRAN (R 3.6.2)
 xml2                     1.2.2    2019-08-09 [2] CRAN (R 3.6.1)
 xtable                   1.8-4    2019-04-21 [2] CRAN (R 3.6.0)
 XVector                  0.26.0   2019-10-29 [2] Bioconductor  
 yaml                     2.2.0    2018-07-25 [2] CRAN (R 3.6.0)
 zeallot                  0.1.0    2018-01-28 [2] CRAN (R 3.6.0)
 zlibbioc                 1.32.0   2019-10-29 [2] Bioconductor  
 zoo                      1.8-7    2020-01-10 [2] CRAN (R 3.6.2)

[1] /home/steveped/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
