<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Steve Pederson" />


<title>Enrichment Analysis: Mutant Vs Wild-Type</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">20170327_Psen2S4Ter_RNASeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Enrichment Analysis: Mutant Vs Wild-Type</h1>
<h4 class="author">Steve Pederson</h4>
<h4 class="date">07 March, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-03-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>20170327_Psen2S4Ter_RNASeq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200119code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200119)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200119code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomUofABioinformaticsHub20170327Psen2S4TerRNASeqtree65ae03b1eb0182515c042bbbe9358ff2958ec1a3targetblank65ae03ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/tree/65ae03b1eb0182515c042bbbe9358ff2958ec1a3" target="_blank">65ae03b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomUofABioinformaticsHub20170327Psen2S4TerRNASeqtree65ae03b1eb0182515c042bbbe9358ff2958ec1a3targetblank65ae03ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  output/AllResults.xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/65ae03b1eb0182515c042bbbe9358ff2958ec1a3/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">65ae03b</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-07
</td>
<td>
Typo
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/3b120d98a05b282f069d3c032b8255fabd97d32e/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">3b120d9</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-07
</td>
<td>
Corrected table justifications
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/a5cdf5d6804cdddeb202915519f7e0588ea4648a/docs/3_Enrichment_MutantVsWT.html" target="_blank">a5cdf5d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
<td>
Finished Revisions
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/2772d42413d55a72b869d7b6929a15ddcd0f2bb6/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">2772d42</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
<td>
Corrected UpSet names
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/a9781d973379ddbcb43886194de4e7ba7afa7ba0/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">a9781d9</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
<td>
Fixed duplicate labels
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/154085bd3661ef17a5a5260df5f3b3c539978b24/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">154085b</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
<td>
Reduced ranked lists to fry only
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/2aa6e06392d2e643c7fefc36c4e661f472ce9f0d/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">2aa6e06</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
<td>
Removed gsea, camera &amp; combining data. Now only using fry + goseq
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/433a7294defb4f538bb7e0db7e18c764c798b225/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">433a729</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
<td>
Prepared to remove EGSEA-type approach
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/956c06574bb65a9ff9a96d32d44aaf0d7ca83be2/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">956c065</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-05
</td>
<td>
Revised GO exploration
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/1385ceb4a1970ff77ddd3c38195e150aa718edf7/docs/3_Enrichment_MutantVsWT.html" target="_blank">1385ceb</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-05
</td>
<td>
Compiled after renaming columns
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/2c56cefee54ceee830d89b3ef09f6063b5a5b27d/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">2c56cef</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-05
</td>
<td>
Tidied column names for enrichment output
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/7fe1505995bf93217d6ecd0c1de1f60f27e27c26/docs/3_Enrichment_MutantVsWT.html" target="_blank">7fe1505</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-05
</td>
<td>
Updated all analyses &amp; heatmaps after changing to cpmPostNorm instead of fit<span class="math inline">\(fitted.values for enrichment/heatmaps&lt;/td&gt; &lt;/tr&gt; &lt;tr&gt; &lt;td&gt;Rmd&lt;/td&gt; &lt;td&gt;&lt;a href=&quot;https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/bad81c1731ba8bc4487d25ec2719e92ae6917e4a/analysis/3_Enrichment_MutantVsWT.Rmd&quot; target=&quot;_blank&quot;&gt;bad81c1&lt;/a&gt;&lt;/td&gt; &lt;td&gt;Steve Ped&lt;/td&gt; &lt;td&gt;2020-03-05&lt;/td&gt; &lt;td&gt;Updated MutVsWt UpSet plots for KEGG/HALLMARK&lt;/td&gt; &lt;/tr&gt; &lt;tr&gt; &lt;td&gt;Rmd&lt;/td&gt; &lt;td&gt;&lt;a href=&quot;https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/377410a12ca13a7e233e39fb2f77f165e846d830/analysis/3_Enrichment_MutantVsWT.Rmd&quot; target=&quot;_blank&quot;&gt;377410a&lt;/a&gt;&lt;/td&gt; &lt;td&gt;Steve Ped&lt;/td&gt; &lt;td&gt;2020-03-05&lt;/td&gt; &lt;td&gt;Finished revision of Enrichment Analysis&lt;/td&gt; &lt;/tr&gt; &lt;tr&gt; &lt;td&gt;Rmd&lt;/td&gt; &lt;td&gt;&lt;a href=&quot;https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/c0b89ed8f761b0c7222538c353d21cd5aa402b94/analysis/3_Enrichment_MutantVsWT.Rmd&quot; target=&quot;_blank&quot;&gt;c0b89ed&lt;/a&gt;&lt;/td&gt; &lt;td&gt;Steve Ped&lt;/td&gt; &lt;td&gt;2020-03-05&lt;/td&gt; &lt;td&gt;Reran after changing to cpmPotNorm instead of the incorrect fit\)</span>fitted.values
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/7e59d3c3eeb91b99f90f7d919772149389e7b85a/docs/3_Enrichment_MutantVsWT.html" target="_blank">7e59d3c</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
<td>
Recompiled
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/4ede7527068c637893720c4e3e9511258b136320/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">4ede752</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
<td>
Added headings
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/01688249b9535123903e827dfd11a894c24ce8d8/docs/3_Enrichment_MutantVsWT.html" target="_blank">0168824</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
<td>
Rebult after resizing
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/de9c4b30542683957dc03d9ed779638e162750c5/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">de9c4b3</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
<td>
Resized heatmaps again
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/86ffa07adc34510ac982dcb32159b051f50a7b9a/docs/3_Enrichment_MutantVsWT.html" target="_blank">86ffa07</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
<td>
Updated saample layout figure
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/4d2af23dbbdd8c6f32404a1df10f65fcf0dbb266/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">4d2af23</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
<td>
Modified heatmap sizes &amp; initial network layout
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/8fe35b027e18fc26afa1c9029b907a71c2a2e083/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">8fe35b0</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
<td>
Added multiple heatmaps &amp; UpSet plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/679b10d0c260303f8b87b439535ebf16aab302ac/docs/3_Enrichment_MutantVsWT.html" target="_blank">679b10d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-19
</td>
<td>
Rendered OXPHOS Plot
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/66ba4e059abaaa0cb27a59cf949a530f31216bf7/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">66ba4e0</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-19
</td>
<td>
Added OXPHOS plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/e0288c6ba5aec521ebe8beea27d31941a6d303f7/docs/3_Enrichment_MutantVsWT.html" target="_blank">e0288c6</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-19
</td>
<td>
Generated enrichment tables
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/541fbf03bb7a1f3bcc98da5cdee8b493ace38637/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">541fbf0</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-19
</td>
<td>
Revised Hom Vs Het Enrichment
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/77f167d0437034fc61acae57b7bea02b00d06d24/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">77f167d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-19
</td>
<td>
Revised enrichment
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/876e40fc8b5e69e983e028d62c49f977da71a90d/docs/3_Enrichment_MutantVsWT.html" target="_blank">876e40f</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-17
</td>
<td>
Compiled after minor corrections
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/53ed0e30b1eff5b43093d31a0de4443f96125be6/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">53ed0e3</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-17
</td>
<td>
Corrected Ensembl Release
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/f55be858a66d2812a95303ad7146f952727953ba/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">f55be85</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-17
</td>
<td>
Added commas
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/8f29458fe6b2b7523c023b18629b270b2e500425/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">8f29458</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-17
</td>
<td>
Minor tweaks to formatting
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/9104ecdc9ad8512ec277c6c131d36de86edcf845/docs/3_Enrichment_MutantVsWT.html" target="_blank">9104ecd</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
First draft of Hom Vs Het
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/207cdc8d7398f80115e58d65fef3a9d2f0bbf044/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">207cdc8</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Added code for Hom Vs Het Enrichment
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/468e6e3a5e47e1e25be13e92d15dc4e1d7067130/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">468e6e3</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Ran enrichment of Mut Vs WT
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/468e6e3a5e47e1e25be13e92d15dc4e1d7067130/docs/3_Enrichment_MutantVsWT.html" target="_blank">468e6e3</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Ran enrichment of Mut Vs WT
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/3a9933c5330cee6a4957bdfdf7794251f9346dd2/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">3a9933c</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Finished Enrichment analysis on MutVsWt
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/3a9933c5330cee6a4957bdfdf7794251f9346dd2/docs/3_Enrichment_MutantVsWT.html" target="_blank">3a9933c</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Finished Enrichment analysis on MutVsWt
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level1">
<h1>Setup</h1>
<pre class="r"><code>library(tidyverse)
library(magrittr)
library(edgeR)
library(scales)
library(pander)
library(goseq)
library(msigdbr)
library(AnnotationDbi)
library(RColorBrewer)
library(ngsReports)
library(UpSetR)
library(pheatmap)</code></pre>
<pre class="r"><code>theme_set(theme_bw())
panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;table.style&quot;, &quot;rmarkdown&quot;)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)</code></pre>
<pre class="r"><code>samples &lt;- read_csv(&quot;data/samples.csv&quot;) %&gt;%
  distinct(sampleName, .keep_all = TRUE) %&gt;%
  dplyr::select(sample = sampleName, sampleID, genotype) %&gt;%
  mutate(
    genotype = factor(genotype, levels = c(&quot;WT&quot;, &quot;Het&quot;, &quot;Hom&quot;)),
    mutant = genotype %in% c(&quot;Het&quot;, &quot;Hom&quot;),
    homozygous = genotype == &quot;Hom&quot;
  )
genoCols &lt;- samples$genotype %&gt;%
  levels() %&gt;%
  length() %&gt;%
  brewer.pal(&quot;Set1&quot;) %&gt;%
  setNames(levels(samples$genotype))</code></pre>
<pre class="r"><code>dgeList &lt;- here::here(&quot;data/dgeList.rds&quot;) %&gt;% read_rds()
cpmPostNorm &lt;- here::here(&quot;data/cpmPostNorm.rds&quot;) %&gt;% read_rds()
entrezGenes &lt;- dgeList$genes %&gt;%
  dplyr::filter(!is.na(entrezid)) %&gt;%
  unnest(entrezid) %&gt;%
  dplyr::rename(entrez_gene = entrezid)</code></pre>
<pre class="r"><code>deTable &lt;- here::here(&quot;output&quot;, &quot;psen2VsWT.csv&quot;) %&gt;% 
  read_csv() %&gt;%
  mutate(
    entrezid = dgeList$genes$entrezid[gene_id]
  )</code></pre>
<pre class="r"><code>formatP &lt;- function(p, m = 0.0001){
  out &lt;- rep(&quot;&quot;, length(p))
  out[p &lt; m] &lt;- sprintf(&quot;%.2e&quot;, p[p&lt;m])
  out[p &gt;= m] &lt;- sprintf(&quot;%.4f&quot;, p[p&gt;=m])
  out
}</code></pre>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Enrichment analysis for this dataset present some challenges. Despite normalisation to account for gene length and GC bias, some appeared to still be present in the final results. In addition, the confounding of incomplete rRNA removal with genotype may lead to other distortions in both DE genes and ranking statistics.</p>
<p>Two steps for enrichment analysis will be undertaken.</p>
<ol style="list-style-type: decimal">
<li>Testing for enrichment within <em>discrete sets of DE genes</em> as defined in the previous steps</li>
<li>Testing for enrichment using the complete set of results using <code>fry</code>.</li>
</ol>
<p>Testing for enrichment <em>within discrete gene sets</em> will be performed using <code>goseq</code> as this allows for the incorporation of a single covariate as a predictor of differential expression. GC content, gene length and correlation with rRNA removal can all be supplied as separate covariates.</p>
<p>For enrichment within larger gene lists, <code>fry</code> can take into account inter-gene correlations. Values supplied will be logCPM for each gene/sample after being adjusted for GC and length biases.</p>
</div>
<div id="databases-used-for-testing" class="section level1">
<h1>Databases used for testing</h1>
<p>Data was sourced using the <code>msigdbr</code> package. The initial database used for testing was the Hallmark Gene Sets, with mappings from gene-set to EntrezGene IDs performed by the package authors.</p>
<div id="hallmark-gene-sets" class="section level2">
<h2>Hallmark Gene Sets</h2>
<pre class="r"><code>hm &lt;- msigdbr(&quot;Danio rerio&quot;, category = &quot;H&quot;)  %&gt;% 
  left_join(entrezGenes) %&gt;%
  dplyr::filter(!is.na(gene_id)) %&gt;%
  distinct(gs_name, gene_id, .keep_all = TRUE)
hmByGene &lt;- hm %&gt;%
  split(f = .$gene_id) %&gt;%
  lapply(extract2, &quot;gs_name&quot;)
hmByID &lt;- hm %&gt;%
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;gene_id&quot;)</code></pre>
<p>Mappings are required from gene to pathway, and Ensembl identifiers were used to map from gene to pathway, based on the mappings in the previously used annotations (Ensembl Release 98). A total of 3,459 Ensembl IDs were mapped to pathways from the hallmark gene sets.</p>
</div>
<div id="kegg-gene-sets" class="section level2">
<h2>KEGG Gene Sets</h2>
<pre class="r"><code>kg &lt;- msigdbr(&quot;Danio rerio&quot;, category = &quot;C2&quot;, subcategory = &quot;CP:KEGG&quot;)  %&gt;% 
  left_join(entrezGenes) %&gt;%
  dplyr::filter(!is.na(gene_id)) %&gt;%
  distinct(gs_name, gene_id, .keep_all = TRUE)
kgByGene &lt;- kg  %&gt;%
  split(f = .$gene_id) %&gt;%
  lapply(extract2, &quot;gs_name&quot;)
kgByID &lt;- kg  %&gt;%
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;gene_id&quot;)</code></pre>
<p>The same mapping process was applied to KEGG gene sets. A total of 3,614 Ensembl IDs were mapped to pathways from the KEGG gene sets.</p>
</div>
<div id="gene-ontology-gene-sets" class="section level2">
<h2>Gene Ontology Gene Sets</h2>
<pre class="r"><code>goSummaries &lt;- url(&quot;https://uofabioinformaticshub.github.io/summaries2GO/data/goSummaries.RDS&quot;) %&gt;%
  readRDS() %&gt;%
  mutate(
    Term = Term(id),
    gs_name = Term %&gt;% str_to_upper() %&gt;% str_replace_all(&quot;[ -]&quot;, &quot;_&quot;),
    gs_name = paste0(&quot;GO_&quot;, gs_name)
    )
minPath &lt;- 3</code></pre>
<pre class="r"><code>go &lt;- msigdbr(&quot;Danio rerio&quot;, category = &quot;C5&quot;) %&gt;% 
  left_join(entrezGenes) %&gt;%
  dplyr::filter(!is.na(gene_id)) %&gt;%
  left_join(goSummaries) %&gt;% 
  dplyr::filter(shortest_path &gt;= minPath) %&gt;%
  distinct(gs_name, gene_id, .keep_all = TRUE)
goByGene &lt;- go %&gt;%
  split(f = .$gene_id) %&gt;%
  lapply(extract2, &quot;gs_name&quot;)
goByID &lt;- go %&gt;%
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;gene_id&quot;)</code></pre>
<p>For analysis of gene-sets from the GO database, gene-sets were restricted to those with 3 or more steps back to the ontology root terms. A total of 11,245 Ensembl IDs were mapped to pathways from restricted database of 8,834 GO gene sets.</p>
<pre class="r"><code>gsSizes &lt;- bind_rows(hm, kg, go) %&gt;% 
  dplyr::select(gs_name, gene_symbol, gene_id) %&gt;% 
  chop(c(gene_symbol, gene_id)) %&gt;%
  mutate(
    gs_size = vapply(gene_symbol, length, integer(1)),
    de_id = lapply(
      X = gene_id, 
      FUN = intersect, 
      y = dplyr::filter(deTable, DE)$gene_id
      ),
    de_size = vapply(de_id, length, integer(1))
  )</code></pre>
</div>
</div>
<div id="enrichment-in-the-de-gene-set" class="section level1">
<h1>Enrichment in the DE Gene Set</h1>
<p>The first step of analysis using <code>goseq</code>, regardless of the gene-set, is estimation of the probability weight function (PWF) which quantifies the probability of a gene being considered as DE based on a single covariate. As GC content and length should have been accounted for during conditional-quantile normalisation, these were not required for any bias. However, the <em>gene-level correlations with rRNA contamination</em> were instead used a predictor of bias in selection of a gene as being DE.</p>
<pre class="r"><code>rawFqc &lt;- list.files(
  path = here::here(&quot;data/0_rawData/FastQC/&quot;),
  pattern = &quot;zip&quot;,
  full.names = TRUE
) %&gt;%
  FastqcDataList()
gc &lt;- getModule(rawFqc, &quot;Per_sequence_GC&quot;) %&gt;%
  group_by(Filename) %&gt;% 
  mutate(Freq = Count / sum(Count)) %&gt;%
  ungroup()
gcDev &lt;- gc %&gt;%
  left_join(getGC(gcTheoretical, &quot;Drerio&quot;, &quot;Trans&quot;)) %&gt;%
  mutate(
    sample = str_remove(Filename, &quot;_R[12].fastq.gz&quot;),
    resid = Freq - Drerio
  ) %&gt;% 
  left_join(samples) %&gt;%
  group_by(sample) %&gt;%
  summarise(
    ss = sum(resid^2),
    n = n(),
    sd = sqrt(ss / (n - 1))
  )</code></pre>
<pre class="r"><code>riboVec &lt;- structure(gcDev$sd, names = gcDev$sample)
riboCors &lt;- cpm(dgeList, log = TRUE) %&gt;%
  apply(1, function(x){
    cor(x, riboVec[names(x)])
  })</code></pre>
<p>Values were calculated as per the previous steps, using the logCPM values for each gene, with the sample-level standard deviations from the theoretical GC distribution being used as a measure of rRNA contamination. For estimation of the probability weight function, squared correlations were used to place negative and positive correlations on the same scale. This accounts for genes which are both negatively &amp; positively biased by the presence of excessive rRNA. Clearly, the confounding of genotype with rRNA means that some genes driven by the genuine biology may be down-weighted under this approach.</p>
<pre class="r"><code>riboPwf &lt;- deTable %&gt;%
  mutate(riboCors = riboCors[gene_id]^2) %&gt;%
  dplyr::select(gene_id, DE, riboCors) %&gt;%
  distinct(gene_id, .keep_all = TRUE) %&gt;% 
  with(
    nullp(
      DEgenes = structure(
        as.integer(DE), names = gene_id
      ), 
      genome = &quot;danRer10&quot;, 
      id = &quot;ensGene&quot;, 
      bias.data = riboCors,
      plot.fit = FALSE
    )
  )
plotPWF(riboPwf, main = &quot;Bias from rRNA proportions&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/riboPwf-1.png" alt="*Using this approach, it was clear that correlation with rRNA proportions significantly biased the probability of a gene being considered as DE.*" width="384" />
<p class="caption">
<em>Using this approach, it was clear that correlation with rRNA proportions significantly biased the probability of a gene being considered as DE.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-riboPwf-1">
Past versions of riboPwf-1.png
</button>
</p>
<div id="fig-riboPwf-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/468e6e3a5e47e1e25be13e92d15dc4e1d7067130/docs/figure/3_Enrichment_MutantVsWT.Rmd/riboPwf-1.png" target="_blank">468e6e3</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>All gene-sets were then tested using this PWF.</p>
<div id="hallmark-gene-sets-1" class="section level2">
<h2>Hallmark Gene Sets</h2>
<pre class="r"><code>hmRiboGoseq &lt;- goseq(riboPwf, gene2cat = hmByGene) %&gt;%
  as_tibble %&gt;%
  dplyr::filter(numDEInCat &gt; 0) %&gt;%
  mutate( 
    adjP = p.adjust(over_represented_pvalue, method = &quot;bonf&quot;),
    FDR = p.adjust(over_represented_pvalue, method = &quot;fdr&quot;)
  ) %&gt;%
  dplyr::select(-contains(&quot;under&quot;)) %&gt;%
  dplyr::rename(
    gs_name = category,
    PValue = over_represented_pvalue
  )</code></pre>
<p>No gene-sets achieved significance in the DE genes with the lowest FDR being 41%</p>
</div>
<div id="kegg-gene-sets-1" class="section level2">
<h2>KEGG Gene Sets</h2>
<pre class="r"><code>kgRiboGoseq &lt;- goseq(riboPwf, gene2cat = kgByGene) %&gt;%
  as_tibble %&gt;%
  dplyr::filter(numDEInCat &gt; 0) %&gt;%
  mutate( 
    adjP = p.adjust(over_represented_pvalue, method = &quot;bonf&quot;),
    FDR = p.adjust(over_represented_pvalue, method = &quot;fdr&quot;)
  ) %&gt;%
  dplyr::select(-contains(&quot;under&quot;))  %&gt;%
  dplyr::rename(
    gs_name = category,
    PValue = over_represented_pvalue
  )</code></pre>
<pre class="r"><code>kgRiboGoseq %&gt;%
  dplyr::slice(1:5) %&gt;%
  mutate(
    p = formatP(PValue),
    adjP = formatP(adjP),
    FDR = formatP(FDR)
  ) %&gt;%
  dplyr::select(
    `Gene Set` = gs_name,
    DE = numDEInCat,
    `Set Size` = numInCat,
    PValue,
    `p~bonf~` = adjP,
    `p~FDR~` = FDR
  ) %&gt;%
  pander(
    justify = &quot;lrrrrr&quot;,
    caption = paste(
      &quot;The&quot;, nrow(.), &quot;most highly-ranked KEGG pathways.&quot;,
      &quot;Bonferroni-adjusted p-values are the most stringent and give high&quot;,
      &quot;confidence when below 0.05.&quot;
    )
  )</code></pre>
<table>
<caption>The 5 most highly-ranked KEGG pathways. Bonferroni-adjusted p-values are the most stringent and give high confidence when below 0.05.</caption>
<thead>
<tr class="header">
<th align="left">Gene Set</th>
<th align="right">DE</th>
<th align="right">Set Size</th>
<th align="right">PValue</th>
<th align="right">p<sub>bonf</sub></th>
<th align="right">p<sub>FDR</sub></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">KEGG_RIBOSOME</td>
<td align="right">26</td>
<td align="right">80</td>
<td align="right">4.754e-09</td>
<td align="right">5.32e-07</td>
<td align="right">5.32e-07</td>
</tr>
<tr class="even">
<td align="left">KEGG_PRIMARY_IMMUNODEFICIENCY</td>
<td align="right">2</td>
<td align="right">15</td>
<td align="right">0.01642</td>
<td align="right">1.0000</td>
<td align="right">0.7173</td>
</tr>
<tr class="odd">
<td align="left">KEGG_CYSTEINE_AND_METHIONINE_METABOLISM</td>
<td align="right">4</td>
<td align="right">26</td>
<td align="right">0.02959</td>
<td align="right">1.0000</td>
<td align="right">0.7173</td>
</tr>
<tr class="even">
<td align="left">KEGG_ASTHMA</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">0.04088</td>
<td align="right">1.0000</td>
<td align="right">0.7173</td>
</tr>
<tr class="odd">
<td align="left">KEGG_RETINOL_METABOLISM</td>
<td align="right">3</td>
<td align="right">25</td>
<td align="right">0.0512</td>
<td align="right">1.0000</td>
<td align="right">0.7173</td>
</tr>
</tbody>
</table>
<p>Notably, the KEGG gene-set for Ribosomal genes was detected as enriched in the set of DE genes, with no other KEGG gene-sets being considered significant.</p>
</div>
<div id="go-gene-sets" class="section level2">
<h2>GO Gene Sets</h2>
<pre class="r"><code>goRiboGoseq &lt;- goseq(riboPwf, gene2cat = goByGene) %&gt;%
  as_tibble %&gt;%
  dplyr::filter(numDEInCat &gt; 0) %&gt;%
  mutate( 
    adjP = p.adjust(over_represented_pvalue, method = &quot;bonf&quot;),
    FDR = p.adjust(over_represented_pvalue, method = &quot;fdr&quot;)
  ) %&gt;%
  dplyr::select(-contains(&quot;under&quot;)) %&gt;%
  dplyr::rename(
    gs_name = category,
    PValue = over_represented_pvalue
  )</code></pre>
<pre class="r"><code>goRiboGoseq %&gt;%
  dplyr::filter(adjP &lt; 0.05) %&gt;%
  mutate(
    p = formatP(PValue),
    adjP = formatP(adjP),
    FDR = formatP(FDR)
  ) %&gt;%
  dplyr::select(
    `Gene Set` = gs_name,
    DE = numDEInCat,
    `Set Size` = numInCat,
    PValue,
    `p~bonf~` = adjP,
    `p~FDR~` = FDR
  ) %&gt;%
  pander(
    justify = &quot;lrrrrr&quot;,
    caption = paste(
      &quot;*The&quot;, nrow(.), &quot;most highly-ranked GO terms.&quot;,
      &quot;Bonferroni-adjusted p-values are the most stringent and give high&quot;,
      &quot;confidence when below 0.05, with all terms here reaching this threshold.&quot;,
      &quot;However, most terms once again indicate the presence of rRNA.*&quot;
    )
  )</code></pre>
<table>
<caption><em>The 14 most highly-ranked GO terms. Bonferroni-adjusted p-values are the most stringent and give high confidence when below 0.05, with all terms here reaching this threshold. However, most terms once again indicate the presence of rRNA.</em></caption>
<colgroup>
<col width="59%" />
<col width="3%" />
<col width="8%" />
<col width="9%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Gene Set</th>
<th align="right">DE</th>
<th align="right">Set Size</th>
<th align="right">PValue</th>
<th align="right">p<sub>bonf</sub></th>
<th align="right">p<sub>FDR</sub></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE</td>
<td align="right">28</td>
<td align="right">93</td>
<td align="right">7.36e-10</td>
<td align="right">2.83e-06</td>
<td align="right">2.03e-06</td>
</tr>
<tr class="even">
<td align="left">GO_CYTOSOLIC_RIBOSOME</td>
<td align="right">27</td>
<td align="right">97</td>
<td align="right">1.056e-09</td>
<td align="right">4.06e-06</td>
<td align="right">2.03e-06</td>
</tr>
<tr class="odd">
<td align="left">GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM</td>
<td align="right">27</td>
<td align="right">105</td>
<td align="right">1.6e-08</td>
<td align="right">6.16e-05</td>
<td align="right">2.05e-05</td>
</tr>
<tr class="even">
<td align="left">GO_TRANSLATIONAL_INITIATION</td>
<td align="right">29</td>
<td align="right">177</td>
<td align="right">2.178e-07</td>
<td align="right">0.0008</td>
<td align="right">0.0002</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEIN_TARGETING_TO_MEMBRANE</td>
<td align="right">30</td>
<td align="right">171</td>
<td align="right">2.533e-07</td>
<td align="right">0.0010</td>
<td align="right">0.0002</td>
</tr>
<tr class="even">
<td align="left">GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM</td>
<td align="right">27</td>
<td align="right">128</td>
<td align="right">3.849e-07</td>
<td align="right">0.0015</td>
<td align="right">0.0002</td>
</tr>
<tr class="odd">
<td align="left">GO_CYTOSOLIC_PART</td>
<td align="right">30</td>
<td align="right">206</td>
<td align="right">4.424e-07</td>
<td align="right">0.0017</td>
<td align="right">0.0002</td>
</tr>
<tr class="even">
<td align="left">GO_VIRAL_GENE_EXPRESSION</td>
<td align="right">29</td>
<td align="right">172</td>
<td align="right">6.939e-07</td>
<td align="right">0.0027</td>
<td align="right">0.0003</td>
</tr>
<tr class="odd">
<td align="left">GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS</td>
<td align="right">30</td>
<td align="right">183</td>
<td align="right">7.911e-07</td>
<td align="right">0.0030</td>
<td align="right">0.0003</td>
</tr>
<tr class="even">
<td align="left">GO_RIBOSOMAL_SUBUNIT</td>
<td align="right">28</td>
<td align="right">175</td>
<td align="right">1.99e-06</td>
<td align="right">0.0077</td>
<td align="right">0.0008</td>
</tr>
<tr class="odd">
<td align="left">GO_RIBOSOME</td>
<td align="right">30</td>
<td align="right">210</td>
<td align="right">2.659e-06</td>
<td align="right">0.0102</td>
<td align="right">0.0009</td>
</tr>
<tr class="even">
<td align="left">GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT</td>
<td align="right">15</td>
<td align="right">51</td>
<td align="right">2.826e-06</td>
<td align="right">0.0109</td>
<td align="right">0.0009</td>
</tr>
<tr class="odd">
<td align="left">GO_SYMPORTER_ACTIVITY</td>
<td align="right">10</td>
<td align="right">94</td>
<td align="right">7.571e-06</td>
<td align="right">0.0291</td>
<td align="right">0.0022</td>
</tr>
<tr class="even">
<td align="left">GO_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">17</td>
<td align="right">228</td>
<td align="right">1.23e-05</td>
<td align="right">0.0473</td>
<td align="right">0.0034</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="enrichment-testing-on-the-complete-set-of-genes" class="section level1">
<h1>Enrichment Testing on the Complete Set of Genes</h1>
<div id="hallmark-gene-sets-2" class="section level2">
<h2>Hallmark Gene Sets</h2>
<pre class="r"><code>hmFry &lt;- cpmPostNorm %&gt;%
  fry(
    index = hmByID,
    design = dgeList$design,
    contrast = &quot;mutant&quot;,
    sort = &quot;directional&quot;
    ) %&gt;%
  rownames_to_column(&quot;gs_name&quot;) %&gt;%
  as_tibble()</code></pre>
<pre class="r"><code>hmFry %&gt;%
  dplyr::filter(
    PValue &lt; 0.01
  ) %&gt;%
  left_join(gsSizes) %&gt;%
  mutate(gs_name = str_remove(gs_name, &quot;HALLMARK_&quot;)) %&gt;%
  dplyr::select(
    `Hallmark Gene Set` = gs_name,
    `Expressed Genes` = NGenes,
    `DE Genes` = de_size,
    Direction,
    PValue, FDR
    ) %&gt;%
  pander(
    justify = &quot;lrrlrr&quot;,
    caption = &quot;Enriched Hallmark Gene Sets, using a threshold of p &lt; 0.01, which corresponds to an FDR of 0.02&quot;
  )</code></pre>
<table>
<caption>Enriched Hallmark Gene Sets, using a threshold of p &lt; 0.01, which corresponds to an FDR of 0.02</caption>
<thead>
<tr class="header">
<th align="left">Hallmark Gene Set</th>
<th align="right">Expressed Genes</th>
<th align="right">DE Genes</th>
<th align="left">Direction</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">UV_RESPONSE_UP</td>
<td align="right">141</td>
<td align="right">7</td>
<td align="left">Up</td>
<td align="right">4.187e-06</td>
<td align="right">0.0002094</td>
</tr>
<tr class="even">
<td align="left">XENOBIOTIC_METABOLISM</td>
<td align="right">146</td>
<td align="right">11</td>
<td align="left">Up</td>
<td align="right">8.577e-05</td>
<td align="right">0.001756</td>
</tr>
<tr class="odd">
<td align="left">REACTIVE_OXYGEN_SPECIES_PATHWAY</td>
<td align="right">45</td>
<td align="right">5</td>
<td align="left">Up</td>
<td align="right">0.0001128</td>
<td align="right">0.001756</td>
</tr>
<tr class="even">
<td align="left">FATTY_ACID_METABOLISM</td>
<td align="right">135</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">0.0001574</td>
<td align="right">0.001756</td>
</tr>
<tr class="odd">
<td align="left">MTORC1_SIGNALING</td>
<td align="right">194</td>
<td align="right">11</td>
<td align="left">Up</td>
<td align="right">0.0001756</td>
<td align="right">0.001756</td>
</tr>
<tr class="even">
<td align="left">GLYCOLYSIS</td>
<td align="right">172</td>
<td align="right">8</td>
<td align="left">Up</td>
<td align="right">0.0002864</td>
<td align="right">0.002387</td>
</tr>
<tr class="odd">
<td align="left">PEROXISOME</td>
<td align="right">95</td>
<td align="right">2</td>
<td align="left">Up</td>
<td align="right">0.0004238</td>
<td align="right">0.003027</td>
</tr>
<tr class="even">
<td align="left">ADIPOGENESIS</td>
<td align="right">186</td>
<td align="right">13</td>
<td align="left">Up</td>
<td align="right">0.0005475</td>
<td align="right">0.003236</td>
</tr>
<tr class="odd">
<td align="left">BILE_ACID_METABOLISM</td>
<td align="right">86</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.0006345</td>
<td align="right">0.003236</td>
</tr>
<tr class="even">
<td align="left">PROTEIN_SECRETION</td>
<td align="right">92</td>
<td align="right">3</td>
<td align="left">Up</td>
<td align="right">0.0006473</td>
<td align="right">0.003236</td>
</tr>
<tr class="odd">
<td align="left">COAGULATION</td>
<td align="right">80</td>
<td align="right">3</td>
<td align="left">Up</td>
<td align="right">0.0008365</td>
<td align="right">0.003624</td>
</tr>
<tr class="even">
<td align="left">UNFOLDED_PROTEIN_RESPONSE</td>
<td align="right">113</td>
<td align="right">4</td>
<td align="left">Up</td>
<td align="right">0.0008697</td>
<td align="right">0.003624</td>
</tr>
<tr class="odd">
<td align="left">CHOLESTEROL_HOMEOSTASIS</td>
<td align="right">69</td>
<td align="right">2</td>
<td align="left">Up</td>
<td align="right">0.001557</td>
<td align="right">0.005666</td>
</tr>
<tr class="even">
<td align="left">NOTCH_SIGNALING</td>
<td align="right">29</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.001623</td>
<td align="right">0.005666</td>
</tr>
<tr class="odd">
<td align="left">ESTROGEN_RESPONSE_LATE</td>
<td align="right">153</td>
<td align="right">3</td>
<td align="left">Up</td>
<td align="right">0.0017</td>
<td align="right">0.005666</td>
</tr>
<tr class="even">
<td align="left">MYC_TARGETS_V1</td>
<td align="right">192</td>
<td align="right">15</td>
<td align="left">Up</td>
<td align="right">0.0021</td>
<td align="right">0.005993</td>
</tr>
<tr class="odd">
<td align="left">HYPOXIA</td>
<td align="right">169</td>
<td align="right">9</td>
<td align="left">Up</td>
<td align="right">0.002131</td>
<td align="right">0.005993</td>
</tr>
<tr class="even">
<td align="left">OXIDATIVE_PHOSPHORYLATION</td>
<td align="right">201</td>
<td align="right">15</td>
<td align="left">Up</td>
<td align="right">0.002158</td>
<td align="right">0.005993</td>
</tr>
<tr class="odd">
<td align="left">EPITHELIAL_MESENCHYMAL_TRANSITION</td>
<td align="right">156</td>
<td align="right">7</td>
<td align="left">Up</td>
<td align="right">0.003453</td>
<td align="right">0.009086</td>
</tr>
<tr class="even">
<td align="left">TNFA_SIGNALING_VIA_NFKB</td>
<td align="right">149</td>
<td align="right">7</td>
<td align="left">Up</td>
<td align="right">0.004565</td>
<td align="right">0.01141</td>
</tr>
<tr class="odd">
<td align="left">IL2_STAT5_SIGNALING</td>
<td align="right">147</td>
<td align="right">2</td>
<td align="left">Up</td>
<td align="right">0.005095</td>
<td align="right">0.01213</td>
</tr>
<tr class="even">
<td align="left">ANDROGEN_RESPONSE</td>
<td align="right">90</td>
<td align="right">2</td>
<td align="left">Up</td>
<td align="right">0.006364</td>
<td align="right">0.01446</td>
</tr>
<tr class="odd">
<td align="left">PI3K_AKT_MTOR_SIGNALING</td>
<td align="right">98</td>
<td align="right">2</td>
<td align="left">Up</td>
<td align="right">0.00925</td>
<td align="right">0.02011</td>
</tr>
</tbody>
</table>
<div id="exploration-of-significant-hallmark-gene-sets" class="section level3">
<h3>Exploration of significant Hallmark Gene Sets</h3>
<pre class="r"><code>hmFry %&gt;% 
  left_join(gsSizes) %&gt;%
  dplyr::filter(de_size &gt;= 5, FDR &lt; 0.05) %&gt;%
  dplyr::select(gs_name, de_id) %&gt;%
  unnest(de_id) %&gt;% 
  mutate(
    gs_name = str_remove(gs_name, &quot;HALLMARK_&quot;),
    gs_name = fct_lump(gs_name, n = 14)
    ) %&gt;% 
  split(f = .$gs_name) %&gt;% 
  lapply(magrittr::extract2, &quot;de_id&quot;) %&gt;% 
  fromList() %&gt;% 
  upset(
    nsets = length(.), 
    nintersects = 20,
    order.by = &quot;freq&quot;, 
    mb.ratio = c(0.6, 0.4),
    sets.x.label = &quot;Number Of DE Genes&quot;
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/hmFryUpset-1.png" alt="*UpSet plot indicating distribution of DE genes within all significant HALLMARK gene sets. Gene sets were restricted to those with an FDR &lt; 0.05 and at least 5 DE genes. The plot is truncated at the right for simplicity. Most gene sets seem relatively independent of each other with regard to DE genes.*" width="960" />
<p class="caption">
<em>UpSet plot indicating distribution of DE genes within all significant HALLMARK gene sets. Gene sets were restricted to those with an FDR &lt; 0.05 and at least 5 DE genes. The plot is truncated at the right for simplicity. Most gene sets seem relatively independent of each other with regard to DE genes.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-hmFryUpset-1">
Past versions of hmFryUpset-1.png
</button>
</p>
<div id="fig-hmFryUpset-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/a5cdf5d6804cdddeb202915519f7e0588ea4648a/docs/figure/3_Enrichment_MutantVsWT.Rmd/hmFryUpset-1.png" target="_blank">a5cdf5d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>hmHeat &lt;- hmFry %&gt;% 
  left_join(gsSizes) %&gt;%
  dplyr::filter(de_size &gt;= 12, FDR &lt; 0.05) %&gt;%
  dplyr::select(gs_name, de_id) %&gt;%
  unnest(de_id) %&gt;%
  left_join(dgeList$genes, by = c(&quot;de_id&quot; = &quot;gene_id&quot;)) %&gt;%
  dplyr::select(gs_name, de_id, gene_name) %&gt;%
  mutate(belongs = TRUE) %&gt;%
  pivot_wider(
    id_cols = c(de_id, gene_name),
    names_from = gs_name,
    values_from = belongs,
    values_fill = list(belongs = FALSE)
  ) %&gt;%
  left_join(
    cpmPostNorm %&gt;%
      as.data.frame() %&gt;%
      rownames_to_column(&quot;de_id&quot;)
  ) 
hmHeat %&gt;%
  dplyr::select(gene_name, starts_with(&quot;Ps&quot;)) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(&quot;gene_name&quot;) %&gt;%
  as.matrix() %&gt;%
  pheatmap(
    color = viridis_pal(option = &quot;magma&quot;)(100),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), &quot;logCPM\n&quot;),
    labels_col = hmHeat %&gt;%
      dplyr::select(starts_with(&quot;Ps&quot;)) %&gt;% 
      colnames() %&gt;% 
      enframe(name = NULL) %&gt;% 
      dplyr::rename(sample = value) %&gt;%
      left_join(dgeList$samples) %&gt;%
      dplyr::select(sample, sampleID) %&gt;%
      with(
        structure(sampleID, names = sample)
      ),
    cutree_rows = 6,
    cutree_cols = 2,
    annotation_names_row = FALSE,
    annotation_row = hmHeat %&gt;%
      dplyr::select(gene_name, starts_with(&quot;HALLMARK_&quot;)) %&gt;%
      mutate_at(vars(starts_with(&quot;HALL&quot;)), as.character) %&gt;%
      as.data.frame() %&gt;%
      column_to_rownames(&quot;gene_name&quot;) %&gt;%
      set_colnames(
        str_remove(colnames(.), &quot;HALLMARK_&quot;)
      )
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/hmHeat-1.png" alt="*Gene expression patterns for all DE genes in HALLMARK gene sets, containing more than 12 DE genes.*" width="960" />
<p class="caption">
<em>Gene expression patterns for all DE genes in HALLMARK gene sets, containing more than 12 DE genes.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-hmHeat-1">
Past versions of hmHeat-1.png
</button>
</p>
<div id="fig-hmHeat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/a5cdf5d6804cdddeb202915519f7e0588ea4648a/docs/figure/3_Enrichment_MutantVsWT.Rmd/hmHeat-1.png" target="_blank">a5cdf5d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7fe1505995bf93217d6ecd0c1de1f60f27e27c26/docs/figure/3_Enrichment_MutantVsWT.Rmd/hmHeat-1.png" target="_blank">7fe1505</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/86ffa07adc34510ac982dcb32159b051f50a7b9a/docs/figure/3_Enrichment_MutantVsWT.Rmd/hmHeat-1.png" target="_blank">86ffa07</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="kegg-gene-sets-2" class="section level2">
<h2>KEGG Gene Sets</h2>
<pre class="r"><code>kgFry &lt;- cpmPostNorm %&gt;%
  fry(
    index = kgByID,
    design = dgeList$design,
    contrast = &quot;mutant&quot;,
    sort = &quot;directional&quot;
    ) %&gt;%
  rownames_to_column(&quot;gs_name&quot;) %&gt;%
  as_tibble()</code></pre>
<pre class="r"><code>kgFry %&gt;%
  dplyr::filter(
    PValue &lt; 0.005
  ) %&gt;%
  left_join(gsSizes) %&gt;%
  mutate(gs_name = str_remove(gs_name, &quot;HALLMARK_&quot;)) %&gt;%
  dplyr::select(
    `KEGG Gene Set` = gs_name,
    `Expressed Genes` = NGenes,
    `DE Genes` = de_size,
    Direction,
    PValue, FDR
    ) %&gt;%
  pander(
    justify = &quot;lrrlrr&quot;,
    caption = &quot;Enriched KEGG Gene Sets, using a threshold p &lt; 0.005, which corresponds to an FDR of 0.02&quot;
  )</code></pre>
<table>
<caption>Enriched KEGG Gene Sets, using a threshold p &lt; 0.005, which corresponds to an FDR of 0.02</caption>
<colgroup>
<col width="52%" />
<col width="13%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">KEGG Gene Set</th>
<th align="right">Expressed Genes</th>
<th align="right">DE Genes</th>
<th align="left">Direction</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">KEGG_GLUTATHIONE_METABOLISM</td>
<td align="right">36</td>
<td align="right">4</td>
<td align="left">Up</td>
<td align="right">1.313e-05</td>
<td align="right">0.002442</td>
</tr>
<tr class="even">
<td align="left">KEGG_GLYCINE_SERINE_AND_THREONINE_METABOLISM</td>
<td align="right">24</td>
<td align="right">2</td>
<td align="left">Up</td>
<td align="right">9.541e-05</td>
<td align="right">0.005307</td>
</tr>
<tr class="odd">
<td align="left">KEGG_ARGININE_AND_PROLINE_METABOLISM</td>
<td align="right">39</td>
<td align="right">2</td>
<td align="left">Up</td>
<td align="right">9.545e-05</td>
<td align="right">0.005307</td>
</tr>
<tr class="even">
<td align="left">KEGG_LIMONENE_AND_PINENE_DEGRADATION</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.0001542</td>
<td align="right">0.005307</td>
</tr>
<tr class="odd">
<td align="left">KEGG_PRIMARY_IMMUNODEFICIENCY</td>
<td align="right">15</td>
<td align="right">2</td>
<td align="left">Down</td>
<td align="right">0.0001638</td>
<td align="right">0.005307</td>
</tr>
<tr class="even">
<td align="left">KEGG_PORPHYRIN_AND_CHLOROPHYLL_METABOLISM</td>
<td align="right">25</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.0002012</td>
<td align="right">0.005307</td>
</tr>
<tr class="odd">
<td align="left">KEGG_ASCORBATE_AND_ALDARATE_METABOLISM</td>
<td align="right">7</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.000218</td>
<td align="right">0.005307</td>
</tr>
<tr class="even">
<td align="left">KEGG_BETA_ALANINE_METABOLISM</td>
<td align="right">18</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.0002525</td>
<td align="right">0.005307</td>
</tr>
<tr class="odd">
<td align="left">KEGG_GLYCEROLIPID_METABOLISM</td>
<td align="right">37</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.0002568</td>
<td align="right">0.005307</td>
</tr>
<tr class="even">
<td align="left">KEGG_PPAR_SIGNALING_PATHWAY</td>
<td align="right">53</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.0002947</td>
<td align="right">0.005397</td>
</tr>
<tr class="odd">
<td align="left">KEGG_FATTY_ACID_METABOLISM</td>
<td align="right">38</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.0003304</td>
<td align="right">0.005397</td>
</tr>
<tr class="even">
<td align="left">KEGG_BUTANOATE_METABOLISM</td>
<td align="right">26</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.0003879</td>
<td align="right">0.005397</td>
</tr>
<tr class="odd">
<td align="left">KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY</td>
<td align="right">79</td>
<td align="right">3</td>
<td align="left">Down</td>
<td align="right">0.0004045</td>
<td align="right">0.005397</td>
</tr>
<tr class="even">
<td align="left">KEGG_AMINOACYL_TRNA_BIOSYNTHESIS</td>
<td align="right">37</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.0004062</td>
<td align="right">0.005397</td>
</tr>
<tr class="odd">
<td align="left">KEGG_TYROSINE_METABOLISM</td>
<td align="right">29</td>
<td align="right">2</td>
<td align="left">Up</td>
<td align="right">0.0004776</td>
<td align="right">0.005698</td>
</tr>
<tr class="even">
<td align="left">KEGG_METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450</td>
<td align="right">26</td>
<td align="right">3</td>
<td align="left">Up</td>
<td align="right">0.0004901</td>
<td align="right">0.005698</td>
</tr>
<tr class="odd">
<td align="left">KEGG_DRUG_METABOLISM_CYTOCHROME_P450</td>
<td align="right">26</td>
<td align="right">3</td>
<td align="left">Up</td>
<td align="right">0.0005606</td>
<td align="right">0.005827</td>
</tr>
<tr class="even">
<td align="left">KEGG_PYRUVATE_METABOLISM</td>
<td align="right">27</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.0006084</td>
<td align="right">0.005827</td>
</tr>
<tr class="odd">
<td align="left">KEGG_CYSTEINE_AND_METHIONINE_METABOLISM</td>
<td align="right">26</td>
<td align="right">4</td>
<td align="left">Up</td>
<td align="right">0.0006305</td>
<td align="right">0.005827</td>
</tr>
<tr class="even">
<td align="left">KEGG_BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS</td>
<td align="right">21</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.0006464</td>
<td align="right">0.005827</td>
</tr>
<tr class="odd">
<td align="left">KEGG_GLYOXYLATE_AND_DICARBOXYLATE_METABOLISM</td>
<td align="right">13</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.0006579</td>
<td align="right">0.005827</td>
</tr>
<tr class="even">
<td align="left">KEGG_SELENOAMINO_ACID_METABOLISM</td>
<td align="right">22</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.0007171</td>
<td align="right">0.005874</td>
</tr>
<tr class="odd">
<td align="left">KEGG_EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION</td>
<td align="right">60</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.0007358</td>
<td align="right">0.005874</td>
</tr>
<tr class="even">
<td align="left">KEGG_VIBRIO_CHOLERAE_INFECTION</td>
<td align="right">47</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.000758</td>
<td align="right">0.005874</td>
</tr>
<tr class="odd">
<td align="left">KEGG_LYSINE_DEGRADATION</td>
<td align="right">39</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.0009213</td>
<td align="right">0.006855</td>
</tr>
<tr class="even">
<td align="left">KEGG_PHENYLALANINE_METABOLISM</td>
<td align="right">11</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.0009648</td>
<td align="right">0.006866</td>
</tr>
<tr class="odd">
<td align="left">KEGG_RENIN_ANGIOTENSIN_SYSTEM</td>
<td align="right">8</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.001033</td>
<td align="right">0.006866</td>
</tr>
<tr class="even">
<td align="left">KEGG_NITROGEN_METABOLISM</td>
<td align="right">13</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.001034</td>
<td align="right">0.006866</td>
</tr>
<tr class="odd">
<td align="left">KEGG_COMPLEMENT_AND_COAGULATION_CASCADES</td>
<td align="right">27</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.00112</td>
<td align="right">0.007181</td>
</tr>
<tr class="even">
<td align="left">KEGG_RIBOSOME</td>
<td align="right">80</td>
<td align="right">26</td>
<td align="left">Up</td>
<td align="right">0.001277</td>
<td align="right">0.007917</td>
</tr>
<tr class="odd">
<td align="left">KEGG_PEROXISOME</td>
<td align="right">70</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.00135</td>
<td align="right">0.008102</td>
</tr>
<tr class="even">
<td align="left">KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS</td>
<td align="right">10</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.00144</td>
<td align="right">0.008369</td>
</tr>
<tr class="odd">
<td align="left">KEGG_PATHOGENIC_ESCHERICHIA_COLI_INFECTION</td>
<td align="right">44</td>
<td align="right">3</td>
<td align="left">Up</td>
<td align="right">0.001755</td>
<td align="right">0.009894</td>
</tr>
<tr class="even">
<td align="left">KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION</td>
<td align="right">40</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.001835</td>
<td align="right">0.01004</td>
</tr>
<tr class="odd">
<td align="left">KEGG_RETINOL_METABOLISM</td>
<td align="right">25</td>
<td align="right">3</td>
<td align="left">Up</td>
<td align="right">0.001902</td>
<td align="right">0.01011</td>
</tr>
<tr class="even">
<td align="left">KEGG_PURINE_METABOLISM</td>
<td align="right">135</td>
<td align="right">4</td>
<td align="left">Up</td>
<td align="right">0.001958</td>
<td align="right">0.01011</td>
</tr>
<tr class="odd">
<td align="left">KEGG_LEUKOCYTE_TRANSENDOTHELIAL_MIGRATION</td>
<td align="right">87</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.002044</td>
<td align="right">0.01018</td>
</tr>
<tr class="even">
<td align="left">KEGG_ALZHEIMERS_DISEASE</td>
<td align="right">141</td>
<td align="right">12</td>
<td align="left">Up</td>
<td align="right">0.002128</td>
<td align="right">0.01018</td>
</tr>
<tr class="odd">
<td align="left">KEGG_CITRATE_CYCLE_TCA_CYCLE</td>
<td align="right">26</td>
<td align="right">2</td>
<td align="left">Up</td>
<td align="right">0.002141</td>
<td align="right">0.01018</td>
</tr>
<tr class="even">
<td align="left">KEGG_N_GLYCAN_BIOSYNTHESIS</td>
<td align="right">45</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.00219</td>
<td align="right">0.01018</td>
</tr>
<tr class="odd">
<td align="left">KEGG_OXIDATIVE_PHOSPHORYLATION</td>
<td align="right">119</td>
<td align="right">14</td>
<td align="left">Up</td>
<td align="right">0.002373</td>
<td align="right">0.01065</td>
</tr>
<tr class="even">
<td align="left">KEGG_PROPANOATE_METABOLISM</td>
<td align="right">28</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.002406</td>
<td align="right">0.01065</td>
</tr>
<tr class="odd">
<td align="left">KEGG_PROXIMAL_TUBULE_BICARBONATE_RECLAMATION</td>
<td align="right">15</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.002581</td>
<td align="right">0.01116</td>
</tr>
<tr class="even">
<td align="left">KEGG_PROTEASOME</td>
<td align="right">42</td>
<td align="right">2</td>
<td align="left">Up</td>
<td align="right">0.00284</td>
<td align="right">0.01187</td>
</tr>
<tr class="odd">
<td align="left">KEGG_GLYCOLYSIS_GLUCONEOGENESIS</td>
<td align="right">39</td>
<td align="right">3</td>
<td align="left">Up</td>
<td align="right">0.002871</td>
<td align="right">0.01187</td>
</tr>
<tr class="even">
<td align="left">KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY</td>
<td align="right">64</td>
<td align="right">3</td>
<td align="left">Down</td>
<td align="right">0.003085</td>
<td align="right">0.01247</td>
</tr>
<tr class="odd">
<td align="left">KEGG_GLYCEROPHOSPHOLIPID_METABOLISM</td>
<td align="right">56</td>
<td align="right">1</td>
<td align="left">Up</td>
<td align="right">0.004306</td>
<td align="right">0.01704</td>
</tr>
<tr class="even">
<td align="left">KEGG_ABC_TRANSPORTERS</td>
<td align="right">32</td>
<td align="right">0</td>
<td align="left">Up</td>
<td align="right">0.004993</td>
<td align="right">0.01935</td>
</tr>
</tbody>
</table>
<div id="exploration-of-significant-kegg-gene-sets" class="section level3">
<h3>Exploration of Significant KEGG Gene Sets</h3>
<pre class="r"><code>kgFry %&gt;% 
  left_join(gsSizes) %&gt;%
  dplyr::filter(de_size &gt;= 4, FDR &lt; 0.05) %&gt;%
  dplyr::select(gs_name, de_id) %&gt;%
  unnest(de_id) %&gt;% 
  mutate(
    gs_name = str_remove(gs_name, &quot;KEGG_&quot;),
    gs_name = fct_lump(gs_name, n = 14)
    ) %&gt;% 
  split(f = .$gs_name) %&gt;% 
  lapply(magrittr::extract2, &quot;de_id&quot;) %&gt;% 
  fromList() %&gt;% 
  upset(
    nsets = length(.), 
    nintersects = 20,
    order.by = &quot;freq&quot;, 
    mb.ratio = c(0.6, 0.4),
    sets.x.label = &quot;Number Of DE Genes&quot;
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/kgFryUpset-1.png" alt="*UpSet plot indicating distribution of DE genes within all significant terms from the KEGG gene sets. Gene sets were restricted to those with an FDR &lt; 0.05 and four or more DE genes. There is considerable overlap between DE genes in Oxidative Phosphorylation and both Parkinson's and Huntington's Disease, indicating that these terms essentially capture the same expression signal within this dataset.*" width="768" />
<p class="caption">
<em>UpSet plot indicating distribution of DE genes within all significant terms from the KEGG gene sets. Gene sets were restricted to those with an FDR &lt; 0.05 and four or more DE genes. There is considerable overlap between DE genes in Oxidative Phosphorylation and both Parkinson’s and Huntington’s Disease, indicating that these terms essentially capture the same expression signal within this dataset.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-kgFryUpset-1">
Past versions of kgFryUpset-1.png
</button>
</p>
<div id="fig-kgFryUpset-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/a5cdf5d6804cdddeb202915519f7e0588ea4648a/docs/figure/3_Enrichment_MutantVsWT.Rmd/kgFryUpset-1.png" target="_blank">a5cdf5d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>riboOx &lt;- kg %&gt;%
  dplyr::filter(
    grepl(&quot;OXIDATIVE_PHOS&quot;, gs_name) | grepl(&quot;RIBOSOME&quot;, gs_name),
    gene_id %in% dplyr::filter(deTable, DE)$gene_id
    ) %&gt;%
  mutate(gs_name = str_remove(gs_name, &quot;(KEGG|HALLMARK)_&quot;)) %&gt;%
  distinct(gs_name, gene_id) %&gt;%
  left_join(
    cpmPostNorm %&gt;% 
      as.data.frame() %&gt;% 
      rownames_to_column(&quot;gene_id&quot;) %&gt;%
      as_tibble()
    ) %&gt;%
  pivot_longer(cols = starts_with(&quot;Ps&quot;), names_to = &quot;sample&quot;, values_to = &quot;CPM&quot;) %&gt;%
  left_join(dgeList$samples) %&gt;% 
  left_join(dgeList$genes) %&gt;%
  dplyr::select(gs_name, gene_name, sampleID, genotype, CPM) %&gt;%
  pivot_wider(
    id_cols = c(gs_name, gene_name),
    values_from = CPM,
    names_from = sampleID
  ) 
riboOx %&gt;% 
  dplyr::select(-gs_name) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(&quot;gene_name&quot;) %&gt;%
  as.matrix() %&gt;%
  pheatmap(
    annotation_row = data.frame(
      GeneSet = riboOx$gs_name %&gt;% str_replace(&quot;OXI.+&quot;, &quot;OXPHOS&quot;), 
      row.names = riboOx$gene_name
      ),
    color = viridis_pal(option = &quot;magma&quot;)(100),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), &quot;logCPM\n&quot;),
    annotation_names_row = FALSE,
    cutree_rows = 3,
    cutree_cols = 2
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/riboOxHeatmap-1.png" alt="*Given that the largest gene sets within the KEGG results were Oxidative Phosphorylation and Ribosomal gene sets, all DE genes associated with these KEGG terms are displayed. All genes appear to show increased expression in mutant samples.*" width="768" />
<p class="caption">
<em>Given that the largest gene sets within the KEGG results were Oxidative Phosphorylation and Ribosomal gene sets, all DE genes associated with these KEGG terms are displayed. All genes appear to show increased expression in mutant samples.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-riboOxHeatmap-1">
Past versions of riboOxHeatmap-1.png
</button>
</p>
<div id="fig-riboOxHeatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/7fe1505995bf93217d6ecd0c1de1f60f27e27c26/docs/figure/3_Enrichment_MutantVsWT.Rmd/riboOxHeatmap-1.png" target="_blank">7fe1505</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/86ffa07adc34510ac982dcb32159b051f50a7b9a/docs/figure/3_Enrichment_MutantVsWT.Rmd/riboOxHeatmap-1.png" target="_blank">86ffa07</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="go-gene-sets-1" class="section level2">
<h2>GO Gene Sets</h2>
<pre class="r"><code>goFry &lt;- cpmPostNorm %&gt;%
  fry(
    index = goByID,
    design = dgeList$design,
    contrast = &quot;mutant&quot;,
    sort = &quot;directional&quot;
    ) %&gt;%
  rownames_to_column(&quot;gs_name&quot;) %&gt;%
  as_tibble()</code></pre>
<pre class="r"><code>goFry %&gt;%
  left_join(gsSizes) %&gt;%
  left_join(distinct(go, gs_name, shortest_path)) %&gt;%
  dplyr::filter(
    FDR &lt; 0.01, 
    shortest_path &gt; 3, 
    NGenes &lt; 1000,
    de_size &gt;= 5
  ) %&gt;%
  dplyr::select(
    `GO Gene Set` = gs_name,
    `Expressed Genes` = NGenes,
    `DE Genes` = de_size,
    Direction,
    PValue, FDR
    ) %&gt;%
  pander(
    justify = &quot;lrrlrr&quot;,
    caption = &quot;Enriched GO Gene Sets, using the thresholds FDR &lt; 0.01, at least 5 DE genes, a set size &lt; 1000 and more than 3 steps back to the ontology root.&quot;
  )</code></pre>
<table>
<caption>Enriched GO Gene Sets, using the thresholds FDR &lt; 0.01, at least 5 DE genes, a set size &lt; 1000 and more than 3 steps back to the ontology root.</caption>
<colgroup>
<col width="53%" />
<col width="13%" />
<col width="7%" />
<col width="8%" />
<col width="8%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">GO Gene Set</th>
<th align="right">Expressed Genes</th>
<th align="right">DE Genes</th>
<th align="left">Direction</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_PEPTIDYL_LYSINE_TRIMETHYLATION</td>
<td align="right">42</td>
<td align="right">5</td>
<td align="left">Down</td>
<td align="right">6.682e-06</td>
<td align="right">0.001904</td>
</tr>
<tr class="even">
<td align="left">GO_FATTY_ACID_BIOSYNTHETIC_PROCESS</td>
<td align="right">96</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">9.201e-06</td>
<td align="right">0.002258</td>
</tr>
<tr class="odd">
<td align="left">GO_ORGANIC_ACID_TRANSPORT</td>
<td align="right">240</td>
<td align="right">14</td>
<td align="left">Up</td>
<td align="right">3.617e-05</td>
<td align="right">0.003991</td>
</tr>
<tr class="even">
<td align="left">GO_ORGANIC_ACID_BIOSYNTHETIC_PROCESS</td>
<td align="right">198</td>
<td align="right">12</td>
<td align="left">Up</td>
<td align="right">4.586e-05</td>
<td align="right">0.004057</td>
</tr>
<tr class="odd">
<td align="left">GO_PEPTIDYL_LYSINE_METHYLATION</td>
<td align="right">111</td>
<td align="right">5</td>
<td align="left">Down</td>
<td align="right">4.848e-05</td>
<td align="right">0.004057</td>
</tr>
<tr class="even">
<td align="left">GO_RESPONSE_TO_CORTICOSTEROID</td>
<td align="right">121</td>
<td align="right">5</td>
<td align="left">Up</td>
<td align="right">4.854e-05</td>
<td align="right">0.004057</td>
</tr>
<tr class="odd">
<td align="left">GO_PEPTIDYL_LYSINE_ACETYLATION</td>
<td align="right">144</td>
<td align="right">5</td>
<td align="left">Down</td>
<td align="right">6.245e-05</td>
<td align="right">0.004211</td>
</tr>
<tr class="even">
<td align="left">GO_MONOCARBOXYLIC_ACID_BIOSYNTHETIC_PROCESS</td>
<td align="right">127</td>
<td align="right">7</td>
<td align="left">Up</td>
<td align="right">7.071e-05</td>
<td align="right">0.004534</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_LIPID_BIOSYNTHETIC_PROCESS</td>
<td align="right">138</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">7.186e-05</td>
<td align="right">0.004534</td>
</tr>
<tr class="even">
<td align="left">GO_RESPONSE_TO_ALCOHOL</td>
<td align="right">177</td>
<td align="right">7</td>
<td align="left">Up</td>
<td align="right">7.293e-05</td>
<td align="right">0.004534</td>
</tr>
<tr class="odd">
<td align="left">GO_HISTONE_METHYLATION</td>
<td align="right">118</td>
<td align="right">5</td>
<td align="left">Down</td>
<td align="right">7.92e-05</td>
<td align="right">0.004536</td>
</tr>
<tr class="even">
<td align="left">GO_MONOCARBOXYLIC_ACID_METABOLIC_PROCESS</td>
<td align="right">315</td>
<td align="right">10</td>
<td align="left">Up</td>
<td align="right">9.221e-05</td>
<td align="right">0.004724</td>
</tr>
<tr class="odd">
<td align="left">GO_NUCLEAR_BODY</td>
<td align="right">649</td>
<td align="right">21</td>
<td align="left">Down</td>
<td align="right">0.0001011</td>
<td align="right">0.004828</td>
</tr>
<tr class="even">
<td align="left">GO_MONOCARBOXYLIC_ACID_TRANSPORT</td>
<td align="right">110</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">0.0001017</td>
<td align="right">0.004829</td>
</tr>
<tr class="odd">
<td align="left">GO_CARBOHYDRATE_BIOSYNTHETIC_PROCESS</td>
<td align="right">148</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">0.0001221</td>
<td align="right">0.005132</td>
</tr>
<tr class="even">
<td align="left">GO_SYMPORTER_ACTIVITY</td>
<td align="right">94</td>
<td align="right">10</td>
<td align="left">Up</td>
<td align="right">0.0001224</td>
<td align="right">0.005132</td>
</tr>
<tr class="odd">
<td align="left">GO_LIPID_BIOSYNTHETIC_PROCESS</td>
<td align="right">508</td>
<td align="right">13</td>
<td align="left">Up</td>
<td align="right">0.0001267</td>
<td align="right">0.005207</td>
</tr>
<tr class="even">
<td align="left">GO_DRUG_TRANSPORT</td>
<td align="right">152</td>
<td align="right">8</td>
<td align="left">Up</td>
<td align="right">0.0001461</td>
<td align="right">0.005379</td>
</tr>
<tr class="odd">
<td align="left">GO_AMINO_ACID_TRANSMEMBRANE_TRANSPORT</td>
<td align="right">69</td>
<td align="right">5</td>
<td align="left">Up</td>
<td align="right">0.0001709</td>
<td align="right">0.005806</td>
</tr>
<tr class="even">
<td align="left">GO_ORGANIC_ACID_TRANSMEMBRANE_TRANSPORT</td>
<td align="right">106</td>
<td align="right">9</td>
<td align="left">Up</td>
<td align="right">0.0001766</td>
<td align="right">0.005887</td>
</tr>
<tr class="odd">
<td align="left">GO_ORGANIC_ANION_TRANSPORT</td>
<td align="right">345</td>
<td align="right">16</td>
<td align="left">Up</td>
<td align="right">0.0001785</td>
<td align="right">0.005928</td>
</tr>
<tr class="even">
<td align="left">GO_SECONDARY_ACTIVE_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">154</td>
<td align="right">13</td>
<td align="left">Up</td>
<td align="right">0.0001819</td>
<td align="right">0.005967</td>
</tr>
<tr class="odd">
<td align="left">GO_COENZYME_METABOLIC_PROCESS</td>
<td align="right">195</td>
<td align="right">5</td>
<td align="left">Up</td>
<td align="right">0.000182</td>
<td align="right">0.005967</td>
</tr>
<tr class="even">
<td align="left">GO_ANION_TRANSMEMBRANE_TRANSPORT</td>
<td align="right">203</td>
<td align="right">15</td>
<td align="left">Up</td>
<td align="right">0.0001854</td>
<td align="right">0.005999</td>
</tr>
<tr class="odd">
<td align="left">GO_ANION_TRANSPORT</td>
<td align="right">434</td>
<td align="right">22</td>
<td align="left">Up</td>
<td align="right">0.0001883</td>
<td align="right">0.006049</td>
</tr>
<tr class="even">
<td align="left">GO_SEQUENCE_SPECIFIC_DNA_BINDING</td>
<td align="right">827</td>
<td align="right">29</td>
<td align="left">Down</td>
<td align="right">0.0001932</td>
<td align="right">0.006093</td>
</tr>
<tr class="odd">
<td align="left">GO_POSITIVE_REGULATION_OF_TRANSCRIPTION_BY_RNA_POLYMERASE_II</td>
<td align="right">903</td>
<td align="right">31</td>
<td align="left">Down</td>
<td align="right">0.000212</td>
<td align="right">0.006139</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATORY_REGION_NUCLEIC_ACID_BINDING</td>
<td align="right">720</td>
<td align="right">24</td>
<td align="left">Down</td>
<td align="right">0.0002154</td>
<td align="right">0.006157</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEIN_LOCALIZATION_TO_MEMBRANE</td>
<td align="right">539</td>
<td align="right">35</td>
<td align="left">Up</td>
<td align="right">0.0002254</td>
<td align="right">0.006217</td>
</tr>
<tr class="even">
<td align="left">GO_PROTEIN_METHYLATION</td>
<td align="right">153</td>
<td align="right">8</td>
<td align="left">Down</td>
<td align="right">0.0002353</td>
<td align="right">0.006313</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_PHOSPHATASE_ACTIVITY</td>
<td align="right">149</td>
<td align="right">8</td>
<td align="left">Down</td>
<td align="right">0.0002413</td>
<td align="right">0.006313</td>
</tr>
<tr class="even">
<td align="left">GO_ACTIVATING_TRANSCRIPTION_FACTOR_BINDING</td>
<td align="right">70</td>
<td align="right">5</td>
<td align="left">Down</td>
<td align="right">0.0002415</td>
<td align="right">0.006313</td>
</tr>
<tr class="odd">
<td align="left">GO_ORGANIC_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">144</td>
<td align="right">11</td>
<td align="left">Up</td>
<td align="right">0.000253</td>
<td align="right">0.006397</td>
</tr>
<tr class="even">
<td align="left">GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_MEMBRANE</td>
<td align="right">287</td>
<td align="right">30</td>
<td align="left">Up</td>
<td align="right">0.0002603</td>
<td align="right">0.006429</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEIN_ACETYLATION</td>
<td align="right">168</td>
<td align="right">5</td>
<td align="left">Down</td>
<td align="right">0.0002605</td>
<td align="right">0.006429</td>
</tr>
<tr class="even">
<td align="left">GO_ISOPRENOID_METABOLIC_PROCESS</td>
<td align="right">68</td>
<td align="right">5</td>
<td align="left">Up</td>
<td align="right">0.0002633</td>
<td align="right">0.006433</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEIN_MATURATION</td>
<td align="right">210</td>
<td align="right">8</td>
<td align="left">Up</td>
<td align="right">0.0002641</td>
<td align="right">0.006433</td>
</tr>
<tr class="even">
<td align="left">GO_DRUG_TRANSMEMBRANE_TRANSPORT</td>
<td align="right">65</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">0.0002665</td>
<td align="right">0.006433</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_PEPTIDE_HORMONE_SECRETION</td>
<td align="right">157</td>
<td align="right">9</td>
<td align="left">Up</td>
<td align="right">0.0002996</td>
<td align="right">0.00691</td>
</tr>
<tr class="even">
<td align="left">GO_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">228</td>
<td align="right">17</td>
<td align="left">Up</td>
<td align="right">0.0003288</td>
<td align="right">0.007143</td>
</tr>
<tr class="odd">
<td align="left">GO_NUCLEAR_SPECK</td>
<td align="right">346</td>
<td align="right">11</td>
<td align="left">Down</td>
<td align="right">0.0003295</td>
<td align="right">0.007143</td>
</tr>
<tr class="even">
<td align="left">GO_PROXIMAL_PROMOTER_SEQUENCE_SPECIFIC_DNA_BINDING</td>
<td align="right">413</td>
<td align="right">15</td>
<td align="left">Down</td>
<td align="right">0.0003484</td>
<td align="right">0.007241</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_DEPHOSPHORYLATION</td>
<td align="right">180</td>
<td align="right">8</td>
<td align="left">Down</td>
<td align="right">0.0003661</td>
<td align="right">0.007384</td>
</tr>
<tr class="even">
<td align="left">GO_COVALENT_CHROMATIN_MODIFICATION</td>
<td align="right">384</td>
<td align="right">13</td>
<td align="left">Down</td>
<td align="right">0.0003797</td>
<td align="right">0.007483</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_CHROMATIN_ORGANIZATION</td>
<td align="right">158</td>
<td align="right">7</td>
<td align="left">Down</td>
<td align="right">0.0003848</td>
<td align="right">0.007483</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_INSULIN_SECRETION</td>
<td align="right">140</td>
<td align="right">9</td>
<td align="left">Up</td>
<td align="right">0.0004112</td>
<td align="right">0.007729</td>
</tr>
<tr class="odd">
<td align="left">GO_ROUGH_ENDOPLASMIC_RETICULUM</td>
<td align="right">67</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">0.0004221</td>
<td align="right">0.00781</td>
</tr>
<tr class="even">
<td align="left">GO_AMINO_ACID_TRANSPORT</td>
<td align="right">119</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">0.0004254</td>
<td align="right">0.00781</td>
</tr>
<tr class="odd">
<td align="left">GO_SEQUENCE_SPECIFIC_DOUBLE_STRANDED_DNA_BINDING</td>
<td align="right">668</td>
<td align="right">23</td>
<td align="left">Down</td>
<td align="right">0.0004287</td>
<td align="right">0.007824</td>
</tr>
<tr class="even">
<td align="left">GO_DOUBLE_STRANDED_DNA_BINDING</td>
<td align="right">740</td>
<td align="right">30</td>
<td align="left">Down</td>
<td align="right">0.0004372</td>
<td align="right">0.007859</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_MRNA_PROCESSING</td>
<td align="right">117</td>
<td align="right">5</td>
<td align="left">Down</td>
<td align="right">0.0004506</td>
<td align="right">0.00795</td>
</tr>
<tr class="even">
<td align="left">GO_NUCLEOSIDE_PHOSPHATE_BIOSYNTHETIC_PROCESS</td>
<td align="right">227</td>
<td align="right">10</td>
<td align="left">Up</td>
<td align="right">0.0004663</td>
<td align="right">0.008076</td>
</tr>
<tr class="odd">
<td align="left">GO_NUCLEOPLASM_PART</td>
<td align="right">943</td>
<td align="right">31</td>
<td align="left">Down</td>
<td align="right">0.0004697</td>
<td align="right">0.008088</td>
</tr>
<tr class="even">
<td align="left">GO_SECRETORY_VESICLE</td>
<td align="right">713</td>
<td align="right">24</td>
<td align="left">Up</td>
<td align="right">0.0004815</td>
<td align="right">0.008218</td>
</tr>
<tr class="odd">
<td align="left">GO_AMINO_ACID_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">60</td>
<td align="right">5</td>
<td align="left">Up</td>
<td align="right">0.000498</td>
<td align="right">0.008306</td>
</tr>
<tr class="even">
<td align="left">GO_PURINE_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS</td>
<td align="right">170</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">0.0004995</td>
<td align="right">0.008306</td>
</tr>
<tr class="odd">
<td align="left">GO_LIPID_CATABOLIC_PROCESS</td>
<td align="right">221</td>
<td align="right">7</td>
<td align="left">Up</td>
<td align="right">0.0005363</td>
<td align="right">0.008614</td>
</tr>
<tr class="even">
<td align="left">GO_INORGANIC_ANION_TRANSPORT</td>
<td align="right">117</td>
<td align="right">7</td>
<td align="left">Up</td>
<td align="right">0.0005584</td>
<td align="right">0.008701</td>
</tr>
<tr class="odd">
<td align="left">GO_MONOVALENT_INORGANIC_CATION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">268</td>
<td align="right">13</td>
<td align="left">Up</td>
<td align="right">0.0005832</td>
<td align="right">0.00893</td>
</tr>
<tr class="even">
<td align="left">GO_PEPTIDE_HORMONE_SECRETION</td>
<td align="right">194</td>
<td align="right">9</td>
<td align="left">Up</td>
<td align="right">0.0005972</td>
<td align="right">0.009027</td>
</tr>
<tr class="odd">
<td align="left">GO_ALPHA_AMINO_ACID_METABOLIC_PROCESS</td>
<td align="right">126</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">0.0006074</td>
<td align="right">0.009142</td>
</tr>
<tr class="even">
<td align="left">GO_ORGANIC_ACID_CATABOLIC_PROCESS</td>
<td align="right">184</td>
<td align="right">5</td>
<td align="left">Up</td>
<td align="right">0.0006398</td>
<td align="right">0.009283</td>
</tr>
<tr class="odd">
<td align="left">GO_IMPORT_INTO_CELL</td>
<td align="right">532</td>
<td align="right">7</td>
<td align="left">Up</td>
<td align="right">0.0006709</td>
<td align="right">0.009508</td>
</tr>
<tr class="even">
<td align="left">GO_RIBOSE_PHOSPHATE_BIOSYNTHETIC_PROCESS</td>
<td align="right">165</td>
<td align="right">7</td>
<td align="left">Up</td>
<td align="right">0.0006731</td>
<td align="right">0.009508</td>
</tr>
<tr class="odd">
<td align="left">GO_TUBULIN_BINDING</td>
<td align="right">258</td>
<td align="right">5</td>
<td align="left">Down</td>
<td align="right">0.0006736</td>
<td align="right">0.009508</td>
</tr>
<tr class="even">
<td align="left">GO_CATION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">458</td>
<td align="right">18</td>
<td align="left">Up</td>
<td align="right">0.0006766</td>
<td align="right">0.009508</td>
</tr>
<tr class="odd">
<td align="left">GO_MRNA_METABOLIC_PROCESS</td>
<td align="right">453</td>
<td align="right">26</td>
<td align="left">Down</td>
<td align="right">0.0006959</td>
<td align="right">0.009529</td>
</tr>
<tr class="even">
<td align="left">GO_GUANYL_NUCLEOTIDE_BINDING</td>
<td align="right">317</td>
<td align="right">6</td>
<td align="left">Up</td>
<td align="right">0.0007188</td>
<td align="right">0.009678</td>
</tr>
<tr class="odd">
<td align="left">GO_ORGANIC_HYDROXY_COMPOUND_BIOSYNTHETIC_PROCESS</td>
<td align="right">183</td>
<td align="right">8</td>
<td align="left">Up</td>
<td align="right">0.000736</td>
<td align="right">0.00984</td>
</tr>
</tbody>
</table>
<div id="exploration-of-significant-go-gene-sets" class="section level3">
<h3>Exploration of Significant GO Gene Sets</h3>
<pre class="r"><code>goFry %&gt;% 
  left_join(gsSizes) %&gt;%
  left_join(distinct(go, gs_name, shortest_path)) %&gt;%
  mutate(
    propDE = de_size / gs_size
  ) %&gt;%
  dplyr::filter(
    de_size &gt;= 15, 
    propDE &gt; 0.05,
    FDR &lt; 0.02,
    shortest_path &gt; 3
  ) %&gt;%
  dplyr::select(gs_name, de_id) %&gt;%
  unnest(de_id) %&gt;% 
  mutate(
    gs_name = str_remove(gs_name, &quot;GO_&quot;),
    gs_name = fct_lump(gs_name, n = 20)
    ) %&gt;% 
  split(f = .$gs_name) %&gt;% 
  lapply(magrittr::extract2, &quot;de_id&quot;) %&gt;% 
  fromList() %&gt;% 
  upset(
    nsets = length(.), 
    nintersects = 20,
    order.by = &quot;freq&quot;, 
    mb.ratio = c(0.6, 0.4),
    sets.x.label = &quot;Number Of DE Genes&quot;
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/goFryUpset-1.png" alt="*UpSet plot indicating distribution of DE genes within significantly enriched terms from the GO gene sets. For this visualisation, GO terms were restricted to those with 15 or more DE genes, where this represented more than 5% of a gene set as being DE, along with an FDR &lt; 0.02 and more than 3 steps back to the ontology root. The 20 largest GO terms satisfying these criteria are shown. The plot is  truncated at the right hand side for simplicity. A group of 28 genes is uniquely attributed to the Mitochondrial Envelope, with a further 18 being relatively unique to mRNA Metabolic Process. The next grouping of 15 genes are unique to Regulation Of Nucleobase-Containing Compound Metabolic Process followed by 25 genes, spread across two clusters of terms which largely represent Ribosomal activity. In between these are 13 genes uniquely associated with Anion Transport.*" width="960" />
<p class="caption">
<em>UpSet plot indicating distribution of DE genes within significantly enriched terms from the GO gene sets. For this visualisation, GO terms were restricted to those with 15 or more DE genes, where this represented more than 5% of a gene set as being DE, along with an FDR &lt; 0.02 and more than 3 steps back to the ontology root. The 20 largest GO terms satisfying these criteria are shown. The plot is truncated at the right hand side for simplicity. A group of 28 genes is uniquely attributed to the Mitochondrial Envelope, with a further 18 being relatively unique to mRNA Metabolic Process. The next grouping of 15 genes are unique to Regulation Of Nucleobase-Containing Compound Metabolic Process followed by 25 genes, spread across two clusters of terms which largely represent Ribosomal activity. In between these are 13 genes uniquely associated with Anion Transport.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-goFryUpset-1">
Past versions of goFryUpset-1.png
</button>
</p>
<div id="fig-goFryUpset-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/a5cdf5d6804cdddeb202915519f7e0588ea4648a/docs/figure/3_Enrichment_MutantVsWT.Rmd/goFryUpset-1.png" target="_blank">a5cdf5d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="genes-associated-with-the-mitochondrial-envelope" class="section level3">
<h3>Genes Associated with the Mitochondrial Envelope</h3>
<pre class="r"><code>go %&gt;%
  dplyr::filter(
    gs_name == &quot;GO_MITOCHONDRIAL_ENVELOPE&quot;,
    gene_id %in% dplyr::filter(deTable, DE)$gene_id
  ) %&gt;%
  dplyr::select(
    gene_id, gene_name
  ) %&gt;%
  left_join(
    cpmPostNorm %&gt;%
      as.data.frame() %&gt;%
      rownames_to_column(&quot;gene_id&quot;)
  ) %&gt;%
  dplyr::select(gene_name, starts_with(&quot;Ps&quot;)) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(&quot;gene_name&quot;) %&gt;%
  as.matrix() %&gt;%
  pheatmap(
    color = viridis_pal(option = &quot;magma&quot;)(100),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), &quot;logCPM\n&quot;),
    labels_col = hmHeat %&gt;%
      dplyr::select(starts_with(&quot;Ps&quot;)) %&gt;% 
      colnames() %&gt;% 
      enframe(name = NULL) %&gt;% 
      dplyr::rename(sample = value) %&gt;%
      left_join(dgeList$samples) %&gt;%
      dplyr::select(sample, sampleID) %&gt;%
      with(
        structure(sampleID, names = sample)
      ),
    annotation_col = dgeList$samples %&gt;%
      dplyr::select(Genotype = genotype),
    cutree_rows = 4,
    cutree_cols = 2
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/heatMito-1.png" alt="*All DE genes associated with the GO term 'Mitochondrial Envelope'.*" width="768" />
<p class="caption">
<em>All DE genes associated with the GO term ‘Mitochondrial Envelope’.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-heatMito-1">
Past versions of heatMito-1.png
</button>
</p>
<div id="fig-heatMito-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/a5cdf5d6804cdddeb202915519f7e0588ea4648a/docs/figure/3_Enrichment_MutantVsWT.Rmd/heatMito-1.png" target="_blank">a5cdf5d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="genes-associated-with-rna-metabolic-process" class="section level3">
<h3>Genes Associated with RNA Metabolic Process</h3>
<pre class="r"><code>go %&gt;%
  dplyr::filter(
    gs_name == &quot;GO_MRNA_METABOLIC_PROCESS&quot;,
    gene_id %in% dplyr::filter(deTable, DE)$gene_id
  ) %&gt;%
  dplyr::select(
    gene_id, gene_name
  ) %&gt;%
  left_join(
    cpmPostNorm %&gt;%
      as.data.frame() %&gt;%
      rownames_to_column(&quot;gene_id&quot;)
  ) %&gt;%
  dplyr::select(gene_name, starts_with(&quot;Ps&quot;)) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(&quot;gene_name&quot;) %&gt;%
  as.matrix() %&gt;%
  pheatmap(
    color = viridis_pal(option = &quot;magma&quot;)(100),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), &quot;logCPM\n&quot;),
    labels_col = hmHeat %&gt;%
      dplyr::select(starts_with(&quot;Ps&quot;)) %&gt;% 
      colnames() %&gt;% 
      enframe(name = NULL) %&gt;% 
      dplyr::rename(sample = value) %&gt;%
      left_join(dgeList$samples) %&gt;%
      dplyr::select(sample, sampleID) %&gt;%
      with(
        structure(sampleID, names = sample)
      ),
    annotation_col = dgeList$samples %&gt;%
      dplyr::select(Genotype = genotype),
    cutree_rows = 4,
    cutree_cols = 2
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/heatRnaMetab-1.png" alt="*All DE genes associated with the GO term 'mRNA Metabolic Process'.*" width="768" />
<p class="caption">
<em>All DE genes associated with the GO term ‘mRNA Metabolic Process’.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-heatRnaMetab-1">
Past versions of heatRnaMetab-1.png
</button>
</p>
<div id="fig-heatRnaMetab-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/a5cdf5d6804cdddeb202915519f7e0588ea4648a/docs/figure/3_Enrichment_MutantVsWT.Rmd/heatRnaMetab-1.png" target="_blank">a5cdf5d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="genes-associated-with-regulation-of-nucleobase-containing-compound-metabolic-process" class="section level3">
<h3>Genes Associated with Regulation Of Nucleobase-Containing Compound Metabolic Process</h3>
<pre class="r"><code>go %&gt;%
  dplyr::filter(
    gs_name == &quot;GO_REGULATION_OF_NUCLEOBASE_CONTAINING_COMPOUND_METABOLIC_PROCESS&quot;,
    gene_id %in% dplyr::filter(deTable, DE)$gene_id
  ) %&gt;%
  dplyr::select(
    gene_id, gene_name
  ) %&gt;%
  left_join(
    cpmPostNorm %&gt;%
      as.data.frame() %&gt;%
      rownames_to_column(&quot;gene_id&quot;)
  ) %&gt;%
  dplyr::select(gene_name, starts_with(&quot;Ps&quot;)) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(&quot;gene_name&quot;) %&gt;%
  as.matrix() %&gt;%
  pheatmap(
    color = viridis_pal(option = &quot;magma&quot;)(100),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), &quot;logCPM\n&quot;),
    labels_col = hmHeat %&gt;%
      dplyr::select(starts_with(&quot;Ps&quot;)) %&gt;% 
      colnames() %&gt;% 
      enframe(name = NULL) %&gt;% 
      dplyr::rename(sample = value) %&gt;%
      left_join(dgeList$samples) %&gt;%
      dplyr::select(sample, sampleID) %&gt;%
      with(
        structure(sampleID, names = sample)
      ),
    annotation_col = dgeList$samples %&gt;%
      dplyr::select(Genotype = genotype),
    cutree_rows = 4,
    cutree_cols = 2
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/heatRegNucleo-1.png" alt="*All DE genes associated with the GO term 'mRNA Metabolic Process'.*" width="768" />
<p class="caption">
<em>All DE genes associated with the GO term ‘mRNA Metabolic Process’.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-heatRegNucleo-1">
Past versions of heatRegNucleo-1.png
</button>
</p>
<div id="fig-heatRegNucleo-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/a5cdf5d6804cdddeb202915519f7e0588ea4648a/docs/figure/3_Enrichment_MutantVsWT.Rmd/heatRegNucleo-1.png" target="_blank">a5cdf5d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="genes-associated-with-anion-transport" class="section level3">
<h3>Genes Associated with Anion Transport</h3>
<pre class="r"><code>go %&gt;%
  dplyr::filter(
    gs_name == &quot;GO_ANION_TRANSPORT&quot;,
    gene_id %in% dplyr::filter(deTable, DE)$gene_id
  ) %&gt;%
  dplyr::select(
    gene_id, gene_name
  ) %&gt;%
  left_join(
    cpmPostNorm %&gt;%
      as.data.frame() %&gt;%
      rownames_to_column(&quot;gene_id&quot;)
  ) %&gt;%
  dplyr::select(gene_name, starts_with(&quot;Ps&quot;)) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(&quot;gene_name&quot;) %&gt;%
  as.matrix() %&gt;%
  pheatmap(
    color = viridis_pal(option = &quot;magma&quot;)(100),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), &quot;logCPM\n&quot;),
    labels_col = hmHeat %&gt;%
      dplyr::select(starts_with(&quot;Ps&quot;)) %&gt;% 
      colnames() %&gt;% 
      enframe(name = NULL) %&gt;% 
      dplyr::rename(sample = value) %&gt;%
      left_join(dgeList$samples) %&gt;%
      dplyr::select(sample, sampleID) %&gt;%
      with(
        structure(sampleID, names = sample)
      ),
    annotation_col = dgeList$samples %&gt;%
      dplyr::select(Genotype = genotype),
    cutree_rows = 3,
    cutree_cols = 2
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/heatAnion-1.png" alt="*All DE genes associated with the GO term 'Anion Transport'.*" width="768" />
<p class="caption">
<em>All DE genes associated with the GO term ‘Anion Transport’.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-heatAnion-1">
Past versions of heatAnion-1.png
</button>
</p>
<div id="fig-heatAnion-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/a5cdf5d6804cdddeb202915519f7e0588ea4648a/docs/figure/3_Enrichment_MutantVsWT.Rmd/heatAnion-1.png" target="_blank">a5cdf5d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="genes-associated-with-ribosomal-activity" class="section level3">
<h3>Genes Associated with Ribosomal Activity</h3>
<pre class="r"><code>go %&gt;%
  dplyr::filter(
    gs_name == &quot;GO_CYTOSOLIC_RIBOSOME&quot;,
    gene_id %in% dplyr::filter(deTable, DE)$gene_id
  ) %&gt;%
  dplyr::select(
    gene_id, gene_name
  ) %&gt;%
  left_join(
    cpmPostNorm %&gt;%
      as.data.frame() %&gt;%
      rownames_to_column(&quot;gene_id&quot;)
  ) %&gt;%
  dplyr::select(gene_name, starts_with(&quot;Ps&quot;)) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(&quot;gene_name&quot;) %&gt;%
  as.matrix() %&gt;%
  pheatmap(
    color = viridis_pal(option = &quot;magma&quot;)(100),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), &quot;logCPM\n&quot;),
    labels_col = hmHeat %&gt;%
      dplyr::select(starts_with(&quot;Ps&quot;)) %&gt;% 
      colnames() %&gt;% 
      enframe(name = NULL) %&gt;% 
      dplyr::rename(sample = value) %&gt;%
      left_join(dgeList$samples) %&gt;%
      dplyr::select(sample, sampleID) %&gt;%
      with(
        structure(sampleID, names = sample)
      ),
    annotation_col = dgeList$samples %&gt;%
      dplyr::select(Genotype = genotype),
    cutree_rows = 3,
    cutree_cols = 2
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/heatRibo-1.png" alt="*All DE genes associated with the GO term 'Cytosolic Ribosome'.*" width="768" />
<p class="caption">
<em>All DE genes associated with the GO term ‘Cytosolic Ribosome’.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-heatRibo-1">
Past versions of heatRibo-1.png
</button>
</p>
<div id="fig-heatRibo-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/a5cdf5d6804cdddeb202915519f7e0588ea4648a/docs/figure/3_Enrichment_MutantVsWT.Rmd/heatRibo-1.png" target="_blank">a5cdf5d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="data-export" class="section level1">
<h1>Data Export</h1>
<p>All enriched gene sets with an FDR adjusted p-value &lt; 0.05 were exported as a single csv file.</p>
<pre class="r"><code>bind_rows(
  hmFry,
  kgFry,
  goFry
) %&gt;%
  dplyr::filter(FDR &lt; 0.05) %&gt;%
  left_join(gsSizes) %&gt;%
  dplyr::select(
    gs_name, NGenes, Direction, PValue, FDR,
    gene_symbol, de_size
  ) %&gt;%
  mutate(
    DE = lapply(
      X = gene_symbol,
      FUN =  intersect, 
      y = dplyr::filter(deTable, DE)$gene_name
    ),
    DE = vapply(DE, paste, character(1), collapse = &quot;;&quot;)
  ) %&gt;%
  dplyr::select(
    `Gene Set` = gs_name, 
    `Nbr Detected Genes` = NGenes, 
    `Nbr DE Genes` = de_size, 
    Direction, PValue, FDR, 
    `DE Genes` = DE
  ) %&gt;%
  write_csv(
    here::here(&quot;output&quot;, &quot;Enrichment_Mutant_V_WT.csv&quot;)
  )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>devtools::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.3 (2020-02-29)
 os       Ubuntu 18.04.4 LTS          
 system   x86_64, linux-gnu           
 ui       X11                         
 language en_AU:en                    
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Melbourne         
 date     2020-03-07                  

─ Packages ───────────────────────────────────────────────────────────────────
 package              * version  date       lib source        
 AnnotationDbi        * 1.48.0   2019-10-29 [2] Bioconductor  
 askpass                1.1      2019-01-13 [2] CRAN (R 3.6.0)
 assertthat             0.2.1    2019-03-21 [2] CRAN (R 3.6.0)
 backports              1.1.5    2019-10-02 [2] CRAN (R 3.6.1)
 BiasedUrn            * 1.07     2015-12-28 [2] CRAN (R 3.6.1)
 Biobase              * 2.46.0   2019-10-29 [2] Bioconductor  
 BiocFileCache          1.10.2   2019-11-08 [2] Bioconductor  
 BiocGenerics         * 0.32.0   2019-10-29 [2] Bioconductor  
 BiocParallel           1.20.1   2019-12-21 [2] Bioconductor  
 biomaRt                2.42.0   2019-10-29 [2] Bioconductor  
 Biostrings             2.54.0   2019-10-29 [2] Bioconductor  
 bit                    1.1-15.2 2020-02-10 [2] CRAN (R 3.6.2)
 bit64                  0.9-7    2017-05-08 [2] CRAN (R 3.6.0)
 bitops                 1.0-6    2013-08-17 [2] CRAN (R 3.6.0)
 blob                   1.2.1    2020-01-20 [2] CRAN (R 3.6.2)
 broom                  0.5.4    2020-01-27 [2] CRAN (R 3.6.2)
 callr                  3.4.2    2020-02-12 [2] CRAN (R 3.6.2)
 cellranger             1.1.0    2016-07-27 [2] CRAN (R 3.6.0)
 cli                    2.0.1    2020-01-08 [2] CRAN (R 3.6.2)
 cluster                2.1.0    2019-06-19 [2] CRAN (R 3.6.1)
 colorspace             1.4-1    2019-03-18 [2] CRAN (R 3.6.0)
 crayon                 1.3.4    2017-09-16 [2] CRAN (R 3.6.0)
 curl                   4.3      2019-12-02 [2] CRAN (R 3.6.2)
 data.table             1.12.8   2019-12-09 [2] CRAN (R 3.6.2)
 DBI                    1.1.0    2019-12-15 [2] CRAN (R 3.6.2)
 dbplyr                 1.4.2    2019-06-17 [2] CRAN (R 3.6.0)
 DelayedArray           0.12.2   2020-01-06 [2] Bioconductor  
 desc                   1.2.0    2018-05-01 [2] CRAN (R 3.6.0)
 devtools               2.2.2    2020-02-17 [2] CRAN (R 3.6.2)
 digest                 0.6.25   2020-02-23 [2] CRAN (R 3.6.2)
 dplyr                * 0.8.4    2020-01-31 [2] CRAN (R 3.6.2)
 edgeR                * 3.28.1   2020-02-26 [2] Bioconductor  
 ellipsis               0.3.0    2019-09-20 [2] CRAN (R 3.6.1)
 evaluate               0.14     2019-05-28 [2] CRAN (R 3.6.0)
 FactoMineR             2.2      2020-02-05 [2] CRAN (R 3.6.2)
 fansi                  0.4.1    2020-01-08 [2] CRAN (R 3.6.2)
 farver                 2.0.3    2020-01-16 [2] CRAN (R 3.6.2)
 flashClust             1.01-2   2012-08-21 [2] CRAN (R 3.6.1)
 forcats              * 0.4.0    2019-02-17 [2] CRAN (R 3.6.0)
 fs                     1.3.1    2019-05-06 [2] CRAN (R 3.6.0)
 geneLenDataBase      * 1.22.0   2019-11-05 [2] Bioconductor  
 generics               0.0.2    2018-11-29 [2] CRAN (R 3.6.0)
 GenomeInfoDb           1.22.0   2019-10-29 [2] Bioconductor  
 GenomeInfoDbData       1.2.2    2019-11-21 [2] Bioconductor  
 GenomicAlignments      1.22.1   2019-11-12 [2] Bioconductor  
 GenomicFeatures        1.38.2   2020-02-15 [2] Bioconductor  
 GenomicRanges          1.38.0   2019-10-29 [2] Bioconductor  
 ggdendro               0.1-20   2016-04-27 [2] CRAN (R 3.6.0)
 ggplot2              * 3.2.1    2019-08-10 [2] CRAN (R 3.6.1)
 ggrepel                0.8.1    2019-05-07 [2] CRAN (R 3.6.0)
 git2r                  0.26.1   2019-06-29 [2] CRAN (R 3.6.1)
 glue                   1.3.1    2019-03-12 [2] CRAN (R 3.6.0)
 GO.db                  3.10.0   2019-11-21 [2] Bioconductor  
 goseq                * 1.38.0   2019-10-29 [2] Bioconductor  
 gridExtra              2.3      2017-09-09 [2] CRAN (R 3.6.0)
 gtable                 0.3.0    2019-03-25 [2] CRAN (R 3.6.0)
 haven                  2.2.0    2019-11-08 [2] CRAN (R 3.6.1)
 here                   0.1      2017-05-28 [2] CRAN (R 3.6.0)
 highr                  0.8      2019-03-20 [2] CRAN (R 3.6.0)
 hms                    0.5.3    2020-01-08 [2] CRAN (R 3.6.2)
 htmltools              0.4.0    2019-10-04 [2] CRAN (R 3.6.1)
 htmlwidgets            1.5.1    2019-10-08 [2] CRAN (R 3.6.1)
 httpuv                 1.5.2    2019-09-11 [2] CRAN (R 3.6.1)
 httr                   1.4.1    2019-08-05 [2] CRAN (R 3.6.1)
 hwriter                1.3.2    2014-09-10 [2] CRAN (R 3.6.0)
 IRanges              * 2.20.2   2020-01-13 [2] Bioconductor  
 jpeg                   0.1-8.1  2019-10-24 [2] CRAN (R 3.6.1)
 jsonlite               1.6.1    2020-02-02 [2] CRAN (R 3.6.2)
 kableExtra             1.1.0    2019-03-16 [2] CRAN (R 3.6.1)
 knitr                  1.28     2020-02-06 [2] CRAN (R 3.6.2)
 labeling               0.3      2014-08-23 [2] CRAN (R 3.6.0)
 later                  1.0.0    2019-10-04 [2] CRAN (R 3.6.1)
 lattice                0.20-40  2020-02-19 [4] CRAN (R 3.6.2)
 latticeExtra           0.6-29   2019-12-19 [2] CRAN (R 3.6.2)
 lazyeval               0.2.2    2019-03-15 [2] CRAN (R 3.6.0)
 leaps                  3.1      2020-01-16 [2] CRAN (R 3.6.2)
 lifecycle              0.1.0    2019-08-01 [2] CRAN (R 3.6.1)
 limma                * 3.42.2   2020-02-03 [2] Bioconductor  
 locfit                 1.5-9.1  2013-04-20 [2] CRAN (R 3.6.0)
 lubridate              1.7.4    2018-04-11 [2] CRAN (R 3.6.0)
 magrittr             * 1.5      2014-11-22 [2] CRAN (R 3.6.0)
 MASS                   7.3-51.5 2019-12-20 [4] CRAN (R 3.6.2)
 Matrix                 1.2-18   2019-11-27 [2] CRAN (R 3.6.1)
 matrixStats            0.55.0   2019-09-07 [2] CRAN (R 3.6.1)
 memoise                1.1.0    2017-04-21 [2] CRAN (R 3.6.0)
 mgcv                   1.8-31   2019-11-09 [4] CRAN (R 3.6.1)
 modelr                 0.1.6    2020-02-22 [2] CRAN (R 3.6.2)
 msigdbr              * 7.0.1    2019-09-04 [2] CRAN (R 3.6.2)
 munsell                0.5.0    2018-06-12 [2] CRAN (R 3.6.0)
 ngsReports           * 1.1.2    2019-10-16 [1] Bioconductor  
 nlme                   3.1-144  2020-02-06 [4] CRAN (R 3.6.2)
 openssl                1.4.1    2019-07-18 [2] CRAN (R 3.6.1)
 pander               * 0.6.3    2018-11-06 [2] CRAN (R 3.6.0)
 pheatmap             * 1.0.12   2019-01-04 [2] CRAN (R 3.6.0)
 pillar                 1.4.3    2019-12-20 [2] CRAN (R 3.6.2)
 pkgbuild               1.0.6    2019-10-09 [2] CRAN (R 3.6.1)
 pkgconfig              2.0.3    2019-09-22 [2] CRAN (R 3.6.1)
 pkgload                1.0.2    2018-10-29 [2] CRAN (R 3.6.0)
 plotly                 4.9.2    2020-02-12 [2] CRAN (R 3.6.2)
 plyr                   1.8.5    2019-12-10 [2] CRAN (R 3.6.2)
 png                    0.1-7    2013-12-03 [2] CRAN (R 3.6.0)
 prettyunits            1.1.1    2020-01-24 [2] CRAN (R 3.6.2)
 processx               3.4.2    2020-02-09 [2] CRAN (R 3.6.2)
 progress               1.2.2    2019-05-16 [2] CRAN (R 3.6.0)
 promises               1.1.0    2019-10-04 [2] CRAN (R 3.6.1)
 ps                     1.3.2    2020-02-13 [2] CRAN (R 3.6.2)
 purrr                * 0.3.3    2019-10-18 [2] CRAN (R 3.6.1)
 R6                     2.4.1    2019-11-12 [2] CRAN (R 3.6.1)
 rappdirs               0.3.1    2016-03-28 [2] CRAN (R 3.6.0)
 RColorBrewer         * 1.1-2    2014-12-07 [2] CRAN (R 3.6.0)
 Rcpp                   1.0.3    2019-11-08 [2] CRAN (R 3.6.1)
 RCurl                  1.98-1.1 2020-01-19 [2] CRAN (R 3.6.2)
 readr                * 1.3.1    2018-12-21 [2] CRAN (R 3.6.0)
 readxl                 1.3.1    2019-03-13 [2] CRAN (R 3.6.0)
 remotes                2.1.1    2020-02-15 [2] CRAN (R 3.6.2)
 reprex                 0.3.0    2019-05-16 [2] CRAN (R 3.6.0)
 reshape2               1.4.3    2017-12-11 [2] CRAN (R 3.6.0)
 rlang                  0.4.4    2020-01-28 [2] CRAN (R 3.6.2)
 rmarkdown              2.1      2020-01-20 [2] CRAN (R 3.6.2)
 rprojroot              1.3-2    2018-01-03 [2] CRAN (R 3.6.0)
 Rsamtools              2.2.3    2020-02-23 [2] Bioconductor  
 RSQLite                2.2.0    2020-01-07 [2] CRAN (R 3.6.2)
 rstudioapi             0.11     2020-02-07 [2] CRAN (R 3.6.2)
 rtracklayer            1.46.0   2019-10-29 [2] Bioconductor  
 rvest                  0.3.5    2019-11-08 [2] CRAN (R 3.6.1)
 S4Vectors            * 0.24.3   2020-01-18 [2] Bioconductor  
 scales               * 1.1.0    2019-11-18 [2] CRAN (R 3.6.1)
 scatterplot3d          0.3-41   2018-03-14 [2] CRAN (R 3.6.1)
 sessioninfo            1.1.1    2018-11-05 [2] CRAN (R 3.6.0)
 ShortRead              1.44.3   2020-02-03 [2] Bioconductor  
 statmod                1.4.34   2020-02-17 [2] CRAN (R 3.6.2)
 stringi                1.4.6    2020-02-17 [2] CRAN (R 3.6.2)
 stringr              * 1.4.0    2019-02-10 [2] CRAN (R 3.6.0)
 SummarizedExperiment   1.16.1   2019-12-19 [2] Bioconductor  
 testthat               2.3.1    2019-12-01 [2] CRAN (R 3.6.1)
 tibble               * 2.1.3    2019-06-06 [2] CRAN (R 3.6.0)
 tidyr                * 1.0.2    2020-01-24 [2] CRAN (R 3.6.2)
 tidyselect             1.0.0    2020-01-27 [2] CRAN (R 3.6.2)
 tidyverse            * 1.3.0    2019-11-21 [2] CRAN (R 3.6.1)
 truncnorm              1.0-8    2018-02-27 [2] CRAN (R 3.6.0)
 UpSetR               * 1.4.0    2019-05-22 [2] CRAN (R 3.6.0)
 usethis                1.5.1    2019-07-04 [2] CRAN (R 3.6.1)
 vctrs                  0.2.3    2020-02-20 [2] CRAN (R 3.6.2)
 viridisLite            0.3.0    2018-02-01 [2] CRAN (R 3.6.0)
 webshot                0.5.2    2019-11-22 [2] CRAN (R 3.6.1)
 whisker                0.4      2019-08-28 [2] CRAN (R 3.6.1)
 withr                  2.1.2    2018-03-15 [2] CRAN (R 3.6.0)
 workflowr            * 1.6.0    2019-12-19 [2] CRAN (R 3.6.2)
 xfun                   0.12     2020-01-13 [2] CRAN (R 3.6.2)
 XML                    3.99-0.3 2020-01-20 [2] CRAN (R 3.6.2)
 xml2                   1.2.2    2019-08-09 [2] CRAN (R 3.6.1)
 XVector                0.26.0   2019-10-29 [2] Bioconductor  
 yaml                   2.2.1    2020-02-01 [2] CRAN (R 3.6.2)
 zlibbioc               1.32.0   2019-10-29 [2] Bioconductor  
 zoo                    1.8-7    2020-01-10 [2] CRAN (R 3.6.2)

[1] /home/steveped/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
