<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Steve Pederson" />


<title>Enrichment Analysis: Mutant Vs Wild-Type</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">20170327_Psen2S4Ter_RNASeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Enrichment Analysis: Mutant Vs Wild-Type</h1>
<h4 class="author">Steve Pederson</h4>
<h4 class="date">17 February, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-02-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>20170327_Psen2S4Ter_RNASeq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200119code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200119)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200119code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomUofABioinformaticsHub20170327Psen2S4TerRNASeqtree53ed0e30b1eff5b43093d31a0de4443f96125be6targetblank53ed0e3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/tree/53ed0e30b1eff5b43093d31a0de4443f96125be6" target="_blank">53ed0e3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomUofABioinformaticsHub20170327Psen2S4TerRNASeqtree53ed0e30b1eff5b43093d31a0de4443f96125be6targetblank53ed0e3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/53ed0e30b1eff5b43093d31a0de4443f96125be6/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">53ed0e3</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-17
</td>
<td>
Corrected Ensembl Release
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/f55be858a66d2812a95303ad7146f952727953ba/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">f55be85</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-17
</td>
<td>
Added commas
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/8f29458fe6b2b7523c023b18629b270b2e500425/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">8f29458</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-17
</td>
<td>
Minor tweaks to formatting
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/9104ecdc9ad8512ec277c6c131d36de86edcf845/docs/3_Enrichment_MutantVsWT.html" target="_blank">9104ecd</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
First draft of Hom Vs Het
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/207cdc8d7398f80115e58d65fef3a9d2f0bbf044/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">207cdc8</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Added code for Hom Vs Het Enrichment
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/468e6e3a5e47e1e25be13e92d15dc4e1d7067130/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">468e6e3</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Ran enrichment of Mut Vs WT
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/468e6e3a5e47e1e25be13e92d15dc4e1d7067130/docs/3_Enrichment_MutantVsWT.html" target="_blank">468e6e3</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Ran enrichment of Mut Vs WT
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/3a9933c5330cee6a4957bdfdf7794251f9346dd2/analysis/3_Enrichment_MutantVsWT.Rmd" target="_blank">3a9933c</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Finished Enrichment analysis on MutVsWt
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/3a9933c5330cee6a4957bdfdf7794251f9346dd2/docs/3_Enrichment_MutantVsWT.html" target="_blank">3a9933c</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Finished Enrichment analysis on MutVsWt
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level1">
<h1>Setup</h1>
<pre class="r"><code>library(tidyverse)
library(magrittr)
library(edgeR)
library(scales)
library(pander)
library(goseq)
library(msigdbr)
library(AnnotationDbi)
library(RColorBrewer)
library(ngsReports)
library(fgsea)
library(metap)</code></pre>
<pre class="r"><code>theme_set(theme_bw())
panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;table.style&quot;, &quot;rmarkdown&quot;)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)</code></pre>
<pre class="r"><code>samples &lt;- read_csv(&quot;data/samples.csv&quot;) %&gt;%
  distinct(sampleName, .keep_all = TRUE) %&gt;%
  dplyr::select(sample = sampleName, sampleID, genotype) %&gt;%
  mutate(
    genotype = factor(genotype, levels = c(&quot;WT&quot;, &quot;Het&quot;, &quot;Hom&quot;)),
    mutant = genotype %in% c(&quot;Het&quot;, &quot;Hom&quot;),
    homozygous = genotype == &quot;Hom&quot;
  )
genoCols &lt;- samples$genotype %&gt;%
  levels() %&gt;%
  length() %&gt;%
  brewer.pal(&quot;Set1&quot;) %&gt;%
  setNames(levels(samples$genotype))</code></pre>
<pre class="r"><code>dgeList &lt;- read_rds(&quot;data/dgeList.rds&quot;)
fit &lt;- read_rds(&quot;data/fit.rds&quot;)
entrezGenes &lt;- dgeList$genes %&gt;%
  dplyr::filter(!is.na(entrezid)) %&gt;%
  unnest(entrezid) %&gt;%
  dplyr::rename(entrez_gene = entrezid)</code></pre>
<pre class="r"><code>formatP &lt;- function(p, m = 0.0001){
  out &lt;- rep(&quot;&quot;, length(p))
  out[p &lt; m] &lt;- sprintf(&quot;%.2e&quot;, p[p&lt;m])
  out[p &gt;= m] &lt;- sprintf(&quot;%.4f&quot;, p[p&gt;=m])
  out
}</code></pre>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Enrichment analysis for this dataset present some challenges. Despite normalisation to account for gene length and GC bias, some appeared to still be present in the final results. In addition, the confounding of incomplete rRNA removal with genotype may lead to other distortions in both DE genes and ranking statistics.</p>
<p>Two steps for enrichment analysis will be undertaken.</p>
<ol style="list-style-type: decimal">
<li>Testing for enrichment within <em>discrete sets of DE genes</em> as defined in the previous steps</li>
<li>Testing for enrichment within <em>ranked lists</em>, regardless of DE status or statistical significance, before integration of all enrichment analyses using Fisher’s method for combining p-values.</li>
</ol>
<p>Testing for enrichment <em>within discrete gene sets</em> will be performed using <code>goseq</code> as this allows for the incorporation of a single covariate as a predictor of differential expression. GC content, gene length and correlation with rRNA removal can all be supplied as separate covariates.</p>
<p>Testing for enrichment <em>with ranked lists</em> will be performed using:</p>
<ol style="list-style-type: decimal">
<li><code>fry</code> as this can take into account inter-gene correlations. Values supplied will be fitted values for each gene/sample as these have been corrected for GC and length biases.</li>
<li><code>camera</code>, which also accommodates inter-gene correlations. Values supplied will be fitted values for each gene/sample as these have been corrected for GC and length biases.</li>
<li><code>fgsea</code> which is an R implementation of GSEA. This approach simply takes a ranked list and doesn’t directly account for correlations. However, the ranked list will be derived from analysis using CQN-normalisation so will be robust to these technical artefacts.</li>
</ol>
<p>In the case of <code>camera</code>, inter-gene correlations will be calculated for each gene-set prior to analysis to ensure more conservative p-values are obtained.</p>
</div>
<div id="databases-used-for-testing" class="section level1">
<h1>Databases used for testing</h1>
<p>Data was sourced using the <code>msigdbr</code> package. The initial database used for testing was the Hallmark Gene Sets, with mappings from gene-set to EntrezGene IDs performed by the package authors.</p>
<div id="hallmark-gene-sets" class="section level2">
<h2>Hallmark Gene Sets</h2>
<pre class="r"><code>hm &lt;- msigdbr(&quot;Danio rerio&quot;, category = &quot;H&quot;)  %&gt;% 
  left_join(entrezGenes) %&gt;%
  dplyr::filter(!is.na(gene_id)) 
hmByGene &lt;- hm %&gt;%
  split(f = .$gene_id) %&gt;%
  lapply(extract2, &quot;gs_name&quot;)
hmByID &lt;- hm %&gt;%
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;gene_id&quot;)</code></pre>
<p>Mappings are required from gene to pathway, and Ensembl identifiers were used to map from gene to pathway, based on the mappings in the previously used annotations (Ensembl Release 98). A total of 3,459 Ensembl IDs were mapped to pathways from the hallmark gene sets.</p>
</div>
<div id="kegg-gene-sets" class="section level2">
<h2>KEGG Gene Sets</h2>
<pre class="r"><code>kg &lt;- msigdbr(&quot;Danio rerio&quot;, category = &quot;C2&quot;, subcategory = &quot;CP:KEGG&quot;)  %&gt;% 
  left_join(entrezGenes) %&gt;%
  dplyr::filter(!is.na(gene_id)) 
kgByGene &lt;- kg  %&gt;%
  split(f = .$gene_id) %&gt;%
  lapply(extract2, &quot;gs_name&quot;)
kgByID &lt;- kg  %&gt;%
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;gene_id&quot;)</code></pre>
<p>The same mapping process was applied to KEGG gene sets. A total of 3,614 Ensembl IDs were mapped to pathways from the KEGG gene sets.</p>
</div>
<div id="gene-ontology-gene-sets" class="section level2">
<h2>Gene Ontology Gene Sets</h2>
<pre class="r"><code>goSummaries &lt;- url(&quot;https://uofabioinformaticshub.github.io/summaries2GO/data/goSummaries.RDS&quot;) %&gt;%
  readRDS() %&gt;%
  mutate(
    Term = Term(id),
    gs_name = Term %&gt;% str_to_upper() %&gt;% str_replace_all(&quot;[ -]&quot;, &quot;_&quot;),
    gs_name = paste0(&quot;GO_&quot;, gs_name)
    )
minPath &lt;- 3</code></pre>
<pre class="r"><code>go &lt;- msigdbr(&quot;Danio rerio&quot;, category = &quot;C5&quot;) %&gt;% 
  left_join(entrezGenes) %&gt;%
  dplyr::filter(!is.na(gene_id)) %&gt;%
  left_join(goSummaries) %&gt;% 
  dplyr::filter(shortest_path &gt;= minPath) 
goByGene &lt;- go %&gt;%
  split(f = .$gene_id) %&gt;%
  lapply(extract2, &quot;gs_name&quot;)
goByID &lt;- go %&gt;%
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;gene_id&quot;)</code></pre>
<p>For analysis of gene-sets from the GO database, gene-sets were restricted to those with 3 or more steps back to the ontology root terms. A total of 11,245 Ensembl IDs were mapped to pathways from restricted database of 8,834 GO gene sets.</p>
</div>
</div>
<div id="enrichment-in-the-de-gene-set" class="section level1">
<h1>Enrichment in the DE Gene Set</h1>
<pre class="r"><code>deTable &lt;- file.path(&quot;output&quot;, &quot;psen2VsWT.csv&quot;) %&gt;% 
  read_csv() %&gt;%
  mutate(
    entrezid = dgeList$genes$entrezid[gene_id]
  )</code></pre>
<p>The first step of analysis using <code>goseq</code>, regardless of the gene-set, is estimation of the probability weight function (PWF) which quantifies the probability of a gene being considered as DE based on a single covariate. As GC content and length should have been accounted for during conditional-quantile normalisation, these were not required for any bias. However, the <em>gene-level correlations with rRNA contamination</em> were instead used a predictor of bias in selection of a gene as being DE.</p>
<pre class="r"><code>rawFqc &lt;- list.files(
  path = &quot;data/0_rawData/FastQC/&quot;,
  pattern = &quot;zip&quot;,
  full.names = TRUE
) %&gt;%
  FastqcDataList()
gc &lt;- getModule(rawFqc, &quot;Per_sequence_GC&quot;) %&gt;%
  group_by(Filename) %&gt;% 
  mutate(Freq = Count / sum(Count)) %&gt;%
  ungroup()
gcDev &lt;- gc %&gt;%
  left_join(getGC(gcTheoretical, &quot;Drerio&quot;, &quot;Trans&quot;)) %&gt;%
  mutate(
    sample = str_remove(Filename, &quot;_R[12].fastq.gz&quot;),
    resid = Freq - Drerio
  ) %&gt;% 
  left_join(samples) %&gt;%
  group_by(sample) %&gt;%
  summarise(
    ss = sum(resid^2),
    n = n(),
    sd = sqrt(ss / (n - 1))
  )</code></pre>
<pre class="r"><code>riboVec &lt;- structure(gcDev$sd, names = gcDev$sample)
riboCors &lt;- cpm(dgeList, log = TRUE) %&gt;%
  apply(1, function(x){
    cor(x, riboVec[names(x)])
  })</code></pre>
<p>Values were calculated as per the previous steps, using the logCPM values for each gene, with the sample-level standard deviations from the theoretical GC distribution being used as a measure of rRNA contamination. For estimation of the probability weight function, squared correlations were used to place negative and positive correlations on the same scale. This accounts for genes which are both negatively &amp; positively biased by the presence of excessive rRNA. Clearly, the confounding of genotype with rRNA means that some genes driven by the genuine biology may be down-weighted under this approach.</p>
<pre class="r"><code>riboPwf &lt;- deTable %&gt;%
  mutate(riboCors = riboCors[gene_id]^2) %&gt;%
  dplyr::select(gene_id, DE, riboCors) %&gt;%
  distinct(gene_id, .keep_all = TRUE) %&gt;% 
  with(
    nullp(
      DEgenes = structure(
        as.integer(DE), names = gene_id
      ), 
      genome = &quot;danRer10&quot;, 
      id = &quot;ensGene&quot;, 
      bias.data = riboCors,
      plot.fit = FALSE
    )
  )
plotPWF(riboPwf, main = &quot;Bias from rRNA proportions&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3_Enrichment_MutantVsWT.Rmd/riboPwf-1.png" alt="*Using this approach, it was clear that correlation with rRNA proportions significantly biased the probability of a gene being considered as DE.*" width="384" />
<p class="caption">
<em>Using this approach, it was clear that correlation with rRNA proportions significantly biased the probability of a gene being considered as DE.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-riboPwf-1">
Past versions of riboPwf-1.png
</button>
</p>
<div id="fig-riboPwf-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/468e6e3a5e47e1e25be13e92d15dc4e1d7067130/docs/figure/3_Enrichment_MutantVsWT.Rmd/riboPwf-1.png" target="_blank">468e6e3</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>All gene-sets were then tested using this PWF.</p>
<div id="hallmark-gene-sets-1" class="section level2">
<h2>Hallmark Gene Sets</h2>
<pre class="r"><code>hmRiboGoseq &lt;- goseq(riboPwf, gene2cat = hmByGene) %&gt;%
  as_tibble %&gt;%
  dplyr::filter(numDEInCat &gt; 0) %&gt;%
  mutate( 
    adjP = p.adjust(over_represented_pvalue, method = &quot;bonf&quot;),
    FDR = p.adjust(over_represented_pvalue, method = &quot;fdr&quot;)
  ) %&gt;%
  dplyr::select(-contains(&quot;under&quot;)) %&gt;%
  dplyr::rename(
    gs_name = category,
    PValue = over_represented_pvalue
  )</code></pre>
<p>No gene-sets achieved significance in the DE genes with the lowest FDR being 41%</p>
</div>
<div id="kegg-gene-sets-1" class="section level2">
<h2>KEGG Gene Sets</h2>
<pre class="r"><code>kgRiboGoseq &lt;- goseq(riboPwf, gene2cat = kgByGene) %&gt;%
  as_tibble %&gt;%
  dplyr::filter(numDEInCat &gt; 0) %&gt;%
  mutate( 
    adjP = p.adjust(over_represented_pvalue, method = &quot;bonf&quot;),
    FDR = p.adjust(over_represented_pvalue, method = &quot;fdr&quot;)
  ) %&gt;%
  dplyr::select(-contains(&quot;under&quot;))  %&gt;%
  dplyr::rename(
    gs_name = category,
    PValue = over_represented_pvalue
  )</code></pre>
<pre class="r"><code>kgRiboGoseq %&gt;%
  dplyr::slice(1:5) %&gt;%
  mutate(
    p = formatP(PValue),
    adjP = formatP(adjP),
    FDR = formatP(FDR)
  ) %&gt;%
  dplyr::select(
    `Gene Set` = gs_name,
    DE = numDEInCat,
    `Set Size` = numInCat,
    PValue,
    `p~bonf~` = adjP,
    `p~FDR~` = FDR
  ) %&gt;%
  pander(
    justify = &quot;lrrrrr&quot;,
    caption = paste(
      &quot;The&quot;, nrow(.), &quot;most highly-ranked KEGG pathways.&quot;,
      &quot;Bonferroni-adjusted p-values are the most stringent and give high&quot;,
      &quot;confidence when below 0.05.&quot;
    )
  )</code></pre>
<table>
<caption>The 5 most highly-ranked KEGG pathways. Bonferroni-adjusted p-values are the most stringent and give high confidence when below 0.05.</caption>
<thead>
<tr class="header">
<th align="left">Gene Set</th>
<th align="right">DE</th>
<th align="right">Set Size</th>
<th align="right">PValue</th>
<th align="right">p<sub>bonf</sub></th>
<th align="right">p<sub>FDR</sub></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">KEGG_RIBOSOME</td>
<td align="right">26</td>
<td align="right">80</td>
<td align="right">4.754e-09</td>
<td align="right">5.32e-07</td>
<td align="right">5.32e-07</td>
</tr>
<tr class="even">
<td align="left">KEGG_PRIMARY_IMMUNODEFICIENCY</td>
<td align="right">2</td>
<td align="right">15</td>
<td align="right">0.01642</td>
<td align="right">1.0000</td>
<td align="right">0.7173</td>
</tr>
<tr class="odd">
<td align="left">KEGG_CYSTEINE_AND_METHIONINE_METABOLISM</td>
<td align="right">4</td>
<td align="right">26</td>
<td align="right">0.02959</td>
<td align="right">1.0000</td>
<td align="right">0.7173</td>
</tr>
<tr class="even">
<td align="left">KEGG_ASTHMA</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">0.04088</td>
<td align="right">1.0000</td>
<td align="right">0.7173</td>
</tr>
<tr class="odd">
<td align="left">KEGG_RETINOL_METABOLISM</td>
<td align="right">3</td>
<td align="right">25</td>
<td align="right">0.0512</td>
<td align="right">1.0000</td>
<td align="right">0.7173</td>
</tr>
</tbody>
</table>
<p>Notably, the KEGG gene-set for Ribosomal genes was detected as enriched in the set of DE genes, with no other KEGG gene-sets being considered significant.</p>
</div>
<div id="go-gene-sets" class="section level2">
<h2>GO Gene Sets</h2>
<pre class="r"><code>goRiboGoseq &lt;- goseq(riboPwf, gene2cat = goByGene) %&gt;%
  as_tibble %&gt;%
  dplyr::filter(numDEInCat &gt; 0) %&gt;%
  mutate( 
    adjP = p.adjust(over_represented_pvalue, method = &quot;bonf&quot;),
    FDR = p.adjust(over_represented_pvalue, method = &quot;fdr&quot;)
  ) %&gt;%
  dplyr::select(-contains(&quot;under&quot;)) %&gt;%
  dplyr::rename(
    gs_name = category,
    PValue = over_represented_pvalue
  )</code></pre>
<pre class="r"><code>goRiboGoseq %&gt;%
  dplyr::filter(adjP &lt; 0.05) %&gt;%
  mutate(
    p = formatP(PValue),
    adjP = formatP(adjP),
    FDR = formatP(FDR)
  ) %&gt;%
  dplyr::select(
    `Gene Set` = gs_name,
    DE = numDEInCat,
    `Set Size` = numInCat,
    PValue,
    `p~bonf~` = adjP,
    `p~FDR~` = FDR
  ) %&gt;%
  pander(
    justify = &quot;lrrrrr&quot;,
    caption = paste(
      &quot;*The&quot;, nrow(.), &quot;most highly-ranked GO terms.&quot;,
      &quot;Bonferroni-adjusted p-values are the most stringent and give high&quot;,
      &quot;confidence when below 0.05, with all terms here reaching this threshold.&quot;,
      &quot;However, most terms once again indicate the presence of rRNA.*&quot;
    )
  )</code></pre>
<table>
<caption><em>The 14 most highly-ranked GO terms. Bonferroni-adjusted p-values are the most stringent and give high confidence when below 0.05, with all terms here reaching this threshold. However, most terms once again indicate the presence of rRNA.</em></caption>
<colgroup>
<col width="59%" />
<col width="3%" />
<col width="8%" />
<col width="9%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Gene Set</th>
<th align="right">DE</th>
<th align="right">Set Size</th>
<th align="right">PValue</th>
<th align="right">p<sub>bonf</sub></th>
<th align="right">p<sub>FDR</sub></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE</td>
<td align="right">28</td>
<td align="right">93</td>
<td align="right">7.36e-10</td>
<td align="right">2.83e-06</td>
<td align="right">2.03e-06</td>
</tr>
<tr class="even">
<td align="left">GO_CYTOSOLIC_RIBOSOME</td>
<td align="right">27</td>
<td align="right">97</td>
<td align="right">1.056e-09</td>
<td align="right">4.06e-06</td>
<td align="right">2.03e-06</td>
</tr>
<tr class="odd">
<td align="left">GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM</td>
<td align="right">27</td>
<td align="right">105</td>
<td align="right">1.6e-08</td>
<td align="right">6.16e-05</td>
<td align="right">2.05e-05</td>
</tr>
<tr class="even">
<td align="left">GO_TRANSLATIONAL_INITIATION</td>
<td align="right">29</td>
<td align="right">177</td>
<td align="right">2.178e-07</td>
<td align="right">0.0008</td>
<td align="right">0.0002</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEIN_TARGETING_TO_MEMBRANE</td>
<td align="right">30</td>
<td align="right">171</td>
<td align="right">2.533e-07</td>
<td align="right">0.0010</td>
<td align="right">0.0002</td>
</tr>
<tr class="even">
<td align="left">GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM</td>
<td align="right">27</td>
<td align="right">128</td>
<td align="right">3.849e-07</td>
<td align="right">0.0015</td>
<td align="right">0.0002</td>
</tr>
<tr class="odd">
<td align="left">GO_CYTOSOLIC_PART</td>
<td align="right">30</td>
<td align="right">206</td>
<td align="right">4.424e-07</td>
<td align="right">0.0017</td>
<td align="right">0.0002</td>
</tr>
<tr class="even">
<td align="left">GO_VIRAL_GENE_EXPRESSION</td>
<td align="right">29</td>
<td align="right">172</td>
<td align="right">6.939e-07</td>
<td align="right">0.0027</td>
<td align="right">0.0003</td>
</tr>
<tr class="odd">
<td align="left">GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS</td>
<td align="right">30</td>
<td align="right">183</td>
<td align="right">7.911e-07</td>
<td align="right">0.0030</td>
<td align="right">0.0003</td>
</tr>
<tr class="even">
<td align="left">GO_RIBOSOMAL_SUBUNIT</td>
<td align="right">28</td>
<td align="right">175</td>
<td align="right">1.99e-06</td>
<td align="right">0.0077</td>
<td align="right">0.0008</td>
</tr>
<tr class="odd">
<td align="left">GO_RIBOSOME</td>
<td align="right">30</td>
<td align="right">210</td>
<td align="right">2.659e-06</td>
<td align="right">0.0102</td>
<td align="right">0.0009</td>
</tr>
<tr class="even">
<td align="left">GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT</td>
<td align="right">15</td>
<td align="right">51</td>
<td align="right">2.826e-06</td>
<td align="right">0.0109</td>
<td align="right">0.0009</td>
</tr>
<tr class="odd">
<td align="left">GO_SYMPORTER_ACTIVITY</td>
<td align="right">10</td>
<td align="right">94</td>
<td align="right">7.571e-06</td>
<td align="right">0.0291</td>
<td align="right">0.0022</td>
</tr>
<tr class="even">
<td align="left">GO_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">17</td>
<td align="right">228</td>
<td align="right">1.23e-05</td>
<td align="right">0.0473</td>
<td align="right">0.0034</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="enrichment-testing-on-ranked-lists" class="section level1">
<h1>Enrichment Testing on Ranked Lists</h1>
<pre class="r"><code>rnk &lt;- structure(
  -sign(deTable$logFC)*log10(deTable$PValue), 
  names = deTable$gene_id
) %&gt;% sort()
np &lt;- 1e5</code></pre>
<p>Genes were ranked by -sign(logFC)*log<sub>10</sub>(p) for approaches which required a ranked list. Multiple approaches were first calculated individually, before being combined for the final integrated set of results.</p>
<div id="hallmark-gene-sets-2" class="section level2">
<h2>Hallmark Gene Sets</h2>
<pre class="r"><code>hmFry &lt;- fit$fitted.values %&gt;%
  cpm(log = TRUE) %&gt;%
  fry(
    index = hmByID,
    design = fit$design,
    contrast = &quot;mutant&quot;,
    sort = &quot;directional&quot;
    ) %&gt;%
  rownames_to_column(&quot;gs_name&quot;) %&gt;%
  as_tibble()</code></pre>
<p>For analysis under <code>camera</code> when inter-gene correlations were calculated for a more conservative result.</p>
<pre class="r"><code>hmCamera &lt;- fit$fitted.values %&gt;%
  cpm(log = TRUE) %&gt;%
  camera(
    index = hmByID,
    design = fit$design,
    contrast = &quot;mutant&quot;,
    inter.gene.cor = NULL
    ) %&gt;%
  rownames_to_column(&quot;gs_name&quot;) %&gt;%
  as_tibble()</code></pre>
<p>For generation of the GSEA ranked list, 100,000 permutations were conducted.</p>
<pre class="r"><code>hmGsea &lt;- fgsea(
  pathways = hmByID, 
  stats = rnk,
  nperm = np
) %&gt;%
  as_tibble() %&gt;%
  dplyr::rename(gs_name = pathway, PValue = pval) %&gt;%
  arrange(PValue)</code></pre>
<p>Results for all analyses, including goseq were then combined using Wilkinson’s method to combine p-values. For a conservative approach, under <span class="math inline">\(m\)</span> tests, the <span class="math inline">\(m - 1^{\text{th}}\)</span> smallest p-value was chosen.</p>
<pre class="r"><code>hmMeta &lt;- hmFry %&gt;%
  dplyr::select(gs_name, fry = PValue) %&gt;%
  left_join(
    dplyr::select(hmCamera, gs_name, camera = PValue)
  ) %&gt;%
  left_join(
    dplyr::select(hmGsea, gs_name, gsea = PValue)
  ) %&gt;%
  left_join(
    dplyr::select(hmRiboGoseq, gs_name, goseq = PValue)
  ) %&gt;%
  pivot_longer(
    cols = one_of(c(&quot;fry&quot;, &quot;camera&quot;, &quot;gsea&quot;, &quot;goseq&quot;)),
    names_to = &quot;method&quot;,
    values_to = &quot;p&quot;
  ) %&gt;%
  dplyr::filter(!is.na(p)) %&gt;%
  group_by(gs_name) %&gt;%
  summarise(
    n = n(),
    p = wilkinsonp(p, r = n - 1)$p
  ) %&gt;%
  arrange(p) %&gt;%
  mutate(
    FDR = p.adjust(p, &quot;fdr&quot;),
    adjP = p.adjust(p, &quot;bonf&quot;)
  ) </code></pre>
<pre class="r"><code>hmMeta %&gt;%
  dplyr::filter(FDR &lt; 0.05) %&gt;%
  mutate_at(vars(one_of(c(&quot;p&quot;, &quot;FDR&quot;, &quot;adjP&quot;))), formatP) %&gt;%
  dplyr::select(`Gene Set` = gs_name, p, `p~FDR~` = FDR, `p~bonf~` = adjP) %&gt;%
  pander(
    caption = &quot;Results from combining all above approaches for the Hallmark Gene Sets. All terms are significant to an FDR of 0.05.&quot;,
    justify = &quot;lrrr&quot;
  )</code></pre>
<table>
<caption>Results from combining all above approaches for the Hallmark Gene Sets. All terms are significant to an FDR of 0.05.</caption>
<thead>
<tr class="header">
<th align="left">Gene Set</th>
<th align="right">p</th>
<th align="right">p<sub>FDR</sub></th>
<th align="right">p<sub>bonf</sub></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HALLMARK_XENOBIOTIC_METABOLISM</td>
<td align="right">0.0001</td>
<td align="right">0.0057</td>
<td align="right">0.0057</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_MYC_TARGETS_V1</td>
<td align="right">0.0003</td>
<td align="right">0.0069</td>
<td align="right">0.0138</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY</td>
<td align="right">0.0009</td>
<td align="right">0.0143</td>
<td align="right">0.0428</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_ADIPOGENESIS</td>
<td align="right">0.0019</td>
<td align="right">0.0243</td>
<td align="right">0.0971</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_OXIDATIVE_PHOSPHORYLATION</td>
<td align="right">0.0024</td>
<td align="right">0.0243</td>
<td align="right">0.1217</td>
</tr>
</tbody>
</table>
</div>
<div id="kegg-gene-sets-2" class="section level2">
<h2>KEGG Gene Sets</h2>
<pre class="r"><code>kgFry &lt;- fit$fitted.values %&gt;%
  cpm(log = TRUE) %&gt;%
  fry(
    index = kgByID,
    design = fit$design,
    contrast = &quot;mutant&quot;,
    sort = &quot;directional&quot;
    ) %&gt;%
  rownames_to_column(&quot;gs_name&quot;) %&gt;%
  as_tibble()</code></pre>
<p>For analysis under <code>camera</code> when inter-gene correlations were calculated for a more conservative result.</p>
<pre class="r"><code>kgCamera &lt;- fit$fitted.values %&gt;%
  cpm(log = TRUE) %&gt;%
  camera(
    index = kgByID,
    design = fit$design,
    contrast = &quot;mutant&quot;,
    inter.gene.cor = NULL
    ) %&gt;%
  rownames_to_column(&quot;gs_name&quot;) %&gt;%
  as_tibble()</code></pre>
<p>For generation of the GSEA ranked list, 100,000 permutations were conducted.</p>
<pre class="r"><code>kgGsea &lt;- fgsea(
  pathways = kgByID, 
  stats = rnk,
  nperm = np
) %&gt;%
  as_tibble() %&gt;%
  dplyr::rename(gs_name = pathway, PValue = pval) %&gt;%
  arrange(PValue)</code></pre>
<p>Results for all analyses, including goseq were then combined using Wilkinson’s method to combine p-values. For a conservative approach, under <span class="math inline">\(m\)</span> tests, the <span class="math inline">\(m - 1^{\text{th}}\)</span> smallest p-value was chosen.</p>
<pre class="r"><code>kgMeta &lt;- kgFry %&gt;%
  dplyr::select(gs_name, fry = PValue) %&gt;%
  left_join(
    dplyr::select(kgCamera, gs_name, camera = PValue)
  ) %&gt;%
  left_join(
    dplyr::select(kgGsea, gs_name, gsea = PValue)
  ) %&gt;%
  left_join(
    dplyr::select(kgRiboGoseq, gs_name, goseq = PValue)
  ) %&gt;%
  pivot_longer(
    cols = one_of(c(&quot;fry&quot;, &quot;camera&quot;, &quot;gsea&quot;, &quot;goseq&quot;)),
    names_to = &quot;method&quot;,
    values_to = &quot;p&quot;
  ) %&gt;%
  dplyr::filter(!is.na(p)) %&gt;%
  group_by(gs_name) %&gt;%
  summarise(
    n = n(),
    p = wilkinsonp(p, r = n - 1)$p
  ) %&gt;%
  arrange(p) %&gt;%
  mutate(
    FDR = p.adjust(p, &quot;fdr&quot;),
    adjP = p.adjust(p, &quot;bonf&quot;)
  ) </code></pre>
<pre class="r"><code>kgMeta %&gt;%
  dplyr::filter(FDR &lt; 0.01) %&gt;%
  mutate_at(vars(one_of(c(&quot;p&quot;, &quot;FDR&quot;, &quot;adjP&quot;))), formatP) %&gt;%
  dplyr::select(`Gene Set` = gs_name, p, `p~FDR~` = FDR, `p~bonf~` = adjP) %&gt;%
  pander(
    caption = &quot;Results from combining all above approaches for the Hallmark Gene Sets. All terms are significant to an FDR of 0.01.&quot;,
    justify = &quot;lrrr&quot;
  )</code></pre>
<table>
<caption>Results from combining all above approaches for the Hallmark Gene Sets. All terms are significant to an FDR of 0.01.</caption>
<thead>
<tr class="header">
<th align="left">Gene Set</th>
<th align="right">p</th>
<th align="right">p<sub>FDR</sub></th>
<th align="right">p<sub>bonf</sub></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">KEGG_RIBOSOME</td>
<td align="right">5.16e-14</td>
<td align="right">9.60e-12</td>
<td align="right">9.60e-12</td>
</tr>
<tr class="even">
<td align="left">KEGG_BUTANOATE_METABOLISM</td>
<td align="right">9.17e-09</td>
<td align="right">8.53e-07</td>
<td align="right">1.71e-06</td>
</tr>
<tr class="odd">
<td align="left">KEGG_PORPHYRIN_AND_CHLOROPHYLL_METABOLISM</td>
<td align="right">1.80e-07</td>
<td align="right">1.11e-05</td>
<td align="right">3.34e-05</td>
</tr>
<tr class="even">
<td align="left">KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION</td>
<td align="right">2.87e-07</td>
<td align="right">1.33e-05</td>
<td align="right">5.33e-05</td>
</tr>
<tr class="odd">
<td align="left">KEGG_LIMONENE_AND_PINENE_DEGRADATION</td>
<td align="right">1.29e-06</td>
<td align="right">4.79e-05</td>
<td align="right">0.0002</td>
</tr>
<tr class="even">
<td align="left">KEGG_ASCORBATE_AND_ALDARATE_METABOLISM</td>
<td align="right">1.78e-06</td>
<td align="right">5.52e-05</td>
<td align="right">0.0003</td>
</tr>
<tr class="odd">
<td align="left">KEGG_BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS</td>
<td align="right">2.76e-06</td>
<td align="right">7.33e-05</td>
<td align="right">0.0005</td>
</tr>
<tr class="even">
<td align="left">KEGG_PRIMARY_IMMUNODEFICIENCY</td>
<td align="right">1.75e-05</td>
<td align="right">0.0004</td>
<td align="right">0.0033</td>
</tr>
<tr class="odd">
<td align="left">KEGG_PROXIMAL_TUBULE_BICARBONATE_RECLAMATION</td>
<td align="right">1.99e-05</td>
<td align="right">0.0004</td>
<td align="right">0.0037</td>
</tr>
<tr class="even">
<td align="left">KEGG_OXIDATIVE_PHOSPHORYLATION</td>
<td align="right">3.59e-05</td>
<td align="right">0.0007</td>
<td align="right">0.0067</td>
</tr>
<tr class="odd">
<td align="left">KEGG_CYSTEINE_AND_METHIONINE_METABOLISM</td>
<td align="right">0.0001</td>
<td align="right">0.0017</td>
<td align="right">0.0189</td>
</tr>
<tr class="even">
<td align="left">KEGG_TERPENOID_BACKBONE_BIOSYNTHESIS</td>
<td align="right">0.0002</td>
<td align="right">0.0025</td>
<td align="right">0.0297</td>
</tr>
<tr class="odd">
<td align="left">KEGG_PEROXISOME</td>
<td align="right">0.0002</td>
<td align="right">0.0033</td>
<td align="right">0.0427</td>
</tr>
<tr class="even">
<td align="left">KEGG_ASTHMA</td>
<td align="right">0.0003</td>
<td align="right">0.0035</td>
<td align="right">0.0493</td>
</tr>
<tr class="odd">
<td align="left">KEGG_PARKINSONS_DISEASE</td>
<td align="right">0.0003</td>
<td align="right">0.0035</td>
<td align="right">0.0534</td>
</tr>
<tr class="even">
<td align="left">KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS</td>
<td align="right">0.0003</td>
<td align="right">0.0035</td>
<td align="right">0.0565</td>
</tr>
<tr class="odd">
<td align="left">KEGG_RETINOL_METABOLISM</td>
<td align="right">0.0005</td>
<td align="right">0.0056</td>
<td align="right">0.0960</td>
</tr>
</tbody>
</table>
</div>
<div id="go-gene-sets-1" class="section level2">
<h2>GO Gene Sets</h2>
<pre class="r"><code>goFry &lt;- fit$fitted.values %&gt;%
  cpm(log = TRUE) %&gt;%
  fry(
    index = goByID,
    design = fit$design,
    contrast = &quot;mutant&quot;,
    sort = &quot;directional&quot;
    ) %&gt;%
  rownames_to_column(&quot;gs_name&quot;) %&gt;%
  as_tibble()</code></pre>
<p>For analysis under <code>camera</code> inter-gene correlations were directly calculated for a more conservative result.</p>
<pre class="r"><code>goCamera &lt;- fit$fitted.values %&gt;%
  cpm(log = TRUE) %&gt;%
  camera(
    index = goByID,
    design = fit$design,
    contrast = &quot;mutant&quot;,
    inter.gene.cor = NULL
    ) %&gt;%
  rownames_to_column(&quot;gs_name&quot;) %&gt;%
  as_tibble()</code></pre>
<p>For generation of the GSEA ranked list, 100,000 permutations were conducted.</p>
<pre class="r"><code>goGsea &lt;- fgsea(
  pathways = goByID, 
  stats = rnk,
  nperm = np
) %&gt;%
  as_tibble() %&gt;%
  dplyr::rename(gs_name = pathway, PValue = pval) %&gt;%
  arrange(PValue)</code></pre>
<p>Results for all analyses, including goseq were then combined using Wilkinson’s method to combine p-values. For a conservative approach, under <span class="math inline">\(m\)</span> tests, the <span class="math inline">\(m - 1^{\text{th}}\)</span> smallest p-value was chosen.</p>
<pre class="r"><code>goMeta &lt;- goFry %&gt;%
  dplyr::select(gs_name, fry = PValue) %&gt;%
  left_join(
    dplyr::select(goCamera, gs_name, camera = PValue)
  ) %&gt;%
  left_join(
    dplyr::select(goGsea, gs_name, gsea = PValue)
  ) %&gt;%
  left_join(
    dplyr::select(goRiboGoseq, gs_name, goseq = PValue)
  ) %&gt;%
  pivot_longer(
    cols = one_of(c(&quot;fry&quot;, &quot;camera&quot;, &quot;gsea&quot;, &quot;goseq&quot;)),
    names_to = &quot;method&quot;,
    values_to = &quot;p&quot;
  ) %&gt;%
  dplyr::filter(!is.na(p)) %&gt;%
  group_by(gs_name) %&gt;%
  summarise(
    n = n(),
    p = wilkinsonp(p, r = n - 1)$p
  ) %&gt;%
  arrange(p) %&gt;%
  mutate(
    FDR = p.adjust(p, &quot;fdr&quot;),
    adjP = p.adjust(p, &quot;bonf&quot;)
  ) </code></pre>
<pre class="r"><code>goMeta %&gt;%
  dplyr::filter(adjP &lt; 0.01) %&gt;%
  mutate_at(vars(one_of(c(&quot;p&quot;, &quot;FDR&quot;, &quot;adjP&quot;))), formatP) %&gt;%
  dplyr::select(`Gene Set` = gs_name, p, `p~FDR~` = FDR, `p~bonf~` = adjP) %&gt;%  
  pander(
    caption = &quot;Results from combining all above approaches for the Hallmark Gene Sets. All terms are significant using a Bonferroni-adjusted p-value &lt; 0.01.&quot;,
    justify = &quot;lrrr&quot;
  )</code></pre>
<table style="width:100%;">
<caption>Results from combining all above approaches for the Hallmark Gene Sets. All terms are significant using a Bonferroni-adjusted p-value &lt; 0.01.</caption>
<colgroup>
<col width="76%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Gene Set</th>
<th align="right">p</th>
<th align="right">p<sub>FDR</sub></th>
<th align="right">p<sub>bonf</sub></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT</td>
<td align="right">4.88e-14</td>
<td align="right">5.38e-11</td>
<td align="right">4.31e-10</td>
</tr>
<tr class="even">
<td align="left">GO_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE</td>
<td align="right">5.46e-14</td>
<td align="right">5.38e-11</td>
<td align="right">4.82e-10</td>
</tr>
<tr class="odd">
<td align="left">GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM</td>
<td align="right">5.64e-14</td>
<td align="right">5.38e-11</td>
<td align="right">4.98e-10</td>
</tr>
<tr class="even">
<td align="left">GO_CYTOSOLIC_RIBOSOME</td>
<td align="right">5.64e-14</td>
<td align="right">5.38e-11</td>
<td align="right">4.98e-10</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM</td>
<td align="right">6.04e-14</td>
<td align="right">5.38e-11</td>
<td align="right">5.33e-10</td>
</tr>
<tr class="even">
<td align="left">GO_VIRAL_GENE_EXPRESSION</td>
<td align="right">6.62e-14</td>
<td align="right">5.38e-11</td>
<td align="right">5.85e-10</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEIN_TARGETING_TO_MEMBRANE</td>
<td align="right">6.63e-14</td>
<td align="right">5.38e-11</td>
<td align="right">5.85e-10</td>
</tr>
<tr class="even">
<td align="left">GO_TRANSLATIONAL_INITIATION</td>
<td align="right">6.70e-14</td>
<td align="right">5.38e-11</td>
<td align="right">5.92e-10</td>
</tr>
<tr class="odd">
<td align="left">GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS</td>
<td align="right">6.79e-14</td>
<td align="right">5.38e-11</td>
<td align="right">6.00e-10</td>
</tr>
<tr class="even">
<td align="left">GO_RIBOSOMAL_SUBUNIT</td>
<td align="right">6.81e-14</td>
<td align="right">5.38e-11</td>
<td align="right">6.01e-10</td>
</tr>
<tr class="odd">
<td align="left">GO_CYTOSOLIC_PART</td>
<td align="right">7.30e-14</td>
<td align="right">5.38e-11</td>
<td align="right">6.45e-10</td>
</tr>
<tr class="even">
<td align="left">GO_RIBOSOME</td>
<td align="right">7.30e-14</td>
<td align="right">5.38e-11</td>
<td align="right">6.45e-10</td>
</tr>
<tr class="odd">
<td align="left">GO_LARGE_RIBOSOMAL_SUBUNIT</td>
<td align="right">8.71e-12</td>
<td align="right">5.92e-09</td>
<td align="right">7.70e-08</td>
</tr>
<tr class="even">
<td align="left">GO_RNA_CATABOLIC_PROCESS</td>
<td align="right">3.70e-11</td>
<td align="right">2.33e-08</td>
<td align="right">3.27e-07</td>
</tr>
<tr class="odd">
<td align="left">GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_MEMBRANE</td>
<td align="right">4.55e-11</td>
<td align="right">2.68e-08</td>
<td align="right">4.02e-07</td>
</tr>
<tr class="even">
<td align="left">GO_CYTOSOLIC_SMALL_RIBOSOMAL_SUBUNIT</td>
<td align="right">6.56e-11</td>
<td align="right">3.62e-08</td>
<td align="right">5.80e-07</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEIN_TARGETING</td>
<td align="right">2.20e-10</td>
<td align="right">1.14e-07</td>
<td align="right">1.94e-06</td>
</tr>
<tr class="even">
<td align="left">GO_RRNA_BINDING</td>
<td align="right">3.48e-10</td>
<td align="right">1.71e-07</td>
<td align="right">3.07e-06</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTON_EXPORTING_ATPASE_ACTIVITY</td>
<td align="right">4.98e-09</td>
<td align="right">2.27e-06</td>
<td align="right">4.40e-05</td>
</tr>
<tr class="even">
<td align="left">GO_SYMPORTER_ACTIVITY</td>
<td align="right">5.13e-09</td>
<td align="right">2.27e-06</td>
<td align="right">4.53e-05</td>
</tr>
<tr class="odd">
<td align="left">GO_DRUG_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">1.08e-08</td>
<td align="right">4.53e-06</td>
<td align="right">9.52e-05</td>
</tr>
<tr class="even">
<td align="left">GO_PHAGOSOME_ACIDIFICATION</td>
<td align="right">1.52e-08</td>
<td align="right">6.10e-06</td>
<td align="right">0.0001</td>
</tr>
<tr class="odd">
<td align="left">GO_ORGANIC_ACID_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">1.81e-08</td>
<td align="right">6.96e-06</td>
<td align="right">0.0002</td>
</tr>
<tr class="even">
<td align="left">GO_ORGANIC_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">2.33e-08</td>
<td align="right">8.58e-06</td>
<td align="right">0.0002</td>
</tr>
<tr class="odd">
<td align="left">GO_TRABECULA_FORMATION</td>
<td align="right">2.79e-08</td>
<td align="right">9.86e-06</td>
<td align="right">0.0002</td>
</tr>
<tr class="even">
<td align="left">GO_ORGANIC_CYCLIC_COMPOUND_CATABOLIC_PROCESS</td>
<td align="right">3.73e-08</td>
<td align="right">1.27e-05</td>
<td align="right">0.0003</td>
</tr>
<tr class="odd">
<td align="left">GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ORGANELLE</td>
<td align="right">7.96e-08</td>
<td align="right">2.60e-05</td>
<td align="right">0.0007</td>
</tr>
<tr class="even">
<td align="left">GO_ORGANIC_ACID_TRANSMEMBRANE_TRANSPORT</td>
<td align="right">8.97e-08</td>
<td align="right">2.83e-05</td>
<td align="right">0.0008</td>
</tr>
<tr class="odd">
<td align="left">GO_NEGATIVE_REGULATION_OF_INSULIN_SECRETION_INVOLVED_IN_CELLULAR_RESPONSE_TO_GLUCOSE_STIMULUS</td>
<td align="right">9.82e-08</td>
<td align="right">2.99e-05</td>
<td align="right">0.0009</td>
</tr>
<tr class="even">
<td align="left">GO_PROTON_TRANSPORTING_V_TYPE_ATPASE_COMPLEX</td>
<td align="right">1.36e-07</td>
<td align="right">3.99e-05</td>
<td align="right">0.0012</td>
</tr>
<tr class="odd">
<td align="left">GO_PYRUVATE_DEHYDROGENASE_COMPLEX</td>
<td align="right">1.60e-07</td>
<td align="right">4.55e-05</td>
<td align="right">0.0014</td>
</tr>
<tr class="even">
<td align="left">GO_SECONDARY_ACTIVE_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">1.87e-07</td>
<td align="right">5.15e-05</td>
<td align="right">0.0016</td>
</tr>
<tr class="odd">
<td align="left">GO_RIBOSOME_ASSEMBLY</td>
<td align="right">1.97e-07</td>
<td align="right">5.24e-05</td>
<td align="right">0.0017</td>
</tr>
<tr class="even">
<td align="left">GO_DRUG_TRANSMEMBRANE_TRANSPORT</td>
<td align="right">2.02e-07</td>
<td align="right">5.24e-05</td>
<td align="right">0.0018</td>
</tr>
<tr class="odd">
<td align="left">GO_POLYSOMAL_RIBOSOME</td>
<td align="right">2.58e-07</td>
<td align="right">6.42e-05</td>
<td align="right">0.0023</td>
</tr>
<tr class="even">
<td align="left">GO_L_AMINO_ACID_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">2.62e-07</td>
<td align="right">6.42e-05</td>
<td align="right">0.0023</td>
</tr>
<tr class="odd">
<td align="left">GO_AEROBIC_ELECTRON_TRANSPORT_CHAIN</td>
<td align="right">2.99e-07</td>
<td align="right">7.14e-05</td>
<td align="right">0.0026</td>
</tr>
<tr class="even">
<td align="left">GO_CYTOPLASMIC_TRANSLATION</td>
<td align="right">3.11e-07</td>
<td align="right">7.24e-05</td>
<td align="right">0.0028</td>
</tr>
<tr class="odd">
<td align="left">GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX</td>
<td align="right">3.77e-07</td>
<td align="right">8.55e-05</td>
<td align="right">0.0033</td>
</tr>
<tr class="even">
<td align="left">GO_SMALL_RIBOSOMAL_SUBUNIT</td>
<td align="right">3.93e-07</td>
<td align="right">8.68e-05</td>
<td align="right">0.0035</td>
</tr>
<tr class="odd">
<td align="left">GO_ACTIVE_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">4.19e-07</td>
<td align="right">9.03e-05</td>
<td align="right">0.0037</td>
</tr>
<tr class="even">
<td align="left">GO_UBIQUITIN_LIGASE_INHIBITOR_ACTIVITY</td>
<td align="right">6.91e-07</td>
<td align="right">0.0001</td>
<td align="right">0.0061</td>
</tr>
<tr class="odd">
<td align="left">GO_AMINO_ACID_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">7.80e-07</td>
<td align="right">0.0002</td>
<td align="right">0.0069</td>
</tr>
<tr class="even">
<td align="left">GO_GONADAL_MESODERM_DEVELOPMENT</td>
<td align="right">7.82e-07</td>
<td align="right">0.0002</td>
<td align="right">0.0069</td>
</tr>
<tr class="odd">
<td align="left">GO_PEPTIDE_BIOSYNTHETIC_PROCESS</td>
<td align="right">9.11e-07</td>
<td align="right">0.0002</td>
<td align="right">0.0081</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>devtools::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.2 (2019-12-12)
 os       Ubuntu 18.04.4 LTS          
 system   x86_64, linux-gnu           
 ui       X11                         
 language en_AU:en                    
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Adelaide          
 date     2020-02-17                  

─ Packages ───────────────────────────────────────────────────────────────────
 package              * version    date       lib source        
 AnnotationDbi        * 1.48.0     2019-10-29 [2] Bioconductor  
 askpass                1.1        2019-01-13 [2] CRAN (R 3.6.0)
 assertthat             0.2.1      2019-03-21 [2] CRAN (R 3.6.0)
 backports              1.1.5      2019-10-02 [2] CRAN (R 3.6.1)
 BiasedUrn            * 1.07       2015-12-28 [2] CRAN (R 3.6.1)
 bibtex                 0.4.2.2    2020-01-02 [2] CRAN (R 3.6.2)
 Biobase              * 2.46.0     2019-10-29 [2] Bioconductor  
 BiocFileCache          1.10.2     2019-11-08 [2] Bioconductor  
 BiocGenerics         * 0.32.0     2019-10-29 [2] Bioconductor  
 BiocParallel           1.20.1     2019-12-21 [2] Bioconductor  
 biomaRt                2.42.0     2019-10-29 [2] Bioconductor  
 Biostrings             2.54.0     2019-10-29 [2] Bioconductor  
 bit                    1.1-15.1   2020-01-14 [2] CRAN (R 3.6.2)
 bit64                  0.9-7      2017-05-08 [2] CRAN (R 3.6.0)
 bitops                 1.0-6      2013-08-17 [2] CRAN (R 3.6.0)
 blob                   1.2.1      2020-01-20 [2] CRAN (R 3.6.2)
 broom                  0.5.4      2020-01-27 [2] CRAN (R 3.6.2)
 callr                  3.4.1      2020-01-24 [2] CRAN (R 3.6.2)
 cellranger             1.1.0      2016-07-27 [2] CRAN (R 3.6.0)
 cli                    2.0.1      2020-01-08 [2] CRAN (R 3.6.2)
 cluster                2.1.0      2019-06-19 [2] CRAN (R 3.6.1)
 codetools              0.2-16     2018-12-24 [4] CRAN (R 3.6.0)
 colorspace             1.4-1      2019-03-18 [2] CRAN (R 3.6.0)
 crayon                 1.3.4      2017-09-16 [2] CRAN (R 3.6.0)
 curl                   4.3        2019-12-02 [2] CRAN (R 3.6.2)
 data.table             1.12.8     2019-12-09 [2] CRAN (R 3.6.2)
 DBI                    1.1.0      2019-12-15 [2] CRAN (R 3.6.2)
 dbplyr                 1.4.2      2019-06-17 [2] CRAN (R 3.6.0)
 DelayedArray           0.12.2     2020-01-06 [2] Bioconductor  
 desc                   1.2.0      2018-05-01 [2] CRAN (R 3.6.0)
 devtools               2.2.1      2019-09-24 [2] CRAN (R 3.6.1)
 digest                 0.6.23     2019-11-23 [2] CRAN (R 3.6.1)
 dplyr                * 0.8.4      2020-01-31 [2] CRAN (R 3.6.2)
 edgeR                * 3.28.0     2019-10-29 [2] Bioconductor  
 ellipsis               0.3.0      2019-09-20 [2] CRAN (R 3.6.1)
 evaluate               0.14       2019-05-28 [2] CRAN (R 3.6.0)
 FactoMineR             2.1        2020-01-17 [2] CRAN (R 3.6.2)
 fansi                  0.4.1      2020-01-08 [2] CRAN (R 3.6.2)
 fastmatch              1.1-0      2017-01-28 [2] CRAN (R 3.6.0)
 fgsea                * 1.12.0     2019-10-29 [2] Bioconductor  
 flashClust             1.01-2     2012-08-21 [2] CRAN (R 3.6.1)
 forcats              * 0.4.0      2019-02-17 [2] CRAN (R 3.6.0)
 fs                     1.3.1      2019-05-06 [2] CRAN (R 3.6.0)
 gbRd                   0.4-11     2012-10-01 [2] CRAN (R 3.6.0)
 geneLenDataBase      * 1.22.0     2019-11-05 [2] Bioconductor  
 generics               0.0.2      2018-11-29 [2] CRAN (R 3.6.0)
 GenomeInfoDb           1.22.0     2019-10-29 [2] Bioconductor  
 GenomeInfoDbData       1.2.2      2019-11-21 [2] Bioconductor  
 GenomicAlignments      1.22.1     2019-11-12 [2] Bioconductor  
 GenomicFeatures        1.38.1     2020-01-22 [2] Bioconductor  
 GenomicRanges          1.38.0     2019-10-29 [2] Bioconductor  
 ggdendro               0.1-20     2016-04-27 [2] CRAN (R 3.6.0)
 ggplot2              * 3.2.1      2019-08-10 [2] CRAN (R 3.6.1)
 ggrepel                0.8.1      2019-05-07 [2] CRAN (R 3.6.0)
 git2r                  0.26.1     2019-06-29 [2] CRAN (R 3.6.1)
 glue                   1.3.1      2019-03-12 [2] CRAN (R 3.6.0)
 GO.db                  3.10.0     2019-11-21 [2] Bioconductor  
 goseq                * 1.38.0     2019-10-29 [2] Bioconductor  
 gridExtra              2.3        2017-09-09 [2] CRAN (R 3.6.0)
 gtable                 0.3.0      2019-03-25 [2] CRAN (R 3.6.0)
 haven                  2.2.0      2019-11-08 [2] CRAN (R 3.6.1)
 highr                  0.8        2019-03-20 [2] CRAN (R 3.6.0)
 hms                    0.5.3      2020-01-08 [2] CRAN (R 3.6.2)
 htmltools              0.4.0      2019-10-04 [2] CRAN (R 3.6.1)
 htmlwidgets            1.5.1      2019-10-08 [2] CRAN (R 3.6.1)
 httpuv                 1.5.2      2019-09-11 [2] CRAN (R 3.6.1)
 httr                   1.4.1      2019-08-05 [2] CRAN (R 3.6.1)
 hwriter                1.3.2      2014-09-10 [2] CRAN (R 3.6.0)
 IRanges              * 2.20.2     2020-01-13 [2] Bioconductor  
 jpeg                   0.1-8.1    2019-10-24 [2] CRAN (R 3.6.1)
 jsonlite               1.6        2018-12-07 [2] CRAN (R 3.6.0)
 kableExtra             1.1.0      2019-03-16 [2] CRAN (R 3.6.1)
 knitr                  1.27       2020-01-16 [2] CRAN (R 3.6.2)
 later                  1.0.0      2019-10-04 [2] CRAN (R 3.6.1)
 lattice                0.20-38    2018-11-04 [4] CRAN (R 3.6.0)
 latticeExtra           0.6-29     2019-12-19 [2] CRAN (R 3.6.2)
 lazyeval               0.2.2      2019-03-15 [2] CRAN (R 3.6.0)
 leaps                  3.1        2020-01-16 [2] CRAN (R 3.6.2)
 lifecycle              0.1.0      2019-08-01 [2] CRAN (R 3.6.1)
 limma                * 3.42.1     2020-01-29 [2] Bioconductor  
 locfit                 1.5-9.1    2013-04-20 [2] CRAN (R 3.6.0)
 lubridate              1.7.4      2018-04-11 [2] CRAN (R 3.6.0)
 magrittr             * 1.5        2014-11-22 [2] CRAN (R 3.6.0)
 MASS                   7.3-51.5   2019-12-20 [4] CRAN (R 3.6.2)
 Matrix                 1.2-18     2019-11-27 [2] CRAN (R 3.6.1)
 matrixStats            0.55.0     2019-09-07 [2] CRAN (R 3.6.1)
 memoise                1.1.0      2017-04-21 [2] CRAN (R 3.6.0)
 metap                * 1.3        2020-01-23 [2] CRAN (R 3.6.2)
 mgcv                   1.8-31     2019-11-09 [4] CRAN (R 3.6.1)
 mnormt                 1.5-5      2016-10-15 [2] CRAN (R 3.6.1)
 modelr                 0.1.5      2019-08-08 [2] CRAN (R 3.6.1)
 msigdbr              * 7.0.1      2019-09-04 [2] CRAN (R 3.6.2)
 multcomp               1.4-12     2020-01-10 [2] CRAN (R 3.6.2)
 multtest               2.42.0     2019-10-29 [2] Bioconductor  
 munsell                0.5.0      2018-06-12 [2] CRAN (R 3.6.0)
 mutoss                 0.1-12     2017-12-04 [2] CRAN (R 3.6.2)
 mvtnorm                1.0-12     2020-01-09 [2] CRAN (R 3.6.2)
 ngsReports           * 1.1.2      2019-10-16 [1] Bioconductor  
 nlme                   3.1-144    2020-02-06 [4] CRAN (R 3.6.2)
 numDeriv               2016.8-1.1 2019-06-06 [2] CRAN (R 3.6.2)
 openssl                1.4.1      2019-07-18 [2] CRAN (R 3.6.1)
 pander               * 0.6.3      2018-11-06 [2] CRAN (R 3.6.0)
 pillar                 1.4.3      2019-12-20 [2] CRAN (R 3.6.2)
 pkgbuild               1.0.6      2019-10-09 [2] CRAN (R 3.6.1)
 pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 3.6.1)
 pkgload                1.0.2      2018-10-29 [2] CRAN (R 3.6.0)
 plotly                 4.9.1      2019-11-07 [2] CRAN (R 3.6.1)
 plotrix                3.7-7      2019-12-05 [2] CRAN (R 3.6.2)
 plyr                   1.8.5      2019-12-10 [2] CRAN (R 3.6.2)
 png                    0.1-7      2013-12-03 [2] CRAN (R 3.6.0)
 prettyunits            1.1.1      2020-01-24 [2] CRAN (R 3.6.2)
 processx               3.4.1      2019-07-18 [2] CRAN (R 3.6.1)
 progress               1.2.2      2019-05-16 [2] CRAN (R 3.6.0)
 promises               1.1.0      2019-10-04 [2] CRAN (R 3.6.1)
 ps                     1.3.0      2018-12-21 [2] CRAN (R 3.6.0)
 purrr                * 0.3.3      2019-10-18 [2] CRAN (R 3.6.1)
 R6                     2.4.1      2019-11-12 [2] CRAN (R 3.6.1)
 rappdirs               0.3.1      2016-03-28 [2] CRAN (R 3.6.0)
 RColorBrewer         * 1.1-2      2014-12-07 [2] CRAN (R 3.6.0)
 Rcpp                 * 1.0.3      2019-11-08 [2] CRAN (R 3.6.1)
 RCurl                  1.98-1.1   2020-01-19 [2] CRAN (R 3.6.2)
 Rdpack                 0.11-1     2019-12-14 [2] CRAN (R 3.6.2)
 readr                * 1.3.1      2018-12-21 [2] CRAN (R 3.6.0)
 readxl                 1.3.1      2019-03-13 [2] CRAN (R 3.6.0)
 remotes                2.1.0      2019-06-24 [2] CRAN (R 3.6.0)
 reprex                 0.3.0      2019-05-16 [2] CRAN (R 3.6.0)
 reshape2               1.4.3      2017-12-11 [2] CRAN (R 3.6.0)
 rlang                  0.4.4      2020-01-28 [2] CRAN (R 3.6.2)
 rmarkdown              2.1        2020-01-20 [2] CRAN (R 3.6.2)
 rprojroot              1.3-2      2018-01-03 [2] CRAN (R 3.6.0)
 Rsamtools              2.2.1      2019-11-06 [2] Bioconductor  
 RSQLite                2.2.0      2020-01-07 [2] CRAN (R 3.6.2)
 rstudioapi             0.10       2019-03-19 [2] CRAN (R 3.6.0)
 rtracklayer            1.46.0     2019-10-29 [2] Bioconductor  
 rvest                  0.3.5      2019-11-08 [2] CRAN (R 3.6.1)
 S4Vectors            * 0.24.3     2020-01-18 [2] Bioconductor  
 sandwich               2.5-1      2019-04-06 [2] CRAN (R 3.6.1)
 scales               * 1.1.0      2019-11-18 [2] CRAN (R 3.6.1)
 scatterplot3d          0.3-41     2018-03-14 [2] CRAN (R 3.6.1)
 sessioninfo            1.1.1      2018-11-05 [2] CRAN (R 3.6.0)
 ShortRead              1.44.1     2019-12-19 [2] Bioconductor  
 sn                     1.5-5      2020-01-30 [2] CRAN (R 3.6.2)
 statmod                1.4.33     2020-01-10 [2] CRAN (R 3.6.2)
 stringi                1.4.5      2020-01-11 [2] CRAN (R 3.6.2)
 stringr              * 1.4.0      2019-02-10 [2] CRAN (R 3.6.0)
 SummarizedExperiment   1.16.1     2019-12-19 [2] Bioconductor  
 survival               3.1-8      2019-12-03 [2] CRAN (R 3.6.2)
 testthat               2.3.1      2019-12-01 [2] CRAN (R 3.6.1)
 TFisher                0.2.0      2018-03-21 [2] CRAN (R 3.6.2)
 TH.data                1.0-10     2019-01-21 [2] CRAN (R 3.6.1)
 tibble               * 2.1.3      2019-06-06 [2] CRAN (R 3.6.0)
 tidyr                * 1.0.2      2020-01-24 [2] CRAN (R 3.6.2)
 tidyselect             1.0.0      2020-01-27 [2] CRAN (R 3.6.2)
 tidyverse            * 1.3.0      2019-11-21 [2] CRAN (R 3.6.1)
 truncnorm              1.0-8      2018-02-27 [2] CRAN (R 3.6.0)
 usethis                1.5.1      2019-07-04 [2] CRAN (R 3.6.1)
 vctrs                  0.2.2      2020-01-24 [2] CRAN (R 3.6.2)
 viridisLite            0.3.0      2018-02-01 [2] CRAN (R 3.6.0)
 webshot                0.5.2      2019-11-22 [2] CRAN (R 3.6.1)
 whisker                0.4        2019-08-28 [2] CRAN (R 3.6.1)
 withr                  2.1.2      2018-03-15 [2] CRAN (R 3.6.0)
 workflowr            * 1.6.0      2019-12-19 [2] CRAN (R 3.6.2)
 xfun                   0.12       2020-01-13 [2] CRAN (R 3.6.2)
 XML                    3.99-0.3   2020-01-20 [2] CRAN (R 3.6.2)
 xml2                   1.2.2      2019-08-09 [2] CRAN (R 3.6.1)
 XVector                0.26.0     2019-10-29 [2] Bioconductor  
 yaml                   2.2.1      2020-02-01 [2] CRAN (R 3.6.2)
 zlibbioc               1.32.0     2019-10-29 [2] Bioconductor  
 zoo                    1.8-7      2020-01-10 [2] CRAN (R 3.6.2)

[1] /home/steveped/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
