<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">20170327_Psen2S4Ter_RNASeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-03-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>20170327_Psen2S4Ter_RNASeq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200119code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200119)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200119code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomUofABioinformaticsHub20170327Psen2S4TerRNASeqtree377410a12ca13a7e233e39fb2f77f165e846d830targetblank377410aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/tree/377410a12ca13a7e233e39fb2f77f165e846d830" target="_blank">377410a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomUofABioinformaticsHub20170327Psen2S4TerRNASeqtree377410a12ca13a7e233e39fb2f77f165e846d830targetblank377410aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   output/Enrichment_Hom_V_Het.csv
    Modified:   output/Enrichment_Mutant_V_WT.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/f75f82a2155f610f0a0f0ae0d8a944411ded63a8/analysis/index.Rmd" target="_blank">f75f82a</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-03-05
</td>
<td>
Added post-CQN PCA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/86ffa07adc34510ac982dcb32159b051f50a7b9a/docs/index.html" target="_blank">86ffa07</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
<td>
Updated saample layout figure
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/4d2af23dbbdd8c6f32404a1df10f65fcf0dbb266/analysis/index.Rmd" target="_blank">4d2af23</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
<td>
Modified heatmap sizes &amp; initial network layout
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/876e40fc8b5e69e983e028d62c49f977da71a90d/docs/index.html" target="_blank">876e40f</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-17
</td>
<td>
Compiled after minor corrections
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/1032a6d0853f2cae2a88b60826c288281f7de202/docs/index.html" target="_blank">1032a6d</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Added link to HomVsHet
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/9104ecdc9ad8512ec277c6c131d36de86edcf845/analysis/index.Rmd" target="_blank">9104ecd</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
First draft of Hom Vs Het
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/468e6e3a5e47e1e25be13e92d15dc4e1d7067130/docs/index.html" target="_blank">468e6e3</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Ran enrichment of Mut Vs WT
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/3a9933c5330cee6a4957bdfdf7794251f9346dd2/analysis/index.Rmd" target="_blank">3a9933c</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-28
</td>
<td>
Finished Enrichment analysis on MutVsWt
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/b086affa08326538432202bf80cb668c0b8bd4cd/docs/index.html" target="_blank">b086aff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-26
</td>
<td>
Added first step of enrichment analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/ba189be6c3080686d32850d6ccd41d7cc76e7a74/analysis/index.Rmd" target="_blank">ba189be</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-26
</td>
<td>
Corrected packages
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/96d8cc7f67d6c50bf5f8f02407771476db931dda/docs/index.html" target="_blank">96d8cc7</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-25
</td>
<td>
Compiled after data export &amp; added compression to output
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/10a285b55e3d5e8c883f78710a0a9f9dd3a6e75a/analysis/index.Rmd" target="_blank">10a285b</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
<td>
Expanded description in index and tried reusing an image
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/10a285b55e3d5e8c883f78710a0a9f9dd3a6e75a/docs/index.html" target="_blank">10a285b</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
<td>
Expanded description in index and tried reusing an image
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/b096ad9021d5452861ab6e55d31f2b986431f597/docs/index.html" target="_blank">b096ad9</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
<td>
Recompiled index
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/01512dac59935dca1dd5e74c9c12478b0b0239eb/analysis/index.Rmd" target="_blank">01512da</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
<td>
Added initial DE analysis to index
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/fbb62422891af2a7c033ef9780b9328b6c71a203/analysis/index.Rmd" target="_blank">fbb6242</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
<td>
Paused DE analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/fbb62422891af2a7c033ef9780b9328b6c71a203/docs/index.html" target="_blank">fbb6242</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-21
</td>
<td>
Paused DE analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/f154205d1606e53258d146024302f5c190e34a7c/docs/index.html" target="_blank">f154205</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-20
</td>
<td>
Got rid of pointless warning
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/bc1210174c2f6056731bac6f3953b2b76d3e64c2/analysis/index.Rmd" target="_blank">bc12101</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-20
</td>
<td>
Added bash pipeline
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/bc1210174c2f6056731bac6f3953b2b76d3e64c2/docs/index.html" target="_blank">bc12101</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-20
</td>
<td>
Added bash pipeline
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/f65bf664e6b273af1c2c82a8823bcbe7ca0e7eca/docs/index.html" target="_blank">f65bf66</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-20
</td>
<td>
Removed License &amp; updated text for PCA
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/39d0381583a1687f9a94ec455f1d3969ae2460a1/analysis/index.Rmd" target="_blank">39d0381</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-20
</td>
<td>
Finished QC
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/39d0381583a1687f9a94ec455f1d3969ae2460a1/docs/index.html" target="_blank">39d0381</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-20
</td>
<td>
Finished QC
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/acb4f8a27a1ed8082c420ae755310cfb3230954b/analysis/index.Rmd" target="_blank">acb4f8a</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-19
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(tidygraph)
library(ggraph)</code></pre>
<pre class="r"><code>samples &lt;- read_csv(&quot;data/samples.csv&quot;) %&gt;%
    distinct(sampleName, .keep_all = TRUE) %&gt;%
    dplyr::select(sample = sampleName, sampleID, genotype) %&gt;%
    mutate(genotype = factor(genotype, levels = c(&quot;WT&quot;, &quot;Het&quot;, &quot;Hom&quot;)))</code></pre>
<p>This dataset is an analysis of RNASeq data from a 3-way comparison of WT zebrafish with Heterozygous mutants (<em>psen2<sup>S4Ter/+</sup></em>) and Homozygous mutants <em>psen2<sup>S4Ter/S4Ter</sup></em>. The <em>psen2<sup>S4Ter</sup></em> mutant allele is expected to be a premature termination of the <em>psen2</em> gene, but instead an alternate downstream start site resulted in a near full-length transcript with significant similarity to a conventional Early Onset Familial Alzhemier’s Disease mutation.</p>
<p>The expected comparisons were intended to be a three-way group comparison, however the final model was defined by fitting the wild-type samples, differences between heterozygotes and wild-type, followed by the difference between the two mutant genotypes</p>
<pre class="r"><code>create_ring(3) %&gt;% 
  mutate(
    name = paste0(levels(samples$genotype), &quot;\n(n = 4)&quot;),
    name = factor(name, levels = name)
  ) %&gt;% 
  activate(edges) %&gt;%
  mutate(comparison = c(&quot;Het Vs WT&quot;, &quot;Hom Vs Het&quot;, &quot;Hom Vs WT&quot;)) %&gt;% 
    dplyr::filter(comparison != &quot;Hom Vs WT&quot;) %&gt;%
  ggraph(layout = &quot;kk&quot;) + 
  geom_edge_link2(
    aes(label = comparison),
    angle_calc = &quot;along&quot;,
    label_dodge = unit(0.02, &quot;npc&quot;),
    start_cap = circle(0.09, &quot;npc&quot;),
    end_cap = circle(0.09, &quot;npc&quot;),
    label_size = 5,
    arrow = arrow(
      length = unit(0.06, &quot;npc&quot;),
      ends = &quot;both&quot;, 
      type = &quot;closed&quot;
    )
  ) +
  geom_node_label(
    aes(label = name, colour = name),
    size = 5,
    fill = rgb(1,1,1,0.7),
    label.padding = unit(0.4, &quot;lines&quot;)
  ) + 
  scale_y_continuous(expand = expand_scale(0.1)) +
  scale_x_continuous(expand = expand_scale(0.1)) +
  theme_void() +
  theme(
    legend.position = &quot;none&quot;
  ) </code></pre>
<p><img src="figure/index.Rmd/plotLayout-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotLayout-1">
Past versions of plotLayout-1.png
</button>
</p>
<div id="fig-plotLayout-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/86ffa07adc34510ac982dcb32159b051f50a7b9a/docs/figure/index.Rmd/plotLayout-1.png" target="_blank">86ffa07</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-02-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/UofABioinformaticsHub/20170327_Psen2S4Ter_RNASeq/blob/10a285b55e3d5e8c883f78710a0a9f9dd3a6e75a/docs/figure/index.Rmd/plotLayout-1.png" target="_blank">10a285b</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-01-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>All samples were female, 6 months old and raised in the same tank as a family, to minimise variability. This age represents sexual maturity and is expected to model a pre-symptomatic brain. Fish were all killed and genotyped by tail-clipping on 07-07-2016.</p>
<p>RNAseq was performed on total RNA (i.e. ribo-reduced) from whole brain tissue, with <span class="math inline">\(n=4\)</span> samples in each group. Sequencing was performed by the sequencing facility at the Centre for Cancer Biology in Adelaide, using an Illimuna NextSeq. Reads were provided in as paired-end, 150bp reads.</p>
<ol style="list-style-type: decimal">
<li><a href="0_bashPipeline.html">Bash Pipeline</a></li>
<li><a href="1_QC.html">Quality Assesment</a></li>
<li><a href="2_DifferentialExpression.html">Differential Expression Analysis</a></li>
<li>Enrichment Analysis
<ol style="list-style-type: lower-alpha">
<li><a href="3_Enrichment_MutantVsWT.html">Mutant Vs Wild Type</a></li>
<li><a href="3_Enrichment_HomVsHet.html">Homozygous Vs Heterzygous Mutants</a></li>
</ol></li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>devtools::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.3 (2020-02-29)
 os       Ubuntu 18.04.4 LTS          
 system   x86_64, linux-gnu           
 ui       X11                         
 language en_AU:en                    
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Melbourne         
 date     2020-03-05                  

─ Packages ───────────────────────────────────────────────────────────────────
 package      * version  date       lib source        
 assertthat     0.2.1    2019-03-21 [2] CRAN (R 3.6.0)
 backports      1.1.5    2019-10-02 [2] CRAN (R 3.6.1)
 broom          0.5.4    2020-01-27 [2] CRAN (R 3.6.2)
 callr          3.4.2    2020-02-12 [2] CRAN (R 3.6.2)
 cellranger     1.1.0    2016-07-27 [2] CRAN (R 3.6.0)
 cli            2.0.1    2020-01-08 [2] CRAN (R 3.6.2)
 colorspace     1.4-1    2019-03-18 [2] CRAN (R 3.6.0)
 crayon         1.3.4    2017-09-16 [2] CRAN (R 3.6.0)
 DBI            1.1.0    2019-12-15 [2] CRAN (R 3.6.2)
 dbplyr         1.4.2    2019-06-17 [2] CRAN (R 3.6.0)
 desc           1.2.0    2018-05-01 [2] CRAN (R 3.6.0)
 devtools       2.2.2    2020-02-17 [2] CRAN (R 3.6.2)
 digest         0.6.25   2020-02-23 [2] CRAN (R 3.6.2)
 dplyr        * 0.8.4    2020-01-31 [2] CRAN (R 3.6.2)
 ellipsis       0.3.0    2019-09-20 [2] CRAN (R 3.6.1)
 evaluate       0.14     2019-05-28 [2] CRAN (R 3.6.0)
 fansi          0.4.1    2020-01-08 [2] CRAN (R 3.6.2)
 farver         2.0.3    2020-01-16 [2] CRAN (R 3.6.2)
 forcats      * 0.4.0    2019-02-17 [2] CRAN (R 3.6.0)
 fs             1.3.1    2019-05-06 [2] CRAN (R 3.6.0)
 generics       0.0.2    2018-11-29 [2] CRAN (R 3.6.0)
 ggforce        0.3.1    2019-08-20 [2] CRAN (R 3.6.1)
 ggplot2      * 3.2.1    2019-08-10 [2] CRAN (R 3.6.1)
 ggraph       * 2.0.1    2020-02-07 [2] CRAN (R 3.6.2)
 ggrepel        0.8.1    2019-05-07 [2] CRAN (R 3.6.0)
 git2r          0.26.1   2019-06-29 [2] CRAN (R 3.6.1)
 glue           1.3.1    2019-03-12 [2] CRAN (R 3.6.0)
 graphlayouts   0.5.0    2019-08-20 [2] CRAN (R 3.6.1)
 gridExtra      2.3      2017-09-09 [2] CRAN (R 3.6.0)
 gtable         0.3.0    2019-03-25 [2] CRAN (R 3.6.0)
 haven          2.2.0    2019-11-08 [2] CRAN (R 3.6.1)
 hms            0.5.3    2020-01-08 [2] CRAN (R 3.6.2)
 htmltools      0.4.0    2019-10-04 [2] CRAN (R 3.6.1)
 httpuv         1.5.2    2019-09-11 [2] CRAN (R 3.6.1)
 httr           1.4.1    2019-08-05 [2] CRAN (R 3.6.1)
 igraph         1.2.4.2  2019-11-27 [2] CRAN (R 3.6.1)
 jsonlite       1.6.1    2020-02-02 [2] CRAN (R 3.6.2)
 knitr          1.28     2020-02-06 [2] CRAN (R 3.6.2)
 labeling       0.3      2014-08-23 [2] CRAN (R 3.6.0)
 later          1.0.0    2019-10-04 [2] CRAN (R 3.6.1)
 lattice        0.20-40  2020-02-19 [4] CRAN (R 3.6.2)
 lazyeval       0.2.2    2019-03-15 [2] CRAN (R 3.6.0)
 lifecycle      0.1.0    2019-08-01 [2] CRAN (R 3.6.1)
 lubridate      1.7.4    2018-04-11 [2] CRAN (R 3.6.0)
 magrittr       1.5      2014-11-22 [2] CRAN (R 3.6.0)
 MASS           7.3-51.5 2019-12-20 [4] CRAN (R 3.6.2)
 memoise        1.1.0    2017-04-21 [2] CRAN (R 3.6.0)
 modelr         0.1.6    2020-02-22 [2] CRAN (R 3.6.2)
 munsell        0.5.0    2018-06-12 [2] CRAN (R 3.6.0)
 nlme           3.1-144  2020-02-06 [4] CRAN (R 3.6.2)
 pillar         1.4.3    2019-12-20 [2] CRAN (R 3.6.2)
 pkgbuild       1.0.6    2019-10-09 [2] CRAN (R 3.6.1)
 pkgconfig      2.0.3    2019-09-22 [2] CRAN (R 3.6.1)
 pkgload        1.0.2    2018-10-29 [2] CRAN (R 3.6.0)
 polyclip       1.10-0   2019-03-14 [2] CRAN (R 3.6.0)
 prettyunits    1.1.1    2020-01-24 [2] CRAN (R 3.6.2)
 processx       3.4.2    2020-02-09 [2] CRAN (R 3.6.2)
 promises       1.1.0    2019-10-04 [2] CRAN (R 3.6.1)
 ps             1.3.2    2020-02-13 [2] CRAN (R 3.6.2)
 purrr        * 0.3.3    2019-10-18 [2] CRAN (R 3.6.1)
 R6             2.4.1    2019-11-12 [2] CRAN (R 3.6.1)
 Rcpp           1.0.3    2019-11-08 [2] CRAN (R 3.6.1)
 readr        * 1.3.1    2018-12-21 [2] CRAN (R 3.6.0)
 readxl         1.3.1    2019-03-13 [2] CRAN (R 3.6.0)
 remotes        2.1.1    2020-02-15 [2] CRAN (R 3.6.2)
 reprex         0.3.0    2019-05-16 [2] CRAN (R 3.6.0)
 rlang          0.4.4    2020-01-28 [2] CRAN (R 3.6.2)
 rmarkdown      2.1      2020-01-20 [2] CRAN (R 3.6.2)
 rprojroot      1.3-2    2018-01-03 [2] CRAN (R 3.6.0)
 rstudioapi     0.11     2020-02-07 [2] CRAN (R 3.6.2)
 rvest          0.3.5    2019-11-08 [2] CRAN (R 3.6.1)
 scales         1.1.0    2019-11-18 [2] CRAN (R 3.6.1)
 sessioninfo    1.1.1    2018-11-05 [2] CRAN (R 3.6.0)
 stringi        1.4.6    2020-02-17 [2] CRAN (R 3.6.2)
 stringr      * 1.4.0    2019-02-10 [2] CRAN (R 3.6.0)
 testthat       2.3.1    2019-12-01 [2] CRAN (R 3.6.1)
 tibble       * 2.1.3    2019-06-06 [2] CRAN (R 3.6.0)
 tidygraph    * 1.1.2    2019-02-18 [2] CRAN (R 3.6.0)
 tidyr        * 1.0.2    2020-01-24 [2] CRAN (R 3.6.2)
 tidyselect     1.0.0    2020-01-27 [2] CRAN (R 3.6.2)
 tidyverse    * 1.3.0    2019-11-21 [2] CRAN (R 3.6.1)
 tweenr         1.0.1    2018-12-14 [2] CRAN (R 3.6.0)
 usethis        1.5.1    2019-07-04 [2] CRAN (R 3.6.1)
 vctrs          0.2.3    2020-02-20 [2] CRAN (R 3.6.2)
 viridis        0.5.1    2018-03-29 [2] CRAN (R 3.6.0)
 viridisLite    0.3.0    2018-02-01 [2] CRAN (R 3.6.0)
 whisker        0.4      2019-08-28 [2] CRAN (R 3.6.1)
 withr          2.1.2    2018-03-15 [2] CRAN (R 3.6.0)
 workflowr    * 1.6.0    2019-12-19 [2] CRAN (R 3.6.2)
 xfun           0.12     2020-01-13 [2] CRAN (R 3.6.2)
 xml2           1.2.2    2019-08-09 [2] CRAN (R 3.6.1)
 yaml           2.2.1    2020-02-01 [2] CRAN (R 3.6.2)

[1] /home/steveped/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library</code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
