---
title: "Differential Expression"
author: "Steve Pederson"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  autodep = TRUE,
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = "center"
)
```

# Setup

```{r loadPackages}
library(ngsReports)
library(tidyverse)
library(magrittr)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(scales)
library(pander)
library(cqn)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
```

```{r setOptions}
if (interactive()) setwd(here::here())
theme_set(theme_bw())
panderOptions("big.mark", ",")
panderOptions("table.split.table", Inf)
panderOptions("table.style", "rmarkdown")
```

## Annotations

```{r annotationSetup}
ah <- AnnotationHub() %>%
  subset(species == "Danio rerio") %>%
  subset(rdataclass == "EnsDb")
ensDb <- ah[["AH74989"]]
```

```{r transAnnotation}
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
  width() %>%
  vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]
```


```{r geneAnnotation}
gcGene <- grTrans %>%
  mcols() %>%
  as.data.frame() %>%
  dplyr::select(gene_id, tx_id, gc_content, length) %>%
  as_tibble() %>%
  group_by(gene_id) %>%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )
grGenes <- genes(ensDb)
mcols(grGenes) %<>%
  as.data.frame() %>%
  left_join(gcGene) %>%
  as.data.frame() %>%
  DataFrame()
```

Similarly to the Quality Assessment steps, `GRanges` objects were formed at the gene and transcript levels, to enable estimation of GC content and length for each transcript and gene.
GC content and transcript length are available for each transcript, and for gene-level estimates, GC content was taken as the sum of all GC bases divided by the sum of all transcript lengths, effectively averaging across all transcripts.
Gene length was defined as the median transcript length.


```{r samplesAndLabels}
samples <- read_csv("data/samples.csv") %>%
  distinct(sampleName, .keep_all = TRUE) %>%
  dplyr::select(sample = sampleName, sampleID, genotype) %>%
  mutate(
    genotype = factor(genotype, levels = c("WT", "Het", "Hom")),
    mutant = genotype %in% c("Het", "Hom"),
    homozygous = genotype == "Hom"
  )
genoCols <- samples$genotype %>%
  levels() %>%
  length() %>%
  brewer.pal("Set1") %>%
  setNames(levels(samples$genotype))
```

Sample metadata was also loaded, with only the sampleID and genotype being retained. 
All other fields were considered irrelevant.

## Count Data

```{r dgeList}
minCPM <- 1.5
minSamples <- 4
dgeList <- file.path("data", "2_alignedData", "featureCounts", "genes.out") %>%
  read_delim(delim = "\t") %>%
  set_names(basename(names(.))) %>%
  as.data.frame() %>%
  column_to_rownames("Geneid") %>%
  as.matrix() %>% 
  set_colnames(str_remove(colnames(.), "Aligned.sortedByCoord.out.bam")) %>%
  .[rowSums(cpm(.) >= minCPM) >= minSamples,] %>%
  DGEList(
    samples = tibble(sample = colnames(.)) %>%
      left_join(samples),
    genes = grGenes[rownames(.)] %>%
      as.data.frame() %>%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, 
        entrezid, gc_content, length
      )
  ) %>%
  .[!grepl("rRNA", .$genes$gene_biotype),] %>%
  calcNormFactors()
```

Gene-level count data as output by `featureCounts`, was loaded and formed into a `DGEList` object.
During this process, genes were removed if:

- They were not considered as detectable (CPM < `r minCPM` in > `r ncol(dgeList) - minSamples` samples). This translates to > `r ceiling(min(minCPM * dgeList$samples$lib.size/1e6))` reads assigned a gene in all samples from one or more of the genotype groups
- The `gene_biotype` was any type of `rRNA`.

These filtering steps returned gene-level counts for `r comma(nrow(dgeList))` genes, with total library sizes between `r pander(comma(range(dgeList$samples$lib.size)))` reads assigned to genes.
It was noted that these library sizes were about 1.5-fold larger than the transcript-level counts used for the QA steps.

```{r plotDensities, fig.width=5, fig.height=4, fig.cap="*Expression density plots for all samples after filtering, showing logCPM values.*"}
cpm(dgeList, log = TRUE) %>%
  as.data.frame() %>%
  pivot_longer(
    cols = everything(),
    names_to = "sample",
    values_to = "logCPM"
  ) %>%
  split(f = .$sample) %>%
  lapply(function(x){
    d <- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %>%
  bind_rows() %>%
  left_join(samples) %>%
  ggplot(aes(x, y, colour = genotype, group = sample)) +
  geom_line() +
  scale_colour_manual(
    values = genoCols
  ) +
  labs(
    x = "logCPM",
    y = "Density",
    colour = "Genotype"
  )
```

### Additional Functions

```{r labellers}
contLabeller <- as_labeller(
  c(
    HetVsWT = "S4Ter/+ Vs +/+",
    HomVsWT = "S4Ter/S4Ter Vs +/+",
    HomVsHet = "S4Ter/S4Ter Vs S4Ter/+",
    Hom = "S4Ter/S4Ter",
    Het = "S4Ter/+",
    WT = "+/+",
    mutant = "S4Ter Vs WT",
    homozygous = "S4Ter/S4Ter Vs S4Ter/+"
  )
)
geneLabeller <- structure(grGenes$gene_name, names = grGenes$gene_id) %>%
  as_labeller()
```

Labeller functions for genotypes, contrasts and gene names were additionally defined for simpler plotting using `ggplot2`.

## Correlations with rRNA contaminants

In order to asses the impact of the potential rRNA contaminant, correlations between the initial estimate of contamination and each gene's expression pattern were also investigated.
Conceptually, genes which are highly correlated with rRNA may be physically associated with these complexes.
Similary, genes with high negative correlations may be actively selected against in the presence of rRNA complexes.

```{r rawFqc}
rawFqc <- list.files(
  path = "data/0_rawData/FastQC/",
  pattern = "zip",
  full.names = TRUE
) %>%
  FastqcDataList()
gc <- getModule(rawFqc, "Per_sequence_GC") %>%
  group_by(Filename) %>% 
  mutate(Freq = Count / sum(Count)) %>%
  ungroup()
rawGC <- gc %>%
  dplyr::filter(GC_Content > 70) %>% 
  group_by(Filename) %>%
  summarise(Freq = sum(Freq)) %>% 
  arrange(desc(Freq)) %>%
  mutate(sample = str_remove(Filename, "_R[12].fastq.gz")) %>%
  group_by(sample) %>%
  summarise(Freq = mean(Freq)) %>%
  left_join(samples) 
```


```{r riboCors}
riboVec <- structure(rawGC$Freq, names = rawGC$sample)
riboCors <- cpm(dgeList, log = TRUE) %>%
  apply(1, function(x){
    cor(x, riboVec[names(x)])
  })
```

# Analysis

## Initial PCA

```{r pca}
pca <- dgeList %>%
  cpm(log = TRUE) %>%
  t() %>%
  prcomp() 
pcaVars <- percent_format(0.1)(summary(pca)$importance["Proportion of Variance",])
```


```{r plotPCA, fig.width=5, fig.height=4, fig.cap="*PCA of gene-level counts.*"}
pca$x %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(samples) %>%
  as_tibble() %>%
  ggplot(aes(PC1, PC2, colour = genotype, fill = genotype)) +
  geom_point() +
  geom_text_repel(aes(label = sampleID), show.legend = FALSE) +
  stat_ellipse(geom = "polygon", alpha = 0.05, show.legend = FALSE) +
  guides(fill = FALSE) +
  scale_colour_manual(
    values = genoCols
  ) +
  labs(
    x = paste0("PC1 (", pcaVars[["PC1"]], ")"),
    y = paste0("PC2 (", pcaVars[["PC2"]], ")"),
    colour = "Genotype"
  )
```

A Principal Component Analysis (PCA) was also performed using logCPM values from each sample.
Both mutant genotypes appear to cluster together, however it has previously been noted that GC content appears to track closely with PC1, as a result of variable rRNA removal.

## Model Description

```{r plotLayout, echo=FALSE, out.width='50%'}
here::here("docs/figure/index.Rmd/plotLayout-1.png") %>%
  knitr::include_graphics()
```

The initial expectation was to perform three pairwise comparisons as described in the above figure.
However, given there was a strong similarity between mutants, the model matrix was defined as containing an intercept (i.e. baseline expression in wild-type), with additional columns defining presence of any mutant alleles, and the final column capturing the difference between mutants.
This avoids any issues with the use of hard cutoffs when combing lists, such as would occur when comparing separate results from both mutant genotypes against wild-type samples.

```{r setModelMatrix}
d <- model.matrix(~mutant + homozygous, data = dgeList$samples) %>%
  set_colnames(str_remove(colnames(.), "TRUE"))
```

```{r mmVisualisation, fig.height=3, fig.width=6, fig.cap = "*Visualisation of the design matrix*"}
pheatmap(
  d, 
  cluster_cols = FALSE, 
  cluster_rows = FALSE, 
  color = c("white", "grey50"),
  annotation_row = dgeList$samples["genotype"],
  annotation_colors = list(genotype = genoCols),
  legend = FALSE
)
```


### Conditional Quantile Normalisation

As GC content and length was noted as being of concern for this dataset, *conditional-quantile normalisation* was performed using the `cqn` package.
This adds a gene and sample-level offset for each count which takes into account any systemic bias, such as that identified previously as an artefact of variable rRNA removal.
The resultant `glm.offset` values were added to the original `DGEList` object, and all dispersion estimates were calculated. 

```{r gcCqn}
gcCqn <- cqn(
  counts = dgeList$counts,
  x = dgeList$genes$gc_content,
  lengths = dgeList$genes$length,
  sizeFactors = dgeList$samples$lib.size
)
```


```{r plotCQN, fig.cap ="*Model fits for GC content and gene length under the CQN model. Genotype-specific effects are clearly visible.*"}
par(mfrow = c(1, 2))
cqnplot(gcCqn, n = 1, xlab = "GC Content", col = genoCols)
cqnplot(gcCqn, n = 2, xlab = "Length", col = genoCols)
legend("bottomright", legend = levels(samples$genotype), col = genoCols, lty = 1)
par(mfrow = c(1, 1))
```

```{r addOffset}
dgeList$offset <- gcCqn$glm.offset 
dgeList %<>% estimateDisp(design = d)
```


### PCA After Normalisation

```{r cpmPostNorm}
cpmPostNorm <- gcCqn %>%
  with(y + offset)
```

```{r pcaPost}
pcaPost <- cpmPostNorm %>%
  t() %>%
  prcomp() 
pcaVars <- percent_format(0.1)(summary(pcaPost)$importance["Proportion of Variance",])
```


```{r plotPcaPost, fig.width=5, fig.height=4, fig.cap="*PCA of gene-level counts after conditional quantile normalisation.*"}
pcaPost$x %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(samples) %>%
  as_tibble() %>%
  ggplot(aes(PC1, PC2, colour = genotype, fill = genotype)) +
  geom_point() +
  geom_text_repel(aes(label = sampleID), show.legend = FALSE) +
  stat_ellipse(geom = "polygon", alpha = 0.05, show.legend = FALSE) +
  guides(fill = FALSE) +
  scale_colour_manual(
    values = genoCols
  ) +
  labs(
    x = paste0("PC1 (", pcaVars[["PC1"]], ")"),
    y = paste0("PC2 (", pcaVars[["PC2"]], ")"),
    colour = "Genotype"
  )
```

After conditional quantile normalisation, a second PCA analysis was performed in order to check for improvement in group separation.
However, there appeared to minimal improvement at this level, with PC1 still only accounting for `r pcaVars[["PC1"]]` of the variability.

## Model Fitting

```{r fitModel}
minLfc <- log2(2)
alpha <- 0.01
fit <- glmFit(dgeList)
topTables <- colnames(d)[2:3] %>%
  sapply(function(x){
    glmLRT(fit, coef = x) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
      dplyr::select(
        gene_id, gene_name, logFC, logCPM, PValue, FDR, everything()  
      ) %>%
      mutate(
        coef = x,
        bonfP = p.adjust(PValue, "bonf"),
        DE = case_when(
          bonfP < alpha ~ TRUE,
          FDR < alpha & abs(logFC) > minLfc ~ TRUE
        ),
        DE = ifelse(is.na(DE), FALSE, DE)
      )
  }, simplify = FALSE)
```

Models were fit using the negative-binomial approaches of `glmFit()`.
Top Tables were then obtained using likelihood-ratio tests in `glmLRT()`.
These test the standard $H_0$ that the true value of the estimated model coefficient is zero.
These model coefficients effectively estimate:

a. the effect of the presence of a mutation, and 
b. the difference in heterozygous and heterozygous mutants

For enrichment testing, genes were initially considered to be DE using:

a. A Bonferroni-adjusted p-value < `r alpha`
b. An FDR-adjusted p-value < `r alpha` along with an estimated logFC outside of the range $\pm \log_2(`r 2^minLfc`)$.

As fewer genes were detected in the comparisons between homozygous and heterozygous mutants, a simple FDR of 0.05 was subsequently chosen for this comparison.

```{r changeMutantDE}
topTables$homozygous %<>%
  mutate(DE = FDR < 0.05)
```


Using these criteria, the following initial DE gene-sets were defined:

```{r printInitialDE, results='asis'}
topTables %>%
  lapply(dplyr::filter, DE) %>% 
  vapply(nrow, integer(1)) %>%
  set_names(
    case_when(
      names(.) == "mutant" ~ "psen2 mutant",
      names(.) == "homozygous" ~ "HomVsHet"
    )
  ) %>%
  pander()
```

```{r deCols}
deCols <- c(
  `FALSE` = rgb(0.5, 0.5, 0.5, 0.4), 
  `TRUE` = rgb(1, 0, 0, 0.7)
)
```


## Model Checking

## Checks for rRNA impacts

Given the previously identified concerns about variable rRNA removal, correlations were calculated between each gene's expression values and the proportion of the raw libraries with > 70% GC.
Those with the strongest +^ve^ correlation, and for which the FDR is < `r alpha` are shown below. 

Most notably, the most correlated RNA with these rRNA proportions was `eif1b` which is the initiation factor for translation.
As this is a protein coding gene with the *protein product* interacting with the ribosome, the reason for this correlation *at the RNA level* raises difficult questions which cannot be answered here.


```{r riboCorsTable}
topTables$mutant %>%
  mutate(riboCors = riboCors[gene_id]) %>%
  dplyr::filter(
    FDR < alpha
  ) %>%
  dplyr::select(gene_id, gene_name, logFC, logCPM, FDR, riboCors, DE) %>%
  arrange(desc(riboCors)) %>%
  mutate(FDR = case_when(
    FDR >= 0.0001 ~ sprintf("%.4f", FDR),
    FDR < 0.0001 ~ sprintf("%.2e", FDR)
  )
  ) %>%
  dplyr::slice(1:40) %>%
  pander(
    justify = "llrrrrl",
    style = "rmarkdown",
    caption = paste(
      "The", nrow(.), "genes most correlated with the original high GC content.",
      "Many failed the selection criteria for differential expression,",
      "primarily due to the stringent logFC filter.",
      "However, the unusually high number of ribosomal protein coding genes",
      "is currently difficult to explain, but notable.",
      "Two possibilities are these RNAs are somehow translated with physical",
      "linkage to the ribosomal complex, or that there is genuinely a global",
      "shift in the level of ribosomal activity.",
      "Neither explanation is initially satisfying."
    )
  )
```

```{r riboNegativeCorsTable}
topTables$mutant %>%
  mutate(riboCors = riboCors[gene_id]) %>%
  dplyr::filter(
    FDR < alpha
  ) %>%
  dplyr::select(gene_id, gene_name, logFC, logCPM, FDR, riboCors, DE) %>%
  arrange(riboCors) %>%
  mutate(FDR = case_when(
    FDR >= 0.0001 ~ sprintf("%.4f", FDR),
    FDR < 0.0001 ~ sprintf("%.2e", FDR)
  )
  ) %>%
  dplyr::slice(1:20) %>%
  pander(
    justify = "llrrrrl",
    style = "rmarkdown",
    caption = paste(
      "The", nrow(.), "genes most negatively correlated with the original high GC content.",
      "Most failed the selection criteria for differential expression,",
      "primarily due to the stringent logFC filter.",
      "These genes may be being selected against in the presence of rRNA."
    )
  )
```

### GC and Length Bias

```{r checkGCBias, fig.height=5, fig.width=8, fig.cap="*Checks for GC bias in differential expression. GC content is shown against the ranking statistic, using -log10(p) multiplied by the sign of log fold-change. A small amount of bias was noted particularly in the comparison between mutants and wild-type.*"}
topTables %>%
  bind_rows() %>%
  mutate(stat = -sign(logFC)*log10(PValue)) %>%
  ggplot(aes(gc_content, stat)) +
  geom_point(aes(colour = DE), alpha = 0.4) +
  geom_smooth(se = FALSE) +
  facet_wrap(~coef, labeller = contLabeller)  +
  labs(
    x = "GC content (%)",
    y = "Ranking Statistic"
  ) +
  coord_cartesian(ylim = c(-10, 10)) +
  scale_colour_manual(values = deCols) +
  theme(legend.position = "none")
```

```{r checkLengthBias, fig.height=5, fig.width=8, fig.cap="*Checks for length bias in differential expression. Gene length is shown against the ranking statistic, using -log10(p) multiplied by the sign of log fold-change. Again, a small amount of bias was noted particularly in the comparison between mutants and wild-type.*"}
topTables %>%
  bind_rows() %>%
  mutate(stat = -sign(logFC)*log10(PValue)) %>%
  ggplot(aes(length, stat)) +
  geom_point(aes(colour = DE), alpha = 0.4) +
  geom_smooth(se = FALSE) +
  facet_wrap(~coef, labeller = contLabeller)  +
  labs(
    x = "Gene Length (bp)",
    y = "Ranking Statistic"
  ) +
  coord_cartesian(ylim = c(-10, 10)) +
  scale_x_log10(labels = comma) +
  scale_colour_manual(values = deCols) +
  theme(legend.position = "none")
```

Checks for both GC and length bias on differential expression showed that a small bias remained evident, despite using conditional-quantile normalisation.
However, [an alternative analysis on the same dataset **excluding the CQN steps**](2a_NoCQN.html) revealed vastly different and exaggerated bias.
As such, the impact of CQN normalisation was considered to be appropriate.


## Other Artefacts

```{r maPlots, fig.height=8, fig.width=8, fig.cap="*MA plots checking for any logFC bias across the range of expression values. The small curve in the average at the low end of expression values was considered to be an artefact of the sparse points at this end. Initial DE genes are shown in red, with select points labelled.*"}
topTables %>%
  bind_rows() %>%
  arrange(DE) %>%
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = DE)) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %>% dplyr::filter(DE & abs(logFC) > 2.1)
  ) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %>% dplyr::filter(FDR < 0.05 & coef == "homozygous")
  ) +
  geom_smooth(se = FALSE) +
  geom_hline(
    yintercept = c(-1, 1)*minLfc,
    linetype = 2,
    colour = "red"
  ) +
  facet_wrap(~coef, nrow = 2, labeller = contLabeller) +
  scale_y_continuous(breaks = seq(-8, 8, by = 2)) +
  scale_colour_manual(values = deCols) +
  theme(legend.position = "none")
```

```{r pHist, fig.height=4, fig.width=8, fig.cap="*Histograms of p-values for both sets of coefficients. Values for the difference between mutants follow the expected distribution for when there are very few differences, whilst values for the presence of a mutant allele show the expected distribution for when there are many differences. In particular the spike near zero indicates many genes which are differentially expressed.*"}
topTables %>%
  bind_rows() %>%
  ggplot(aes(PValue)) +
  geom_histogram(
    binwidth = 0.02,
    colour = "black", fill = "grey90"
  ) +
  facet_wrap(~coef, labeller = contLabeller)
```


# Results

```{r volcanoPlots, fig.height=5, fig.width=8, fig.cap="*Volcano Plots showing DE genes against logFC.*"}
topTables %>%
  bind_rows() %>%
  ggplot(aes(logFC, -log10(PValue), colour = DE)) +
  geom_point(alpha = 0.4) +
  geom_text_repel(
    aes(label = gene_name),
    data = . %>% dplyr::filter(PValue < 1e-13 | abs(logFC) > 4)
  ) +
  geom_text_repel(
    aes(label = gene_name),
    data = . %>% dplyr::filter(FDR < 0.05 & coef == "homozygous")
  ) +
  geom_vline(
    xintercept = c(-1, 1)*minLfc,
    linetype = 2,
    colour = "red"
  ) +
  facet_wrap(~coef, nrow = 1, labeller = contLabeller) +
  scale_colour_manual(values = deCols) +
  scale_x_continuous(breaks = seq(-8, 8, by = 2)) +
  theme(legend.position = "none") 
```


```{r deGenes}
deGenes <- topTables %>%
  lapply(dplyr::filter, DE) %>%
  lapply(magrittr::extract2, "gene_id") 
```

## Genes Tracking with *psen2^S4Ter^*

```{r n}
n <- 40
```


A set of `r comma(length(deGenes$mutant))` genes was identified as DE in the presence of the mutant *psen2^S4Ter^* transcript.
The most highly ranked `r n` genes are shown in the following heatmap.
The presence of `SNORNA70` in this list is of some concern as this is an ncRNA which is known to interact with ribosomes.
In particular the expression pattern follows very closely the patterns previosuly identified for rRNA depletion using the percentage of libraries with GC > 70.
For example, WT_86 was the sample with most complete rRNA depletion whilst, Het_84 showed the largest proportion of the library as being derived from rRNA.
These represent the samples with lowest and highest expression of SNORNA70 respectively.


```{r commonSummaryHeatmap, fig.height=8, fig.width=8, fig.cap = paste("*The", n, "most highly-ranked genes by FDR which are commonly considered DE between mutants and WT samples. Plotted values are logCPM based on normalised counts.*")}
genoCols <- RColorBrewer::brewer.pal(3, "Set1") %>%
  setNames(levels(dgeList$samples$genotype))
cpmPostNorm %>%
  extract(deGenes$mutant[seq_len(n)],) %>%
  as.data.frame() %>%
  set_rownames(unlist(geneLabeller(rownames(.)))) %>%
  pheatmap::pheatmap(
    color = viridis_pal(option = "magma")(100),
    labels_col = colnames(.) %>%
      str_replace_all(".+(Het|Hom|WT).+F3(_[0-9]{2}).+", "\\1\\2"),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), "logCPM\n"),
    annotation_col = dgeList$samples %>%
      dplyr::select(Genotype = genotype),
    annotation_names_col = FALSE,
    annotation_colors = list(Genotype = genoCols),
    cutree_cols = 2,
    cutree_rows = 4
  )
```

## Genes showing differences between mutants

When inspecting the genes showing differences between mutants, it was noted that the WT and Homozygous mutant samples clustered together, implying that there was a unique effect on these genes which was specific to the heterozygous condition. 

```{r homozygousHeatmap, fig.height=4, fig.width=7, fig.cap = paste("*The", length(deGenes$homozygous), "most highly-ranked genes by FDR which are DE between between mutants. Plotted values are logCPM based on normalised counts.*")}
cpmPostNorm %>%
  extract(deGenes$homozygous,) %>%
  as.data.frame() %>%
  set_rownames(unlist(geneLabeller(rownames(.)))) %>%
  pheatmap::pheatmap(
    color = viridis_pal(option = "magma")(100),
    labels_col = colnames(.) %>%
      str_replace_all(".+(Het|Hom|WT).+F3(_[0-9]{2}).+", "\\1\\2"),
    legend_breaks = c(seq(-2, 8, by = 2), max(.)),
    legend_labels = c(seq(-2, 8, by = 2), "logCPM\n"),
    annotation_col = dgeList$samples %>%
      dplyr::select(Genotype = genotype),
    annotation_names_col = FALSE,
    annotation_colors = list(Genotype = genoCols),
    cutree_cols = 2,
    cutree_rows = 3
  )
```

# Data Export

Final gene lists were exported as separate csv files, along with  fit and dgeList objects.

```{r writeResults}
write_csv(
  x = topTables$mutant, 
  path = here::here("output/psen2VsWT.csv")
)
write_csv(
  x = topTables$homozygous,
  path = here::here("output/psen2HomVsHet.csv")
)
write_rds(
  x = fit, 
  path = here::here("data/fit.rds"),
  compress = "gz"
)
write_rds(
  x = dgeList,
  path = here::here("data/dgeList.rds"), 
  compress = "gz"
)
write_rds(
  x = cpmPostNorm, 
  path = here::here("data/cpmPostNorm.rds"),
  compress = "gz"
)
```


