---
title: "Analysis Without Using CQN"
author: "Steve Pederson"
date: "2020-01-24"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
    echo = TRUE,
    warning = FALSE,
    message = FALSE,
    fig.align = "center"
)
```

# Introduction

This workflow is an alternative differential gene expression analysis, however *conditional quantile normalisation* was not used.
This was written up specifically to allow for a checking of the impact this method has had on the dataset.


```{r loadPackages}
library(ngsReports)
library(tidyverse)
library(magrittr)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(scales)
library(pander)
library(cowplot)
library(cqn)
library(ggrepel)
library(UpSetR)
```

```{r setOptions}
if (interactive()) setwd(here::here())
theme_set(theme_bw())
panderOptions("big.mark", ",")
panderOptions("table.split.table", Inf)
panderOptions("table.style", "rmarkdown")
twoCols <- c(rgb(0.8, 0.1, 0.1), rgb(0.2, 0.2, 0.8))
```

## Annotations

```{r annotationSetup}
ah <- AnnotationHub() %>%
    subset(species == "Danio rerio") %>%
    subset(rdataclass == "EnsDb")
ensDb <- ah[["AH74989"]]
```

```{r transAnnotation}
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
    width() %>%
    vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]
```


```{r geneAnnotation}
gcGene <- grTrans %>%
    mcols() %>%
    as.data.frame() %>%
    dplyr::select(gene_id, tx_id, gc_content, length) %>%
    as_tibble() %>%
    group_by(gene_id) %>%
    summarise(
        gc_content = sum(gc_content*length) / sum(length),
        length = ceiling(median(length))
    )
grGenes <- genes(ensDb)
mcols(grGenes) %<>%
    as.data.frame() %>%
    left_join(gcGene) %>%
    as.data.frame() %>%
    DataFrame()
```

Similarly to the Quality Assessment steps, `GRanges` objects were formed at the gene and transcript levels, to enable estimation of GC content and length for each transcript and gene.
GC content and transcript length are available for each transcript, and for gene-level estimates, GC content was taken as the sum of all GC bases divided by the sum of all transcript lengths, effectively averaging across all transcripts.
Gene length was defined as the median transcript length.


```{r samplesAndLabels}
samples <- read_csv("data/samples.csv") %>%
    distinct(sampleName, .keep_all = TRUE) %>%
    dplyr::select(sample = sampleName, sampleID, genotype) %>%
    mutate(genotype = factor(genotype, levels = c("WT", "Het", "Hom")))
```

Sample metadata was also loaded, with only the sampleID and genotype being retained. 
All other fields were considered irrelevant.

## Count Data

```{r dgeList}
minCPM <- 1.5
minSamples <- 4
dgeList <- file.path("data", "2_alignedData", "featureCounts", "genes.out") %>%
    read_delim(delim = "\t") %>%
    set_names(basename(names(.))) %>%
    as.data.frame() %>%
    column_to_rownames("Geneid") %>%
    as.matrix() %>% 
    set_colnames(str_remove(colnames(.), "Aligned.sortedByCoord.out.bam")) %>%
    .[rowSums(cpm(.) >= minCPM) >= minCPM,] %>%
    DGEList(
        samples = tibble(sample = colnames(.)) %>%
            left_join(samples),
        genes = grGenes[rownames(.)] %>%
            as.data.frame() %>%
            dplyr::select(
                chromosome = seqnames, start, end, 
                gene_id, gene_name, gene_biotype, description, 
                entrezid, gc_content, length
            )
    ) %>%
    .[!grepl("rRNA", .$genes$gene_biotype),] %>%
    calcNormFactors()
```

Gene-level count data as output by `featureCounts`, was loaded and formed into a `DGEList` object.
During this process, genes were removed if:

- They were not considered as detectable (CPM < `r minCPM` in > `r ncol(dgeList) - minSamples` samples). This translates to > `r ceiling(min(minCPM * dgeList$samples$lib.size/1e6))` reads assigned a gene in all samples from one or more of the genotype groups
- The `gene_biotype` was any type of `rRNA`.

These filtering steps returned gene-level counts for `r comma(nrow(dgeList))` genes, with total library sizes between `r pander(comma(range(dgeList$samples$lib.size)))` reads assigned to genes.
It was noted that these library sizes were about 1.5-fold larger than the transcript-level counts used for the QA steps.


```{r plotDensities, fig.width=5, fig.height=4, fig.cap="*Expression density plots for all samples after filtering, showing logCPM values.*"}
cpm(dgeList, log = TRUE) %>%
    as.data.frame() %>%
    pivot_longer(
        cols = everything(),
        names_to = "sample",
        values_to = "logCPM"
    ) %>%
    split(f = .$sample) %>%
    lapply(function(x){
        d <- density(x$logCPM)
        tibble(
            sample = unique(x$sample),
            x = d$x,
            y = d$y
        )
    }) %>%
    bind_rows() %>%
    left_join(samples) %>%
    ggplot(aes(x, y, colour = genotype, group = sample)) +
    geom_line() +
    labs(
        x = "logCPM",
        y = "Density",
        colour = "Genotype"
    )
```

### Additional Functions

```{r labellers}
contLabeller <- as_labeller(
    c(
        HetVsWT = "S4Ter/+ Vs +/+",
        HomVsWT = "S4Ter/S4Ter Vs +/+",
        HomVsHet = "S4Ter/S4Ter Vs S4Ter/+",
        Hom = "S4Ter/S4Ter",
        Het = "S4Ter/+",
        WT = "+/+"
    )
)
geneLabeller <- structure(grGenes$gene_name, names = grGenes$gene_id) %>%
    as_labeller()
```

Labeller functions for genotypes, contrasts and gene names were additionally defined for simpler plotting using `ggplot2`.

# Analysis Using NB Models


## Model Description

The same model was applied as for the analysis using CQN.

```{r setModelMatrix}
d <- model.matrix(~ 0 + genotype, data = dgeList$samples) %>%
    set_colnames(str_remove_all(colnames(.), "genotype"))
cont <- makeContrasts(
    HetVsWT = Het - WT,
    HomVsWT = Hom - WT,
    HomVsHet = Hom - Het,
    levels = d
)
```


### Normalisation

No GC normalisation was included in this workflow.
Instead, dispersions were calculated using the model matrix as defined in the full workflow.


```{r addOffset}
dgeList %<>% estimateDisp(design = d)
```


## Model Fitting

```{r fitModel}
minLfc <- log2(1.5)
alpha <- 0.01
fit <- glmFit(dgeList)
topTables <- colnames(cont) %>%
    sapply(function(x){
        glmLRT(fit, contrast = cont[,x]) %>%
            topTags(n = Inf) %>%
            .[["table"]] %>%
            as_tibble() %>%
            dplyr::select(
                gene_id, gene_name, logFC, logCPM, PValue, FDR, everything()  
            ) %>%
            mutate(
                comparison = x,
                DE = FDR < alpha & abs(logFC) > minLfc
            )
    },
    simplify = FALSE) 
```

Models were fit using the negative-binomial approaches of `glmFit()`.
Top Tables were then obtained using pair-wise likelihood-ratio tests in `glmLRT()`.
These test the standard $H_0$ that there is no difference in gene expression estimates between genotypes, the gene expression estimates are obtained under the negative binomial model.

```{r expandDeLists}
alpha2 <- 0.05
topTables %<>% 
  bind_rows() %>% 
  split(f = .$gene_id) %>% 
  lapply(function(x){mutate(x, DE = any(DE) & FDR < alpha2)}) %>%
  bind_rows() %>%
  split(f = .$comparison)
```

**In order to remain as comparable as possible, the same secondary gene selection steps were performed as for the main analysis.**

For enrichment testing, genes were initially considered to be DE using an estimated logFC outside of the range $\pm \log_2(`r 2^minLfc`)$ and an FDR-adjusted p-value < `r alpha`.
For genes in any of these initial lists, the logFC filter was subsequently removed from subsequent comparisons in order to minimise issues introduced by the use of a hard cutoff.
Similarly the FDR threshold was raised to `r alpha2` in secondary comparisons for genes which passed the initial round of selection.

Using these criteria, the following initial DE gene-sets were defined. These were slightly higher than previously

```{r printInitialDE, results='asis'}
topTables %>%
  lapply(dplyr::filter, DE) %>% 
  vapply(nrow, integer(1)) %>%
  pander()
```


## Model Checking

```{r maPlots, fig.height=5, fig.width=10, fig.cap="*MA plots checking for any logFC bias across the range of expression values. Both mutant comparisons against wild-type appear to show a biased relationship between logFC and expression level.. Initial DE genes are shown in red, with select points labelled.*"}
topTables %>%
  bind_rows() %>%
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = DE), alpha = 0.4) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %>% dplyr::filter(DE & abs(logFC) > 3)
  ) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %>% dplyr::filter(FDR < 0.05 & comparison == "HomVsHet")
  ) +
  geom_smooth(se = FALSE) +
  geom_hline(
    yintercept = c(-1, 1)*minLfc,
    linetype = 2,
    colour = "red"
  ) +
  facet_wrap(~comparison, nrow = 1, labeller = contLabeller) +
  scale_y_continuous(breaks = seq(-8, 8, by = 2)) +
  scale_colour_manual(values = c("grey50", "red")) +
  theme(legend.position = "none")
```


```{r checkGCBias, fig.height=5, fig.width=10, fig.cap="*Checks for GC bias in differential expression. GC content is shown against the ranking statistic, using -log10(p) multiplied by the sign of log fold-change. A large amount of bias was noted particularly in the comparison between homozygous mutants and wild-type.*"}
topTables %>%
    bind_rows() %>%
    mutate(stat = -sign(logFC)*log10(PValue)) %>%
    ggplot(aes(gc_content, stat)) +
    geom_point(aes(colour = DE), alpha = 0.4) +
    geom_smooth(se = FALSE) +
    facet_wrap(~comparison, labeller = contLabeller)  +
    labs(
        x = "GC content (%)",
        y = "Ranking Statistic"
    ) +
    coord_cartesian(ylim = c(-10, 10)) +
    scale_colour_manual(values = c("grey50", "red")) +
    theme(legend.position = "none")
```

```{r checkLengthBias, fig.height=5, fig.width=10, fig.cap="*Checks for length bias in differential expression. Gene length is shown against the ranking statistic, using -log10(p) multiplied by the sign of log fold-change. Again, a large amount of bias was noted particularly in the comparison between homozygous mutants and wild-type.*"}
topTables %>%
    bind_rows() %>%
    mutate(stat = -sign(logFC)*log10(PValue)) %>%
    ggplot(aes(length, stat)) +
    geom_point(aes(colour = DE), alpha = 0.4) +
    geom_smooth(se = FALSE) +
    facet_wrap(~comparison, labeller = contLabeller)  +
    labs(
        x = "Gene Length (bp)",
        y = "Ranking Statistic"
    ) +
    coord_cartesian(ylim = c(-10, 10)) +
    scale_x_log10(labels = comma) +
    scale_colour_manual(values = c("grey50", "red")) +
    theme(legend.position = "none")
```


# Analysis Using Voom

As a final alternative, the dataset was fit using `voomWithQualityWeights()`.
Given that two samples were relatively divergent from the remainder of the samples, in terms of their rRNA depletion, this strategy may resolve some of these issues.

```{r voomFit}
voomData <- dgeList %>%
    voomWithQualityWeights(design = matrix(1, nrow = ncol(.)))
voomFit <- voomData %>%
    lmFit(design = d) %>%
    contrasts.fit(cont) %>%
    eBayes()
```


```{r plotWeights, fig.height=5, fig.width=10, fig.cap="*Sample-level weights after applying voom. As expected, the highest and lowest samples from the initial rRNA analysis were down-weighted the most strongly*"}
voomData$targets %>%
    ggplot(aes(sampleID, sample.weights, fill = genotype)) +
    geom_bar(stat = "identity") +
    facet_wrap(~genotype, labeller = contLabeller, scales = "free_x") +
    theme(legend.position = "none")
```


```{r voomTables}
voomTables <- colnames(cont) %>%
    sapply(function(x){
        topTable(voomFit, coef = x, number = Inf) %>%
            as_tibble() %>%
            dplyr::select(
                gene_id, gene_name, logFC, AveExpr, P.Value, FDR = adj.P.Val, everything()
            ) %>%
            mutate(
                comparison = x,
                DE = FDR < alpha & abs(logFC) > minLfc
            )
    },
    simplify = FALSE) 
```

## Model Checking

```{r voomMaPlots, fig.height=5, fig.width=10, fig.cap="*MA plots checking for any logFC bias across the range of expression values. Both mutant comparisons against wild-type appear to show a biased relationship between logFC and expression level.. Initial DE genes are shown in red, with select points labelled.*"}
voomTables %>%
  bind_rows() %>%
  ggplot(aes(AveExpr, logFC)) +
  geom_point(aes(colour = DE), alpha = 0.4) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %>% dplyr::filter(DE & abs(logFC) > 3)
  ) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %>% dplyr::filter(FDR < 0.05 & comparison == "HomVsHet")
  ) +
  geom_smooth(se = FALSE) +
  geom_hline(
    yintercept = c(-1, 1)*minLfc,
    linetype = 2,
    colour = "red"
  ) +
  facet_wrap(~comparison, nrow = 1, labeller = contLabeller) +
  scale_y_continuous(breaks = seq(-8, 8, by = 2)) +
  scale_colour_manual(values = c("grey50", "red")) +
  theme(legend.position = "none")
```


```{r voomGCBias, fig.height=5, fig.width=10, fig.cap="*Checks for GC bias in differential expression. GC content is shown against the ranking t-statistic. A large amount of bias was noted particularly in the comparison between homozygous mutants and wild-type.*"}
voomTables %>%
    bind_rows() %>%
    ggplot(aes(gc_content, t)) +
    geom_point(aes(colour = DE), alpha = 0.4) +
    geom_smooth(se = FALSE) +
    facet_wrap(~comparison, labeller = contLabeller)  +
    labs(
        x = "GC content (%)",
        y = "Ranking Statistic (t)"
    ) +
    coord_cartesian(ylim = c(-10, 10)) +
    scale_colour_manual(values = c("grey50", "red")) +
    theme(legend.position = "none")
```

```{r cvoomLengthBias, fig.height=5, fig.width=10, fig.cap="*Checks for length bias in differential expression. Gene length is shown against the ranking t-statistic. Again, a large amount of bias was noted particularly in the comparison between homozygous mutants and wild-type.*"}
voomTables %>%
    bind_rows() %>%
    ggplot(aes(length, t)) +
    geom_point(aes(colour = DE), alpha = 0.4) +
    geom_smooth(se = FALSE) +
    facet_wrap(~comparison, labeller = contLabeller)  +
    labs(
        x = "Gene Length (bp)",
        y = "Ranking Statistic (t)"
    ) +
    coord_cartesian(ylim = c(-10, 10)) +
    scale_x_log10(labels = comma) +
    scale_colour_manual(values = c("grey50", "red")) +
    theme(legend.position = "none")
```

# Conclusion

None of the results in this workflow were for analysis, but were simply to assess the impact of GC and length bias without accounting for it using CQN.
