---
title: "Generate XML For ENA Submission"
author: "Steve Pederson"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
    echo = TRUE,
    warning = FALSE,
    message = FALSE,
    fig.align = "center"
)
```

```{r}
library(tidyverse)
library(xml2)
library(glue)
```


# Introduction

This is an attempt to automatically generate the XML files required for submission to ENA.
Hopefully once complete, this be beneficial for later submissions.

# Generataion of XML Files

## Study XML

This is the first step required and outlines the entire experiment.

```{r}
alias <- "psen2S4Ter"
title <- glue(
    "Transcriptomic analysis of premature termination codon mutation in the zebrafish gene psen2"
)
description <- glue(
    "This study compared gene expression patterns in whole brain samples from zebrafish.
    Samples were homozygous or heterozygous for the psen2S4Ter mutation, or were wild-type at this locus."
)
```


```{r}
glue(
    "<PROJECT_SET>
        <PROJECT alias='{alias}'>
            <TITLE>{title}</TITLE>
            <DESCRIPTION>{description}</DESCRIPTION>
            <SUBMISSION_PROJECT>
                <SEQUENCING_PROJECT/>
            </SUBMISSION_PROJECT>
        </PROJECT>
     </PROJECT_SET>"
) %>%
    as.character() %>%
    write_lines(
        here::here ("output/Study.XML")
    )
```


## Sample XML

